<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Changelog • recount</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Changelog"><meta property="og:image" content="/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-news">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">recount</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.27.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/fakeVignettes/SRP009615-results.html">UNKNOWN TITLE</a>
    </li>
    <li>
      <a href="../articles/recount-quickstart.html">recount quick start guide</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/leekgroup/recount/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
      <h1 data-toc-skip>Changelog <small></small></h1>
      <small>Source: <a href="https://github.com/leekgroup/recount/blob/HEAD/NEWS.md" class="external-link"><code>NEWS.md</code></a></small>
    </div>

    <div class="section level2">
<h2 class="page-header" data-toc-text="1.19.2" id="recount-1192">recount 1.19.2<a class="anchor" aria-label="anchor" href="#recount-1192"></a></h2>
<p>BUG FIXES</p>
<ul><li>Fix a bug in <code><a href="../reference/geo_info.html">geo_info()</a></code> for reading files on Windows where a trailing <code>\r</code> was added to all variables.</li>
<li>Avoid the <code>implicit list embedding of S4 objects is deprecated</code> warning that was noted at <a href="https://github.com/leekgroup/recount/runs/3286046827?check_suite_focus=true#step:20:1417" class="external-link uri">https://github.com/leekgroup/recount/runs/3286046827?check_suite_focus=true#step:20:1417</a>.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.13.2" id="recount-1132">recount 1.13.2<a class="anchor" aria-label="anchor" href="#recount-1132"></a></h2>
<p>SIGNIFICANT USER-VISIBLE CHANGES</p>
<ul><li>Documentation website is now available at <a href="http://leekgroup.github.io/recount/" class="external-link uri">http://leekgroup.github.io/recount/</a>. It gets updated with every commit on the master branch (bioc-devel) using GitHub Actions and pkgdown.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.13.1" id="recount-1131">recount 1.13.1<a class="anchor" aria-label="anchor" href="#recount-1131"></a></h2>
<ul><li>Mention in <code><a href="../reference/reproduce_ranges.html">reproduce_ranges()</a></code> the link to <a href="https://support.bioconductor.org/p/126148/#126173" class="external-link uri">https://support.bioconductor.org/p/126148/#126173</a> which shows how to update the gene symbols in the RSE objects in recount.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.11.14" id="recount-11114">recount 1.11.14<a class="anchor" aria-label="anchor" href="#recount-11114"></a></h2>
<ul><li>Added a <code>NEWS.md</code> file to track changes to the package.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.11.13" id="recount-11113">recount 1.11.13<a class="anchor" aria-label="anchor" href="#recount-11113"></a></h2>
<p>NEW FEATURES</p>
<ul><li>Added the function <code><a href="../reference/getTPM.html">getTPM()</a></code> as discussed in <a href="https://support.bioconductor.org/p/124265" class="external-link uri">https://support.bioconductor.org/p/124265</a> and based on Sonali Arora et al <a href="https://www.biorxiv.org/content/10.1101/445601v2" class="external-link uri">https://www.biorxiv.org/content/10.1101/445601v2</a>.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.11.12" id="recount-11112">recount 1.11.12<a class="anchor" aria-label="anchor" href="#recount-11112"></a></h2>
<p>BUG FIXES</p>
<ul><li>Now <code><a href="../reference/geo_characteristics.html">geo_characteristics()</a></code> can deal with the scenario reported at <a href="https://support.bioconductor.org/p/116480/" class="external-link uri">https://support.bioconductor.org/p/116480/</a> by <a href="https://github.com/Jacques" class="external-link">@Jacques</a>.van-Helden.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.11.7" id="recount-1117">recount 1.11.7<a class="anchor" aria-label="anchor" href="#recount-1117"></a></h2>
<p>SIGNIFICANT USER-VISIBLE CHANGES</p>
<ul><li>Renamed <code>.load_install()</code> as <code>.load_check()</code> as this function now only checks that the package(s) was installed and returns an error if missing. The error shows the user how to install the package(s) they are missing instead of installing them automatically. This complies with Marcel Ramos’ request at <a href="https://github.com/leekgroup/recount/issues/14" class="external-link uri">https://github.com/leekgroup/recount/issues/14</a>.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.11.4" id="recount-1114">recount 1.11.4<a class="anchor" aria-label="anchor" href="#recount-1114"></a></h2>
<p>NEW FEATURES</p>
<ul><li>Added the function <code><a href="../reference/download_retry.html">download_retry()</a></code> based on <a href="http://bioconductor.org/developers/how-to/web-query/" class="external-link uri">http://bioconductor.org/developers/how-to/web-query/</a> such that <code>download_file()</code> and other recount functions will re-try to download a file 3 times before giving up. This should help reduce the number of occasional failed Bioconductor nightly checks.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.9.7" id="recount-197">recount 1.9.7<a class="anchor" aria-label="anchor" href="#recount-197"></a></h2>
<p>SIGNIFICANT USER-VISIBLE CHANGES</p>
<ul><li>Cleaned up the documentation of <code><a href="../reference/add_metadata.html">add_metadata()</a></code> and changed the default <code>source</code> from <code>recount_brain_v1</code> to <code>recount_brain_v2</code>.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.9.6" id="recount-196">recount 1.9.6<a class="anchor" aria-label="anchor" href="#recount-196"></a></h2>
<p>NEW FEATURES</p>
<ul><li>
<code>citation('recount')[5]</code> now lists the <code>recount-brain</code> bioRxiv pre-print citation information.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.9.5" id="recount-195">recount 1.9.5<a class="anchor" aria-label="anchor" href="#recount-195"></a></h2>
<p>NEW FEATURES</p>
<ul><li>We have now released the FANTOM-CAT/recount2 RSE files which you can access now through <code>download_study(type = 'rse-fc')</code>. See Imada EL, Sanchez DF, et al, bioRxiv, 2019 <a href="https://www.biorxiv.org/content/10.1101/659490v1" class="external-link uri">https://www.biorxiv.org/content/10.1101/659490v1</a> for more information.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.9.4" id="recount-194">recount 1.9.4<a class="anchor" aria-label="anchor" href="#recount-194"></a></h2>
<p>BUG FIXES</p>
<ul><li>I made the example code for <code><a href="../reference/geo_characteristics.html">geo_characteristics()</a></code> more robust since currently <code>rentrez</code> can occasionally fails.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.9.3" id="recount-193">recount 1.9.3<a class="anchor" aria-label="anchor" href="#recount-193"></a></h2>
<p>NEW FEATURES</p>
<ul><li>Add ORCID’s following changes at <a href="http://bioconductor.org/developers/package-guidelines/#description" class="external-link uri">http://bioconductor.org/developers/package-guidelines/#description</a>
</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.9.2" id="recount-192">recount 1.9.2<a class="anchor" aria-label="anchor" href="#recount-192"></a></h2>
<p>BUG FIXES</p>
<ul><li>Added the argument <code>async</code> to <code><a href="../reference/snaptron_query.html">snaptron_query()</a></code> which can be set to <code>FALSE</code> to address the issue reported at <a href="https://github.com/ChristopherWilks/snaptron/issues/11" class="external-link uri">https://github.com/ChristopherWilks/snaptron/issues/11</a>
</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.7.7" id="recount-177">recount 1.7.7<a class="anchor" aria-label="anchor" href="#recount-177"></a></h2>
<p>BUG FIXES</p>
<ul><li>Updated <code><a href="../reference/reproduce_ranges.html">reproduce_ranges()</a></code> to match the <code>URL</code> change in Gencode from <a href="ftp://ftp.sanger.ac.uk" class="uri">ftp://ftp.sanger.ac.uk</a> to <a href="ftp://ftp.ebi.ac.uk" class="uri">ftp://ftp.ebi.ac.uk</a>
</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.7.5" id="recount-175">recount 1.7.5<a class="anchor" aria-label="anchor" href="#recount-175"></a></h2>
<p>SIGNIFICANT USER-VISIBLE CHANGES</p>
<ul><li>
<code><a href="../reference/add_metadata.html">add_metadata()</a></code> can now download the <code>recount_brain_v2</code> data.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.7.4" id="recount-174">recount 1.7.4<a class="anchor" aria-label="anchor" href="#recount-174"></a></h2>
<p>BUG FIXES</p>
<ul><li>Fix a <code>NOTE</code> about <code>RefManageR</code>.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.7.3" id="recount-173">recount 1.7.3<a class="anchor" aria-label="anchor" href="#recount-173"></a></h2>
<p>SIGNIFICANT USER-VISIBLE CHANGES</p>
<ul><li>Use <code>BiocManager</code>
</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.7.2" id="recount-172">recount 1.7.2<a class="anchor" aria-label="anchor" href="#recount-172"></a></h2>
<p>BUG FIXES</p>
<ul><li>Fix a unit test.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.7.1" id="recount-171">recount 1.7.1<a class="anchor" aria-label="anchor" href="#recount-171"></a></h2>
<p>SIGNIFICANT USER-VISIBLE CHANGES</p>
<ul><li>
<code>rse_tx</code> URLs now point to v2 to reflect recent changes by Fu et al.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.5.11" id="recount-1511">recount 1.5.11<a class="anchor" aria-label="anchor" href="#recount-1511"></a></h2>
<p>BUG FIXES</p>
<ul><li>Change some examples to dontrun and improve the code that cleans up after the tests. This should reduce the size of files left in tmp although they didn’t seem too big to begin with.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.5.9" id="recount-159">recount 1.5.9<a class="anchor" aria-label="anchor" href="#recount-159"></a></h2>
<p>SIGNIFICANT USER-VISIBLE CHANGES</p>
<ul><li>The functions <code><a href="../reference/add_metadata.html">add_metadata()</a></code> and <code><a href="../reference/add_predictions.html">add_predictions()</a></code> now return the sample metadata or predictions when the <code>rse</code> argument is missing.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.5.6" id="recount-156">recount 1.5.6<a class="anchor" aria-label="anchor" href="#recount-156"></a></h2>
<p>NEW FEATURES</p>
<ul><li>Added the function <code><a href="../reference/add_metadata.html">add_metadata()</a></code> which can be used to append curated metadata to a recount rse object. Currently, <code><a href="../reference/add_metadata.html">add_metadata()</a></code> only supports the <code>recount_brain_v1</code> data available at <a href="http://lieberinstitute.github.io/recount-brain/" class="external-link uri">http://lieberinstitute.github.io/recount-brain/</a> and to be further described in Razmara et al, in prep, 2018.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.5.5" id="recount-155">recount 1.5.5<a class="anchor" aria-label="anchor" href="#recount-155"></a></h2>
<p>BUG FIXES</p>
<ul><li>Fix doc link in <code><a href="../reference/geo_characteristics.html">geo_characteristics()</a></code> which affected the Windows build machines.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.5.4" id="recount-154">recount 1.5.4<a class="anchor" aria-label="anchor" href="#recount-154"></a></h2>
<p>BUG FIXES</p>
<ul><li>Fix a unit test for <code><a href="../reference/download_study.html">download_study()</a></code>, add another test for the versions, and fix a <code>NOTE</code> in <code>R CMD check</code>.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.5.3" id="recount-153">recount 1.5.3<a class="anchor" aria-label="anchor" href="#recount-153"></a></h2>
<p>NEW FEATURES</p>
<ul><li>
<code><a href="../reference/download_study.html">download_study()</a></code> can now download the transcript counts (<code>rse_tx.RData</code>) files. The transcript estimation is described in Fu et al, 2018.</li>
</ul><p>SIGNIFICANT USER-VISIBLE CHANGES</p>
<ul><li>
<code><a href="../reference/download_study.html">download_study()</a></code> now has a version parameter (defaults to 2). This argument controls which version of the files to download based on the change on how exons were defined. Version 1 are reduced exons while version 2 are disjoint exons as described in further detail in the documentation tab of the recount website <a href="https://jhubiostatistics.shinyapps.io/recount/" class="external-link uri">https://jhubiostatistics.shinyapps.io/recount/</a>.</li>
<li>
<code>recount_url</code> and the example <code>rse_gene_SRP009615</code> have been updated to match the changes in version 2.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.3.13" id="recount-1313">recount 1.3.13<a class="anchor" aria-label="anchor" href="#recount-1313"></a></h2>
<p>BUG FIXES</p>
<ul><li>Changed <code><a href="../reference/reproduce_ranges.html">reproduce_ranges()</a></code> since disjoint exons are more useful than reduced exons for downstream analyses.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.3.12" id="recount-1312">recount 1.3.12<a class="anchor" aria-label="anchor" href="#recount-1312"></a></h2>
<p>NEW FEATURES</p>
<ul><li>Added the function <code><a href="../reference/read_counts.html">read_counts()</a></code>.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.3.9" id="recount-139">recount 1.3.9<a class="anchor" aria-label="anchor" href="#recount-139"></a></h2>
<p>SIGNIFICANT USER-VISIBLE CHANGES</p>
<ul><li>Added citations for <a href="https://www.biorxiv.org/content/early/2017/06/03/145656" class="external-link uri">https://www.biorxiv.org/content/early/2017/06/03/145656</a> and <a href="https://f1000research.com/articles/6-1558/v1" class="external-link uri">https://f1000research.com/articles/6-1558/v1</a> as well as mentions to them in the vignette.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.3.7" id="recount-137">recount 1.3.7<a class="anchor" aria-label="anchor" href="#recount-137"></a></h2>
<p>SIGNIFICANT USER-VISIBLE CHANGES</p>
<ul><li>
<code><a href="../reference/add_predictions.html">add_predictions()</a></code> was bumped to version 0.0.05</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.3.5" id="recount-135">recount 1.3.5<a class="anchor" aria-label="anchor" href="#recount-135"></a></h2>
<p>SIGNIFICANT USER-VISIBLE CHANGES</p>
<ul><li>Vignette now uses the new <code><a href="https://rdrr.io/pkg/BiocStyle/man/html_document.html" class="external-link">BiocStyle::html_document</a></code> that was recently released.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.3.2" id="recount-132">recount 1.3.2<a class="anchor" aria-label="anchor" href="#recount-132"></a></h2>
<p>NEW FEATURES</p>
<ul><li>
<code><a href="../reference/coverage_matrix.html">coverage_matrix()</a></code> now has two new arguments: <code>scale</code> and <code>round</code>. Use <code>scale = FALSE</code> to get raw coverage counts, which you can then scale with <code><a href="../reference/scale_counts.html">scale_counts()</a></code>. <code>scale</code> is set to <code>TRUE</code> by default, so the counts are scaled to a library size of 40 million reads. <code>round</code> is set to <code>FALSE</code> by default, but can be set to <code>TRUE</code> if you want to get integer counts, just as in the default of <code><a href="../reference/scale_counts.html">scale_counts()</a></code>.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.3.1" id="recount-131">recount 1.3.1<a class="anchor" aria-label="anchor" href="#recount-131"></a></h2>
<p>SIGNIFICANT USER-VISIBLE CHANGES</p>
<ul><li>Changed the default version argument of <code><a href="../reference/add_predictions.html">add_predictions()</a></code> to <code>latest</code>. Internally, that’s still 0.0.03.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.1.27" id="recount-1127">recount 1.1.27<a class="anchor" aria-label="anchor" href="#recount-1127"></a></h2>
<p>NEW FEATURES</p>
<ul><li>Added the <code><a href="../reference/add_predictions.html">add_predictions()</a></code> function which appends the predicted phenotypes to a RSE object downloaded with recount. The phenotypes were predicted by Shannon Ellis et al, 2017 (citation coming up soon!).</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.1.26" id="recount-1126">recount 1.1.26<a class="anchor" aria-label="anchor" href="#recount-1126"></a></h2>
<p>SIGNIFICANT USER-VISIBLE CHANGES</p>
<ul><li>Changed the citation now that the recount2 paper has been published at <a href="http://www.nature.com/nbt/journal/v35/n4/full/nbt.3838.html" class="external-link uri">http://www.nature.com/nbt/journal/v35/n4/full/nbt.3838.html</a>.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.1.25" id="recount-1125">recount 1.1.25<a class="anchor" aria-label="anchor" href="#recount-1125"></a></h2>
<p>NEW FEATURES</p>
<ul><li>Added the function <code><a href="../reference/getRPKM.html">getRPKM()</a></code> which can be used with <code>RangedSummarizedExperiment</code> objects from <code>recount</code> and from other sources.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.1.24" id="recount-1124">recount 1.1.24<a class="anchor" aria-label="anchor" href="#recount-1124"></a></h2>
<p>SIGNIFICANT USER-VISIBLE CHANGES</p>
<ul><li>
<code>recount_url</code> now includes the URLs for the GTEx bigWig files.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.1.19" id="recount-1119">recount 1.1.19<a class="anchor" aria-label="anchor" href="#recount-1119"></a></h2>
<p>SIGNIFICANT USER-VISIBLE CHANGES</p>
<ul><li>
<code><a href="../reference/coverage_matrix.html">coverage_matrix()</a></code> now returns a RangedSummarizedExperiment object. This matches the behavior of <code>recount.bwtool::coverage_matrix_bwtool()</code> and is more consistent with the use of RSE objects in recount.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.1.18" id="recount-1118">recount 1.1.18<a class="anchor" aria-label="anchor" href="#recount-1118"></a></h2>
<p>BUG FIXES</p>
<ul><li>
<code><a href="../reference/coverage_matrix.html">coverage_matrix()</a></code>’s helper function <code>.read_pheno()</code> was failing for some projects.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.1.16" id="recount-1116">recount 1.1.16<a class="anchor" aria-label="anchor" href="#recount-1116"></a></h2>
<p>BUG FIXES</p>
<ul><li>Fixed a bug in the counts in <code><a href="../reference/coverage_matrix.html">coverage_matrix()</a></code>. They were being incorrectly multiplied by 100.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.1.14" id="recount-1114-1">recount 1.1.14<a class="anchor" aria-label="anchor" href="#recount-1114-1"></a></h2>
<p>SIGNIFICANT USER-VISIBLE CHANGES</p>
<ul><li>Completed the change to Gencode v25 annotation for exon and gene counts.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.1.13" id="recount-1113">recount 1.1.13<a class="anchor" aria-label="anchor" href="#recount-1113"></a></h2>
<p>SIGNIFICANT USER-VISIBLE CHANGES</p>
<ul><li>We dropped <code>TxDb.Hsapiens.UCSC.hg38.knownGene</code> completely from <code>recount</code> and will be using Gencode v25 instead.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.1.12" id="recount-1112">recount 1.1.12<a class="anchor" aria-label="anchor" href="#recount-1112"></a></h2>
<p>BUG FIXES</p>
<ul><li>Updated <code><a href="../reference/snaptron_query.html">snaptron_query()</a></code> to comply with recent changes in Snaptron.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.1.8" id="recount-118">recount 1.1.8<a class="anchor" aria-label="anchor" href="#recount-118"></a></h2>
<p>SIGNIFICANT USER-VISIBLE CHANGES</p>
<ul><li>Updated the package so you can now access TCGA data. Now there’s over 8 terabytes of data available in the <code>recount</code> project!</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.1.6" id="recount-116">recount 1.1.6<a class="anchor" aria-label="anchor" href="#recount-116"></a></h2>
<p>SIGNIFICANT USER-VISIBLE CHANGES</p>
<ul><li>
<code><a href="../reference/snaptron_query.html">snaptron_query()</a></code> can now access GTEx and TCGA data.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.1.5" id="recount-115">recount 1.1.5<a class="anchor" aria-label="anchor" href="#recount-115"></a></h2>
<p>SIGNIFICANT USER-VISIBLE CHANGES</p>
<ul><li>Snaptron changed from stingray.cs.jhu.edu:8090 to snaptron.cs.jhu.edu so <code><a href="../reference/snaptron_query.html">snaptron_query()</a></code> has been changed accordingly.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.1.2" id="recount-112">recount 1.1.2<a class="anchor" aria-label="anchor" href="#recount-112"></a></h2>
<p>SIGNIFICANT USER-VISIBLE CHANGES</p>
<ul><li>The function <code><a href="../reference/reproduce_ranges.html">reproduce_ranges()</a></code> now has the <code>db</code> argument. By default it’s set to <code>TxDb.Hsapiens.UCSC.hg38.knownGene</code> to reproduce the actual information used in <code>recount</code>. But it can also be used with <code>EnsDb.Hsapiens.v79</code> to use the ENSEMBL annotation. Then with <code><a href="../reference/coverage_matrix.html">coverage_matrix()</a></code> you can get the counts for either an updated <code>TxDb.Hsapiens.UCSC.hg38.knownGene</code> or for <code>EnsDb.Hsapiens.v79</code> at the exon and/or gene levels as shown in the vignette.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.1.1" id="recount-111">recount 1.1.1<a class="anchor" aria-label="anchor" href="#recount-111"></a></h2>
<p>SIGNIFICANT USER-VISIBLE CHANGES</p>
<ul><li>The vignette now describes how to download all the data, how to check exon-exon junctions by class, and how to use <code>SciServer</code> compute to access all the <code>recount</code> data (over 6 TB) via <a href="http://www.sciserver.org/" class="external-link uri">http://www.sciserver.org/</a>
</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.99.30" id="recount-09930">recount 0.99.30<a class="anchor" aria-label="anchor" href="#recount-09930"></a></h2>
<p>NEW FEATURES</p>
<ul><li>Added the function <code><a href="../reference/snaptron_query.html">snaptron_query()</a></code> which queries Intropolis via Snaptron to find if an exon-exon junction is present in the data.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.99.29" id="recount-09929">recount 0.99.29<a class="anchor" aria-label="anchor" href="#recount-09929"></a></h2>
<p>BUF FIXES</p>
<ul><li>Fixed an bug in the vignette. Thanks to Michael Love for noticing it!</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.99.0" id="recount-0990">recount 0.99.0<a class="anchor" aria-label="anchor" href="#recount-0990"></a></h2>
<p>NEW FEATURES</p>
<ul><li>Created the package skeleton for <code>recount</code>
</li>
<li>Added the function <code><a href="../reference/reproduce_ranges.html">reproduce_ranges()</a></code> for re-creating the gene or exon level information used in the <code>recount</code> project.</li>
<li>Added the function <code><a href="../reference/abstract_search.html">abstract_search()</a></code> for identifying SRA projects of interest by searching the abstracts.</li>
<li>Added the function <code><a href="../reference/browse_study.html">browse_study()</a></code> for opening a browser tab for further exploring a project.</li>
<li>Added the function <code><a href="../reference/download_study.html">download_study()</a></code> for downloading the data from the <code>recount</code> project.</li>
<li>Added the function <code><a href="../reference/scale_counts.html">scale_counts()</a></code> for properly scaling the counts before performing a differential expression analysis with the <code>RangedSummarizedExperiment</code> objects hosted in the <code>recount</code> project.</li>
<li>Added the function <code><a href="../reference/expressed_regions.html">expressed_regions()</a></code> for defining the expressed regions in a chromosome for a given SRA study.</li>
<li>Added the function <code><a href="../reference/coverage_matrix.html">coverage_matrix()</a></code> for computing the coverage matrix based on the regions of interest for a given SRA study.</li>
<li>Added the function <code><a href="../reference/geo_info.html">geo_info()</a></code> for obtaining sample information from GEO.</li>
<li>Added the function <code><a href="../reference/find_geo.html">find_geo()</a></code> for finding the GEO accession id given a SRA run accession (<code>id</code>). This function will be useful for SRA projects that did not have GEO entries at the time <code>recount</code>’s data was created.</li>
<li>Added the function <code><a href="../reference/geo_characteristics.html">geo_characteristics()</a></code> for building a <code><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame()</a></code> from <code><a href="../reference/geo_info.html">geo_info()</a></code>’s results for the characteristics.</li>
<li>Added the function <code><a href="../reference/all_metadata.html">all_metadata()</a></code> which downloads all the phenotype data for all projects. This function can be useful for identifying projects and/or samples of interests.</li>
</ul></div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>

</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Leonardo Collado-Torres, Jeffrey T. Leek.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

