<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Given a set of regions for a chromosome, compute the coverage matrix for a
given SRA study. — coverage_matrix • recount</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Given a set of regions for a chromosome, compute the coverage matrix for a
given SRA study. — coverage_matrix"><meta property="og:description" content="Given a set of genomic regions as created by expressed_regions, this
function computes the coverage matrix for a library size of 40 million 100 bp
reads for a given SRA study."><meta property="og:image" content="/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">recount</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.27.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/fakeVignettes/SRP009615-results.html">UNKNOWN TITLE</a>
    </li>
    <li>
      <a href="../articles/recount-quickstart.html">recount quick start guide</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/leekgroup/recount/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Given a set of regions for a chromosome, compute the coverage matrix for a
given SRA study.</h1>
    <small class="dont-index">Source: <a href="https://github.com/leekgroup/recount/blob/HEAD/R/coverage_matrix.R" class="external-link"><code>R/coverage_matrix.R</code></a></small>
    <div class="hidden name"><code>coverage_matrix.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Given a set of genomic regions as created by <a href="expressed_regions.html">expressed_regions</a>, this
function computes the coverage matrix for a library size of 40 million 100 bp
reads for a given SRA study.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">coverage_matrix</span><span class="op">(</span></span>
<span>  <span class="va">project</span>,</span>
<span>  <span class="va">chr</span>,</span>
<span>  <span class="va">regions</span>,</span>
<span>  chunksize <span class="op">=</span> <span class="fl">1000</span>,</span>
<span>  bpparam <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  outdir <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  chrlen <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  verboseLoad <span class="op">=</span> <span class="va">verbose</span>,</span>
<span>  scale <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  round <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>project</dt>
<dd><p>A character vector with one SRA study id.</p></dd>


<dt>chr</dt>
<dd><p>A character vector with the name of the chromosome.</p></dd>


<dt>regions</dt>
<dd><p>A <a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges-class</a> object with regions
for <code>chr</code> for which to calculate the coverage matrix.</p></dd>


<dt>chunksize</dt>
<dd><p>A single integer vector defining the chunksize to use for
computing the coverage matrix. Regions will be split into different chunks
which can be useful when using a parallel instance as defined by
<code>bpparam</code>.</p></dd>


<dt>bpparam</dt>
<dd><p>A <a href="https://rdrr.io/pkg/BiocParallel/man/BiocParallelParam-class.html" class="external-link">BiocParallelParam-class</a> instance which
will be used to calculate the coverage matrix in parallel. By default,
<a href="https://rdrr.io/pkg/BiocParallel/man/SerialParam-class.html" class="external-link">SerialParam-class</a> will be used.</p></dd>


<dt>outdir</dt>
<dd><p>The destination directory for the downloaded file(s) that were
previously downloaded with <a href="download_study.html">download_study</a>. If the files are missing,
but <code>outdir</code> is specified, they will get downloaded first. By default
<code>outdir</code> is set to <code>NULL</code> which will use the data from the web.
We only recommend downloading the full data if you will use it several times.</p></dd>


<dt>chrlen</dt>
<dd><p>The chromosome length in base pairs. If it's <code>NULL</code>, the
chromosome length is extracted from the Rail-RNA runs GitHub repository.
Alternatively check the <code>SciServer</code> section on the vignette to see
how to access all the recount data via a R Jupyter Notebook.</p></dd>


<dt>verbose</dt>
<dd><p>If <code>TRUE</code> basic status updates will be printed along the
way.</p></dd>


<dt>verboseLoad</dt>
<dd><p>If <code>TRUE</code> basic status updates for loading the data
will be printed.</p></dd>


<dt>scale</dt>
<dd><p>If <code>TRUE</code>, the coverage counts will be scaled to read
counts based on a library size of 40 million reads. Set <code>scale</code> to
<code>FALSE</code> if you want the raw coverage counts. The scaling method is by
AUC, as in the default option of <a href="scale_counts.html">scale_counts</a>.</p></dd>


<dt>round</dt>
<dd><p>If <code>TRUE</code>, the counts are rounded to integers. Set to
<code>TRUE</code> if you want to match the defaults of <a href="scale_counts.html">scale_counts</a>.</p></dd>


<dt>...</dt>
<dd><p>Additional arguments passed to <a href="download_study.html">download_study</a> when
<code>outdir</code> is specified but the required files are missing.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>A <a href="https://rdrr.io/pkg/SummarizedExperiment/man/RangedSummarizedExperiment-class.html" class="external-link">RangedSummarizedExperiment-class</a></p>


<p>object with the counts stored in the assays slot.</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>When using <code>outdir = NULL</code> the information will be accessed
from the web on the fly. If you encounter internet access problems, it might
be best to first download the BigWig files using <a href="download_study.html">download_study</a>. This
might be the best option if you are accessing all chromosomes for a given
project and/or are thinking of using different sets of <code>regions</code> (for
example, from different cutoffs applied to <a href="expressed_regions.html">expressed_regions</a>).
Alternatively check the <code>SciServer</code> section on the vignette to see
how to access all the recount data via a R Jupyter Notebook.</p>
<p>If you have <code>bwtool</code> installed, you can use
<a href="https://github.com/LieberInstitute/recount.bwtool" class="external-link">https://github.com/LieberInstitute/recount.bwtool</a> for faster results.
Note that you will need to run <a href="scale_counts.html">scale_counts</a> after running
<code>coverage_matrix_bwtool()</code>.</p>
    </div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p><a href="download_study.html">download_study</a>, <a href="https://rdrr.io/pkg/derfinder/man/findRegions.html" class="external-link">findRegions</a>,
<a href="https://rdrr.io/pkg/derfinder/man/railMatrix.html" class="external-link">railMatrix</a></p></div>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Leonardo Collado-Torres</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span></span></span>
<span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="va">.Platform</span><span class="op">$</span><span class="va">OS.type</span> <span class="op">!=</span> <span class="st">"windows"</span><span class="op">)</span> <span class="op">{</span></span></span>
<span class="r-in"><span>    <span class="co">## Reading BigWig files is not supported by rtracklayer on Windows</span></span></span>
<span class="r-in"><span>    <span class="co">## Define expressed regions for study DRP002835, chrY</span></span></span>
<span class="r-in"><span>    <span class="va">regions</span> <span class="op">&lt;-</span> <span class="fu"><a href="expressed_regions.html">expressed_regions</a></span><span class="op">(</span><span class="st">"DRP002835"</span>, <span class="st">"chrY"</span>,</span></span>
<span class="r-in"><span>        cutoff <span class="op">=</span> <span class="fl">5L</span>,</span></span>
<span class="r-in"><span>        maxClusterGap <span class="op">=</span> <span class="fl">3000L</span></span></span>
<span class="r-in"><span>    <span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span>    <span class="co">## Now calculate the coverage matrix for this study</span></span></span>
<span class="r-in"><span>    <span class="va">rse</span> <span class="op">&lt;-</span> <span class="fu">coverage_matrix</span><span class="op">(</span><span class="st">"DRP002835"</span>, <span class="st">"chrY"</span>, <span class="va">regions</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span>    <span class="co">## One row per region</span></span></span>
<span class="r-in"><span>    <span class="fu"><a href="https://rdrr.io/r/base/identical.html" class="external-link">identical</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">regions</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">rse</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2023-05-07 05:53:18.57369 loadCoverage: loading BigWig file http://duffel.rail.bio/recount/DRP002835/bw/mean_DRP002835.bw</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2023-05-07 05:53:22.100386 loadCoverage: applying the cutoff to the merged data</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2023-05-07 05:53:22.16054 filterData: originally there were 57227415 rows, now there are 57227415 rows. Meaning that 0 percent was filtered.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2023-05-07 05:53:22.16651 findRegions: identifying potential segments</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2023-05-07 05:53:22.176142 findRegions: segmenting information</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2023-05-07 05:53:22.176906 .getSegmentsRle: segmenting with cutoff(s) 5</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2023-05-07 05:53:22.220847 findRegions: identifying candidate regions</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2023-05-07 05:53:23.612509 findRegions: identifying region clusters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2023-05-07 05:53:25.312496 railMatrix: processing regions 1 to 372</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2023-05-07 05:53:25.331388 railMatrix: processing file http://duffel.rail.bio/recount/DRP002835/bw/DRR028129.bw</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2023-05-07 05:53:29.002249 railMatrix: processing file http://duffel.rail.bio/recount/DRP002835/bw/DRR028130.bw</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] TRUE</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Leonardo Collado-Torres, Jeffrey T. Leek.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

