<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Download data for a given SRA study id from the recount project — download_study • recount</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Download data for a given SRA study id from the recount project — download_study" />
<meta property="og:description" content="Download the gene or exon level
RangedSummarizedExperiment-class objects
provided by the recount project. Alternatively download the counts, metadata
or file information for a given SRA study id. You can also download the
sample bigWig files or the mean coverage bigWig file." />
<meta property="og:image" content="/logo.png" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">recount</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.21.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/fakeVignettes/SRP009615-results.html">UNKNOWN TITLE</a>
    </li>
    <li>
      <a href="../articles/recount-quickstart.html">recount quick start guide</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/leekgroup/recount/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Download data for a given SRA study id from the recount project</h1>
    <small class="dont-index">Source: <a href='https://github.com/leekgroup/recount/blob/master/R/download_study.R'><code>R/download_study.R</code></a></small>
    <div class="hidden name"><code>download_study.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Download the gene or exon level
<a href='https://rdrr.io/pkg/SummarizedExperiment/man/RangedSummarizedExperiment-class.html'>RangedSummarizedExperiment-class</a> objects
provided by the recount project. Alternatively download the counts, metadata
or file information for a given SRA study id. You can also download the
sample bigWig files or the mean coverage bigWig file.</p>
    </div>

    <pre class="usage"><span class='fu'>download_study</span><span class='op'>(</span>
  <span class='va'>project</span>,
  type <span class='op'>=</span> <span class='st'>"rse-gene"</span>,
  outdir <span class='op'>=</span> <span class='va'>project</span>,
  download <span class='op'>=</span> <span class='cn'>TRUE</span>,
  version <span class='op'>=</span> <span class='fl'>2</span>,
  <span class='va'>...</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>project</th>
      <td><p>A character vector with one SRA study id.</p></td>
    </tr>
    <tr>
      <th>type</th>
      <td><p>Specifies which files to download. The options are:</p><dl>
<dt>rse-gene</dt><dd><p>the gene-level
<a href='https://rdrr.io/pkg/SummarizedExperiment/man/RangedSummarizedExperiment-class.html'>RangedSummarizedExperiment-class</a> object in
a file named rse_gene.Rdata.</p></dd>
<dt>rse-exon</dt><dd><p>the exon-level
<a href='https://rdrr.io/pkg/SummarizedExperiment/man/RangedSummarizedExperiment-class.html'>RangedSummarizedExperiment-class</a> object in
a file named rse_exon.Rdata.</p></dd>
<dt>rse-jx</dt><dd><p>the exon-exon junction level
<a href='https://rdrr.io/pkg/SummarizedExperiment/man/RangedSummarizedExperiment-class.html'>RangedSummarizedExperiment-class</a> object in
a file named rse_jx.Rdata.</p></dd>
<dt>rse-tx</dt><dd><p>the transcript level
<a href='https://rdrr.io/pkg/SummarizedExperiment/man/RangedSummarizedExperiment-class.html'>RangedSummarizedExperiment-class</a> object in
a file named rse_tx.RData.</p></dd>
<dt>counts-gene</dt><dd><p>the gene-level counts in a tsv file named
counts_gene.tsv.gz.</p></dd>
<dt>counts-exon</dt><dd><p>the exon-level counts in a tsv file named
counts_exon.tsv.gz.</p></dd>
<dt>counts-jx</dt><dd><p>the exon-exon junction level counts in a tsv file named
counts_jx.tsv.gz.</p></dd>
<dt>phenotype</dt><dd><p>the phenotype data for the study in a tsv file named
<code>project</code>.tsv.</p></dd>
<dt>files-info</dt><dd><p>the files information for the given study (including
md5sum hashes) in a tsv file named files_info.tsv.</p></dd>
<dt>samples</dt><dd><p>one bigWig file per sample in the study.</p></dd>
<dt>mean</dt><dd><p>one mean bigWig file for the samples in the study,
with each sample normalized to a 40 million 100 bp library using the
total coverage sum (area under the coverage curve, AUC) for the given
sample.</p></dd>
<dt>all</dt><dd><p>Downloads all the above types. Note that it might take some
time if the project has many samples. When using <code>type = 'all'</code> a
small delay will be added before each download request to avoid
request issues.</p></dd>
<dt>rse-fc</dt><dd><p>Downloads the FANTOM-CAT/recount2 rse file described in
Imada, Sanchez, et al., bioRxiv, 2019.</p></dd>

</dl></td>
    </tr>
    <tr>
      <th>outdir</th>
      <td><p>The destination directory for the downloaded file(s).
Alternatively check the <code>SciServer</code> section on the vignette to see
how to access all the recount data via a R Jupyter Notebook.</p></td>
    </tr>
    <tr>
      <th>download</th>
      <td><p>Whether to download the files or just get the download urls.</p></td>
    </tr>
    <tr>
      <th>version</th>
      <td><p>A single integer specifying which version of the files to
download. Valid options are 1 and 2, as described in
<a href='https://jhubiostatistics.shinyapps.io/recount/'>https://jhubiostatistics.shinyapps.io/recount/</a> under the
documentation tab. Briefly, version 1 are counts based on reduced exons while
version 2 are based on disjoint exons. This argument mostly just matters for
the exon counts. Defaults to version 2 (disjoint exons).
Use <code>version = 1</code> for backward compatability with exon counts
prior to version 1.5.3 of the package.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>Additional arguments passed to <a href='https://rdrr.io/pkg/downloader/man/download.html'>download</a>.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>Returns invisibly the URL(s) for the files that were downloaded.</p>
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>Check <a href='http://stackoverflow.com/a/34383991'>http://stackoverflow.com/a/34383991</a> if you need to find
the effective URLs. For example,
<a href='http://duffel.rail.bio/recount/DRP000366/bw/mean_DRP000366.bw'>http://duffel.rail.bio/recount/DRP000366/bw/mean_DRP000366.bw</a> points to
a link from SciServer.</p>
<p>Transcript quantifications are described in Fu et al, bioRxiv, 2018.
<a href='https://www.biorxiv.org/content/10.1101/247346v2'>https://www.biorxiv.org/content/10.1101/247346v2</a></p>
<p>FANTOM-CAT/recount2 quantifications are described in Imada,
Sanchez, et al., bioRxiv, 2019.
<a href='https://www.biorxiv.org/content/10.1101/659490v1'>https://www.biorxiv.org/content/10.1101/659490v1</a></p>
    <h2 class="hasAnchor" id="author"><a class="anchor" href="#author"></a>Author</h2>

    <p>Leonardo Collado-Torres</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><span class='r-in'><span class='co'>## Find the URL to download the RangedSummarizedExperiment for the</span></span>
<span class='r-in'><span class='co'>## Geuvadis consortium study.</span></span>
<span class='r-in'><span class='va'>url</span> <span class='op'>&lt;-</span> <span class='fu'>download_study</span><span class='op'>(</span><span class='st'>"ERP001942"</span>, download <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span></span>
<span class='r-in'></span>
<span class='r-in'><span class='co'>## See the actual URL</span></span>
<span class='r-in'><span class='va'>url</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [1] "http://duffel.rail.bio/recount/v2/ERP001942/rse_gene.Rdata"</span>
<span class='r-in'><span class='kw'>if</span> <span class='op'>(</span><span class='cn'>FALSE</span><span class='op'>)</span> <span class='op'>{</span></span>
<span class='r-in'><span class='co'>## Download the example data included in the package for study SRP009615</span></span>
<span class='r-in'></span>
<span class='r-in'><span class='va'>url2</span> <span class='op'>&lt;-</span> <span class='fu'>download_study</span><span class='op'>(</span><span class='st'>"SRP009615"</span><span class='op'>)</span></span>
<span class='r-in'><span class='va'>url2</span></span>
<span class='r-in'></span>
<span class='r-in'><span class='co'>## Load the data</span></span>
<span class='r-in'><span class='fu'><a href='https://rdrr.io/r/base/load.html'>load</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/file.path.html'>file.path</a></span><span class='op'>(</span><span class='st'>"SRP009615"</span>, <span class='st'>"rse_gene.Rdata"</span><span class='op'>)</span><span class='op'>)</span></span>
<span class='r-in'></span>
<span class='r-in'><span class='co'>## Compare the data</span></span>
<span class='r-in'><span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='st'><a href='https://testthat.r-lib.org'>"testthat"</a></span><span class='op'>)</span></span>
<span class='r-in'><span class='fu'><a href='https://testthat.r-lib.org/reference/expect_equivalent.html'>expect_equivalent</a></span><span class='op'>(</span><span class='va'>rse_gene</span>, <span class='va'>rse_gene_SRP009615</span><span class='op'>)</span></span>
<span class='r-in'><span class='op'>}</span></span>
<span class='r-in'></span>
</pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Leonardo Collado-Torres, Jeffrey T. Leek.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


