<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Scale the raw counts provided by the recount project — scale_counts • recount</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Scale the raw counts provided by the recount project — scale_counts" />
<meta property="og:description" content="In preparation for a differential expression analysis, you will have to
choose how to scale the raw counts provided by the recount project. Note that
the raw counts are the sum of the base level coverage so you have to take
into account the read length or simply the total coverage for the given
sample (default option). You might want to do some further scaling to take
into account the gene or exon lengths. If you prefer to calculate read counts
without scaling check the function read_counts." />
<meta property="og:image" content="/logo.png" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">recount</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.13.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/fakeVignettes/SRP009615-results.html">UNKNOWN TITLE</a>
    </li>
    <li>
      <a href="../articles/recount-quickstart.html">recount quick start guide</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/leekgroup/recount/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Scale the raw counts provided by the recount project</h1>
    <small class="dont-index">Source: <a href='https://github.com/leekgroup/recount/blob/master/R/scale_counts.R'><code>R/scale_counts.R</code></a></small>
    <div class="hidden name"><code>scale_counts.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>In preparation for a differential expression analysis, you will have to
choose how to scale the raw counts provided by the recount project. Note that
the raw counts are the sum of the base level coverage so you have to take
into account the read length or simply the total coverage for the given
sample (default option). You might want to do some further scaling to take
into account the gene or exon lengths. If you prefer to calculate read counts
without scaling check the function <a href='read_counts.html'>read_counts</a>.</p>
    </div>

    <pre class="usage"><span class='fu'>scale_counts</span>(
  <span class='no'>rse</span>,
  <span class='kw'>by</span> <span class='kw'>=</span> <span class='st'>"auc"</span>,
  <span class='kw'>targetSize</span> <span class='kw'>=</span> <span class='fl'>4e+07</span>,
  <span class='kw'>L</span> <span class='kw'>=</span> <span class='fl'>100</span>,
  <span class='kw'>factor_only</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>round</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>
)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>rse</th>
      <td><p>A <a href='https://rdrr.io/pkg/SummarizedExperiment/man/RangedSummarizedExperiment-class.html'>RangedSummarizedExperiment-class</a>
object as downloaded with <a href='download_study.html'>download_study</a>.</p></td>
    </tr>
    <tr>
      <th>by</th>
      <td><p>Either <code>auc</code> or <code>mapped_reads</code>. If set to <code>auc</code> it
will scale the counts by the total coverage of the sample. That is, the area
under the curve (AUC) of the coverage. If set to <code>mapped_reads</code> it will
scale the counts by the number of mapped reads, whether the library was
paired-end or not, and the desired read length (<code>L</code>).</p></td>
    </tr>
    <tr>
      <th>targetSize</th>
      <td><p>The target library size in number of single end reads.</p></td>
    </tr>
    <tr>
      <th>L</th>
      <td><p>The target read length. Only used when <code>by = 'mapped_reads'</code>
since it cancels out in the calculation when using <code>by = 'auc'</code>.</p></td>
    </tr>
    <tr>
      <th>factor_only</th>
      <td><p>Whether to only return the numeric scaling factor or
to return a <a href='https://rdrr.io/pkg/SummarizedExperiment/man/RangedSummarizedExperiment-class.html'>RangedSummarizedExperiment-class</a>
object with the counts scaled. If set to <code>TRUE</code>, you have to multiply
the sample counts by this scaling factor.</p></td>
    </tr>
    <tr>
      <th>round</th>
      <td><p>Whether to round the counts to integers or not.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>If <code>factor_only = TRUE</code> it returns a numeric vector with the
scaling factor for each sample. If <code>factor_only = FALSE</code> it returns a
<a href='https://rdrr.io/pkg/SummarizedExperiment/man/RangedSummarizedExperiment-class.html'>RangedSummarizedExperiment-class</a> object with
the counts already scaled.</p>
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>Rail-RNA <a href='http://rail.bio'>http://rail.bio</a> uses soft clipping when aligning
which is why we recommed using <code>by = 'auc'</code>.</p>
<p>If the reads are from a paired-end library, then the <code>avg_read_length</code>
is the average fragment length. This is taken into account when using
<code>by = 'mapped_reads'</code>.</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><a href='download_study.html'>download_study</a>, <a href='read_counts.html'>read_counts</a></p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'>## Load an example rse_gene object</span>
<span class='no'>rse_gene</span> <span class='kw'>&lt;-</span> <span class='no'>rse_gene_SRP009615</span>

<span class='co'>## Scale counts</span>
<span class='no'>rse</span> <span class='kw'>&lt;-</span> <span class='fu'>scale_counts</span>(<span class='no'>rse_gene</span>)

<span class='co'>## Find the project used as an example</span>
<span class='no'>project_info</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='abstract_search.html'>abstract_search</a></span>(<span class='st'>"GSE32465"</span>)

<span class='co'>## See some summary information for this project</span>
<span class='no'>project_info</span></div><div class='output co'>#&gt;     number_samples species
#&gt; 340             12   human
#&gt;                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               abstract
#&gt; 340 Summary: K562-shX cells are made in an effort to validate TFBS data and ChIP-seq antibodies in Myers lab (GSE32465). K562 cells are transduced with lentiviral vector having Tet-inducible shRNA targeting a transcription factor gene. Cells with stable integration of shRNA constructs are selected using puromycin in growth media. Doxycycline is added to the growth media to induce the expression of shRNA and a red fluorescent protein marker.  A successful shRNA cell line shows at least a 70% reduction in expression of the target transcription factor as measured by qPCR. For identification, we designated these cell lines as K562-shX, where X is the transcription factor targeted by shRNA and K562 denotes the parent cell line.  For example, K562-shATF3 cells are K562 derived cells selected for stable integration of shRNA targeting the transcription factor ATF3 gene and showed at least a 70% reduction in the expression of ATF3 gene when measured by qPCR. Cells growing without doxycycline (uninduced) are used as a control to measure the change in expression of target transcription factor gene after induction of shRNA using doxycycline. For detailed growth and culturing protocols for these cells please refer to http://hudsonalpha.org/myers-lab/protocols . To identify the potential downstream targets of the candidate transcription factor, analyze the mRNA expression profile of the uninduced and induced K562-shX using RNA-seq.    For data usage terms and conditions, please refer to http://www.genome.gov/27528022 and http://www.genome.gov/Pages/Research/ENCODE/ENCODEDataReleasePolicyFinal2008.pdf    Overall Design: Make K562-shX cells as described in the http://hudsonalpha.org/myers-lab/protocols . Measure the mRNA expression levels in uninduced K562-shX and induced K562-shX cells in two biological replicates using RNA-seq. Identify the potential downstream targets of the candidate transcription factor.
#&gt;       project
#&gt; 340 SRP009615</div><div class='input'>
<span class='co'>## Use the following code to re-download this file</span>
<span class='kw'>if</span> (<span class='fl'>FALSE</span>) {
<span class='co'>## Download</span>
<span class='fu'><a href='download_study.html'>download_study</a></span>(<span class='no'>project_info</span>$<span class='no'>project</span>)

<span class='co'>## Load file</span>
<span class='fu'><a href='https://rdrr.io/r/base/load.html'>load</a></span>(<span class='fu'><a href='https://rdrr.io/r/base/file.path.html'>file.path</a></span>(<span class='no'>project_info</span>$<span class='no'>project</span>, <span class='st'>"rse_gene.Rdata"</span>))
<span class='fu'><a href='https://rdrr.io/r/base/identical.html'>identical</a></span>(<span class='no'>rse_gene</span>, <span class='no'>rse_gene_SRP009615</span>)
}</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Leonardo Collado-Torres, Jeffrey T. Leek.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


