<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>recount quick start guide • recount</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="recount quick start guide">
<meta property="og:description" content="recount">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">recount</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.27.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/fakeVignettes/SRP009615-results.html">UNKNOWN TITLE</a>
    </li>
    <li>
      <a href="../articles/recount-quickstart.html">recount quick start guide</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/leekgroup/recount/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>recount quick start guide</h1>
                        <h4 data-toc-skip class="author">Leonardo
Collado-Torres</h4>
            <address class="author_afil">
      Lieber Institute for Brain Development, Johns Hopkins Medical
CampusCenter for Computational Biology, Johns Hopkins
University<br><a class="author_email" href="mailto:#"></a><a href="mailto:lcolladotor@gmail.com" class="email">lcolladotor@gmail.com</a>
      </address>
                  
            <h4 data-toc-skip class="date">7 May 2023</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/leekgroup/recount/blob/HEAD/vignettes/recount-quickstart.Rmd" class="external-link"><code>vignettes/recount-quickstart.Rmd</code></a></small>
      <div class="hidden name"><code>recount-quickstart.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="basics">Basics<a class="anchor" aria-label="anchor" href="#basics"></a>
</h2>
<div class="section level3">
<h3 id="install-recount">Install <em><a href="https://bioconductor.org/packages/3.16/recount" class="external-link">recount</a></em><a class="anchor" aria-label="anchor" href="#install-recount"></a>
</h3>
<p><code>R</code> is an open-source statistical environment which can be
easily modified to enhance its functionality via packages. <em><a href="https://bioconductor.org/packages/3.16/recount" class="external-link">recount</a></em>
is a <code>R</code> package available via the <a href="http://bioconductor/packages/recount" class="external-link">Bioconductor</a> repository
for packages. <code>R</code> can be installed on any operating system
from <a href="https://cran.r-project.org/" class="external-link">CRAN</a> after which you can
install <em><a href="https://bioconductor.org/packages/3.16/recount" class="external-link">recount</a></em>
by using the following commands in your <code>R</code> session:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"recount"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Check that you have a valid Bioconductor installation</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/valid.html" class="external-link">valid</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="required-knowledge">Required knowledge<a class="anchor" aria-label="anchor" href="#required-knowledge"></a>
</h3>
<p><em><a href="https://bioconductor.org/packages/3.16/recount" class="external-link">recount</a></em>
is based on many other packages and in particular in those that have
implemented the infrastructure needed for dealing with RNA-seq data.
That is, packages like <em><a href="https://bioconductor.org/packages/3.16/GenomicFeatures" class="external-link">GenomicFeatures</a></em>
and <em><a href="https://bioconductor.org/packages/3.16/rtracklayer" class="external-link">rtracklayer</a></em>
that allow you to import the data. A <em><a href="https://bioconductor.org/packages/3.16/recount" class="external-link">recount</a></em>
user is not expected to deal with those packages directly but will need
to be familiar with <em><a href="https://bioconductor.org/packages/3.16/SummarizedExperiment" class="external-link">SummarizedExperiment</a></em>
to understand the results <em><a href="https://bioconductor.org/packages/3.16/recount" class="external-link">recount</a></em>
generates. It might also prove to be highly beneficial to check the</p>
<ul>
<li>
<em><a href="https://bioconductor.org/packages/3.16/DESeq2" class="external-link">DESeq2</a></em>
package for performing differential expression analysis with the
<em>RangedSummarizedExperiment</em> objects,</li>
<li>
<em><a href="https://bioconductor.org/packages/3.16/DEFormats" class="external-link">DEFormats</a></em>
package for switching the objects to those used by other differential
expression packages such as <em><a href="https://bioconductor.org/packages/3.16/edgeR" class="external-link">edgeR</a></em>,</li>
<li>
<em><a href="https://bioconductor.org/packages/3.16/derfinder" class="external-link">derfinder</a></em>
package for performing annotation-agnostic differential expression
analysis.</li>
</ul>
<p>If you are asking yourself the question “Where do I start using
Bioconductor?” you might be interested in <a href="http://lcolladotor.github.io/2014/10/16/startbioc/#.VkOKbq6rRuU" class="external-link">this
blog post</a>.</p>
</div>
<div class="section level3">
<h3 id="asking-for-help">Asking for help<a class="anchor" aria-label="anchor" href="#asking-for-help"></a>
</h3>
<p>As package developers, we try to explain clearly how to use our
packages and in which order to use the functions. But <code>R</code> and
<code>Bioconductor</code> have a steep learning curve so it is critical
to learn where to ask for help. The blog post quoted above mentions some
but we would like to highlight the <a href="https://support.bioconductor.org/" class="external-link">Bioconductor support site</a>
as the main resource for getting help: remember to use the
<code>recount</code> tag and check <a href="https://support.bioconductor.org/t/recount/" class="external-link">the older posts</a>.
Other alternatives are available such as creating GitHub issues and
tweeting. However, please note that if you want to receive help you
should adhere to the <a href="http://www.bioconductor.org/help/support/posting-guide/" class="external-link">posting
guidelines</a>. It is particularly critical that you provide a small
reproducible example and your session information so package developers
can track down the source of the error.</p>
</div>
<div class="section level3">
<h3 id="workflow-using-recount">Workflow using <em><a href="https://bioconductor.org/packages/3.16/recount" class="external-link">recount</a></em><a class="anchor" aria-label="anchor" href="#workflow-using-recount"></a>
</h3>
<p>We have written a workflow on how to use <em><a href="https://bioconductor.org/packages/3.16/recount" class="external-link">recount</a></em>
that explains more details on how to use this package with other
Bioconductor packages as well as the details on the actual counts
provided by <em><a href="https://bioconductor.org/packages/3.16/recount" class="external-link">recount</a></em>.
Check it at <a href="https://f1000research.com/articles/6-1558/v1" class="external-link">f1000research.com/articles/6-1558/v1</a>
or <a href="http://bioconductor.org/help/workflows/recountWorkflow/" class="external-link">bioconductor.org/help/workflows/recountWorkflow/</a>
(Collado-Torres, Nellore, and Jaffe, 2017).</p>
</div>
<div class="section level3">
<h3 id="citing-recount">Citing <em><a href="https://bioconductor.org/packages/3.16/recount" class="external-link">recount</a></em><a class="anchor" aria-label="anchor" href="#citing-recount"></a>
</h3>
<p>We hope that <em><a href="https://bioconductor.org/packages/3.16/recount" class="external-link">recount</a></em>
will be useful for your research. Please use the following information
to cite the package and the overall approach. Thank you!</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Citation info</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/citation.html" class="external-link">citation</a></span><span class="op">(</span><span class="st">"recount"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## To cite package 'recount' in publications use:</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##   Collado-Torres L, Nellore A, Kammers K, Ellis SE, Taub MA, Hansen KD,</span></span>
<span><span class="co">##   Jaffe AE, Langmead B, Leek JT (2017). "Reproducible RNA-seq analysis</span></span>
<span><span class="co">##   using recount2." _Nature Biotechnology_. doi:10.1038/nbt.3838</span></span>
<span><span class="co">##   &lt;https://doi.org/10.1038/nbt.3838&gt;,</span></span>
<span><span class="co">##   &lt;http://www.nature.com/nbt/journal/v35/n4/full/nbt.3838.html&gt;.</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##   Collado-Torres L, Nellore A, Jaffe AE (2017). "recount workflow:</span></span>
<span><span class="co">##   Accessing over 70,000 human RNA-seq samples with Bioconductor</span></span>
<span><span class="co">##   [version 1; referees: 1 approved, 2 approved with reservations]."</span></span>
<span><span class="co">##   _F1000Research_. doi:10.12688/f1000research.12223.1</span></span>
<span><span class="co">##   &lt;https://doi.org/10.12688/f1000research.12223.1&gt;,</span></span>
<span><span class="co">##   &lt;https://f1000research.com/articles/6-1558/v1&gt;.</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##   Ellis SE, Collado-Torres L, Jaffe AE, Leek JT (2018). "Improving the</span></span>
<span><span class="co">##   value of public RNA-seq expression data by phenotype prediction."</span></span>
<span><span class="co">##   _Nucl. Acids Res._. doi:10.1093/nar/gky102</span></span>
<span><span class="co">##   &lt;https://doi.org/10.1093/nar/gky102&gt;,</span></span>
<span><span class="co">##   &lt;https://doi.org/10.1093/nar/gky102&gt;.</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##   Collado-Torres L, Nellore A, Kammers K, Ellis SE, Taub MA, Hansen KD,</span></span>
<span><span class="co">##   Jaffe AE, Langmead B, Leek JT (2023). _Explore and download data from</span></span>
<span><span class="co">##   the recount project_. doi:10.18129/B9.bioc.recount</span></span>
<span><span class="co">##   &lt;https://doi.org/10.18129/B9.bioc.recount&gt;,</span></span>
<span><span class="co">##   https://github.com/leekgroup/recount - R package version 1.27.0,</span></span>
<span><span class="co">##   &lt;http://www.bioconductor.org/packages/recount&gt;.</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##   Frazee AC, Langmead B, Leek JT (2011). "ReCount: A multi-experiment</span></span>
<span><span class="co">##   resource of analysis-ready RNA-seq gene count datasets." _BMC</span></span>
<span><span class="co">##   Bioinformatics_. doi:10.1186/1471-2105-12-449</span></span>
<span><span class="co">##   &lt;https://doi.org/10.1186/1471-2105-12-449&gt;,</span></span>
<span><span class="co">##   &lt;https://doi.org/10.1186/1471-2105-12-449&gt;.</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##   Razmara A, Ellis SE, Sokolowski DJ, Davis S, Wilson MD, Leek JT,</span></span>
<span><span class="co">##   Jaffe AE, Collado-Torres L (2019). "recount-brain: a curated</span></span>
<span><span class="co">##   repository of human brain RNA-seq datasets metadata." _bioRxiv_.</span></span>
<span><span class="co">##   doi:10.1101/618025 &lt;https://doi.org/10.1101/618025&gt;,</span></span>
<span><span class="co">##   &lt;https://doi.org/10.1101/618025&gt;.</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##   Imada E, Sanchez DF, Collado-Torres L, Wilks C, Matam T, Dinalankara</span></span>
<span><span class="co">##   W, Stupnikov A, Lobo-Pereira F, Yip C, Yasuzawa K, Kondo N, Itoh M,</span></span>
<span><span class="co">##   Suzuki H, Kasukawa T, Hon CC, de Hoon MJ, Shin JW, Carninci P, Jaffe</span></span>
<span><span class="co">##   AE, Leek JT, Favorov A, Franco GR, Langmead B, Marchionni L (2020).</span></span>
<span><span class="co">##   "Recounting the FANTOM CAGE–Associated Transcriptome." _Genome</span></span>
<span><span class="co">##   Research_. doi:10.1101/gr.254656.119</span></span>
<span><span class="co">##   &lt;https://doi.org/10.1101/gr.254656.119&gt;,</span></span>
<span><span class="co">##   &lt;https://doi.org/10.1101/gr.254656.119&gt;.</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## To see these entries in BibTeX format, use 'print(&lt;citation&gt;,</span></span>
<span><span class="co">## bibtex=TRUE)', 'toBibtex(.)', or set</span></span>
<span><span class="co">## 'options(citation.bibtex.max=999)'.</span></span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="quick-start-to-using-to-recount">Quick start to using to <em><a href="https://bioconductor.org/packages/3.16/recount" class="external-link">recount</a></em><a class="anchor" aria-label="anchor" href="#quick-start-to-using-to-recount"></a>
</h2>
<p>Main updates:</p>
<ul>
<li><strong>As of January 30, 2017 the annotation used for the exon and
gene counts is Gencode v25.</strong></li>
<li><strong>As of January 12, 2018 transcripts counts are available via
recount2 thanks to the work of <a href="https://www.biorxiv.org/content/10.1101/247346v2" class="external-link">Fu et al,
bioRxiv, 2018</a>. Disjoint exon counts (version 2) were also released
as described in detail in the <a href="https://jhubiostatistics.shinyapps.io/recount/" class="external-link">recount
website</a> documentation tab.</strong></li>
<li><strong>As of April 29, 2019 FANTOM-CAT/recount2 annotation
quantifications area available via recount2 thanks to the work by <a href="https://www.biorxiv.org/content/10.1101/659490v1" class="external-link">Imada, Sanchez,
et al., bioRxiv, 2019</a>.</strong></li>
</ul>
<p><a href="https://jhubiostatistics.shinyapps.io/recount/" class="external-link"><img src="https://raw.githubusercontent.com/leekgroup/recount-website/master/website/recount.png" align="center" width="600"></a></p>
<p>Here is a very quick example of how to download a
<code>RangedSummarizedExperiment</code> object with the gene counts for
a 2 groups project (12 samples) with SRA study id <a href="http://trace.ncbi.nlm.nih.gov/Traces/sra/?study=SRP009615" class="external-link">SRP009615</a>
using the <em><a href="https://bioconductor.org/packages/3.16/recount" class="external-link">recount</a></em>
package (Collado-Torres, Nellore, Kammers, Ellis, Taub, Hansen, Jaffe,
Langmead, and Leek, 2017). The <code>RangedSummarizedExperiment</code>
object is defined in the <em><a href="https://bioconductor.org/packages/3.16/SummarizedExperiment" class="external-link">SummarizedExperiment</a></em>
(Morgan, Obenchain, Hester, and Pagès, 2017) package and can be used for
differential expression analysis with different packages. Here we show
how to use <em><a href="https://bioconductor.org/packages/3.16/DESeq2" class="external-link">DESeq2</a></em>
(Love, Huber, and Anders, 2014) to perform the differential expresion
analysis.</p>
<p>This quick analysis is explained in more detail later on in this
document. Further information about the recount project can be found in
the <a href="https://www.nature.com/articles/nbt.3838" class="external-link">main
publication</a>. Check the <a href="https://jhubiostatistics.shinyapps.io/recount/" class="external-link">recount
website</a> for related publications.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Load library</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/leekgroup/recount" class="external-link">"recount"</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Find a project of interest</span></span>
<span><span class="va">project_info</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/abstract_search.html">abstract_search</a></span><span class="op">(</span><span class="st">"GSE32465"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Download the gene-level RangedSummarizedExperiment data</span></span>
<span><span class="fu"><a href="../reference/download_study.html">download_study</a></span><span class="op">(</span><span class="va">project_info</span><span class="op">$</span><span class="va">project</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Load the data</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">project_info</span><span class="op">$</span><span class="va">project</span>, <span class="st">"rse_gene.Rdata"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Browse the project at SRA</span></span>
<span><span class="fu"><a href="../reference/browse_study.html">browse_study</a></span><span class="op">(</span><span class="va">project_info</span><span class="op">$</span><span class="va">project</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## View GEO ids</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_gene</span><span class="op">)</span><span class="op">$</span><span class="va">geo_accession</span></span>
<span></span>
<span><span class="co">## Extract the sample characteristics</span></span>
<span><span class="va">geochar</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/splitAsList.html" class="external-link">split</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_gene</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_gene</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>, <span class="va">geo_characteristics</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Note that the information for this study is a little inconsistent, so we</span></span>
<span><span class="co">## have to fix it.</span></span>
<span><span class="va">geochar</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/do.call.html" class="external-link">do.call</a></span><span class="op">(</span><span class="va">rbind</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">geochar</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="kw">if</span> <span class="op">(</span><span class="st">"cells"</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>        <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">==</span> <span class="st">"cells"</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"cell.line"</span></span>
<span>        <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span>    <span class="op">}</span> <span class="kw">else</span> <span class="op">{</span></span>
<span>        <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span>    <span class="op">}</span></span>
<span><span class="op">}</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## We can now define some sample information to use</span></span>
<span><span class="va">sample_info</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span>
<span>    run <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_gene</span><span class="op">)</span><span class="op">$</span><span class="va">run</span>,</span>
<span>    group <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"uninduced"</span>, <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_gene</span><span class="op">)</span><span class="op">$</span><span class="va">title</span><span class="op">)</span>, <span class="st">"uninduced"</span>, <span class="st">"induced"</span><span class="op">)</span>,</span>
<span>    gene_target <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_gene</span><span class="op">)</span><span class="op">$</span><span class="va">title</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span></span>
<span>        <span class="fu"><a href="https://rdrr.io/r/base/strsplit.html" class="external-link">strsplit</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/strsplit.html" class="external-link">strsplit</a></span><span class="op">(</span></span>
<span>            <span class="va">x</span>,</span>
<span>            <span class="st">"targeting "</span></span>
<span>        <span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span>, <span class="st">" gene"</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span></span>
<span>    <span class="op">}</span><span class="op">)</span>,</span>
<span>    cell.line <span class="op">=</span> <span class="va">geochar</span><span class="op">$</span><span class="va">cell.line</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Scale counts by taking into account the total coverage per sample</span></span>
<span><span class="va">rse</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/scale_counts.html">scale_counts</a></span><span class="op">(</span><span class="va">rse_gene</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Add sample information for DE analysis</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse</span><span class="op">)</span><span class="op">$</span><span class="va">group</span> <span class="op">&lt;-</span> <span class="va">sample_info</span><span class="op">$</span><span class="va">group</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse</span><span class="op">)</span><span class="op">$</span><span class="va">gene_target</span> <span class="op">&lt;-</span> <span class="va">sample_info</span><span class="op">$</span><span class="va">gene_target</span></span>
<span></span>
<span><span class="co">## Perform differential expression analysis with DESeq2</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/mikelove/DESeq2" class="external-link">"DESeq2"</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Specify design and switch to DESeq2 format</span></span>
<span><span class="va">dds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeqDataSet.html" class="external-link">DESeqDataSet</a></span><span class="op">(</span><span class="va">rse</span>, <span class="op">~</span> <span class="va">gene_target</span> <span class="op">+</span> <span class="va">group</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Perform DE analysis</span></span>
<span><span class="va">dds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeq.html" class="external-link">DESeq</a></span><span class="op">(</span><span class="va">dds</span>, test <span class="op">=</span> <span class="st">"LRT"</span>, reduced <span class="op">=</span> <span class="op">~</span><span class="va">gene_target</span>, fitType <span class="op">=</span> <span class="st">"local"</span><span class="op">)</span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/results.html" class="external-link">results</a></span><span class="op">(</span><span class="va">dds</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Explore results</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/plotMA.html" class="external-link">plotMA</a></span><span class="op">(</span><span class="va">res</span>, main <span class="op">=</span> <span class="st">"DESeq2 results for SRP009615"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Make a report with the results</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/leekgroup/regionReport" class="external-link">"regionReport"</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/regionReport/man/DESeq2Report.html" class="external-link">DESeq2Report</a></span><span class="op">(</span><span class="va">dds</span>,</span>
<span>    res <span class="op">=</span> <span class="va">res</span>, project <span class="op">=</span> <span class="st">"SRP009615"</span>,</span>
<span>    intgroup <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"group"</span>, <span class="st">"gene_target"</span><span class="op">)</span>, outdir <span class="op">=</span> <span class="st">"."</span>,</span>
<span>    output <span class="op">=</span> <span class="st">"SRP009615-results"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>The <a href="https://jhubiostatistics.shinyapps.io/recount/" class="external-link">recount
project</a> also hosts the necessary data to perform annotation-agnostic
differential expression analyses with <em><a href="https://bioconductor.org/packages/3.16/derfinder" class="external-link">derfinder</a></em>
(Collado-Torres, Nellore, Frazee, Wilks, Love, Langmead, Irizarry, Leek,
and Jaffe, 2017). An example analysis would like this:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Define expressed regions for study SRP009615, only for chromosome Y</span></span>
<span><span class="va">regions</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/expressed_regions.html">expressed_regions</a></span><span class="op">(</span><span class="st">"SRP009615"</span>, <span class="st">"chrY"</span>,</span>
<span>    cutoff <span class="op">=</span> <span class="fl">5L</span>,</span>
<span>    maxClusterGap <span class="op">=</span> <span class="fl">3000L</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Compute coverage matrix for study SRP009615, only for chromosome Y</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/system.time.html" class="external-link">system.time</a></span><span class="op">(</span><span class="va">rse_ER</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/coverage_matrix.html">coverage_matrix</a></span><span class="op">(</span><span class="st">"SRP009615"</span>, <span class="st">"chrY"</span>, <span class="va">regions</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Round the coverage matrix to integers</span></span>
<span><span class="va">covMat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">rse_ER</span><span class="op">)</span><span class="op">$</span><span class="va">counts</span>, <span class="fl">0</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Get phenotype data for study SRP009615</span></span>
<span><span class="va">pheno</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_ER</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Complete the phenotype table with the data we got from GEO</span></span>
<span><span class="va">m</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">match</a></span><span class="op">(</span><span class="va">pheno</span><span class="op">$</span><span class="va">run</span>, <span class="va">sample_info</span><span class="op">$</span><span class="va">run</span><span class="op">)</span></span>
<span><span class="va">pheno</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="va">pheno</span>, <span class="va">sample_info</span><span class="op">[</span><span class="va">m</span>, <span class="fl">2</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Build a DESeqDataSet</span></span>
<span><span class="va">dds_ers</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeqDataSet.html" class="external-link">DESeqDataSetFromMatrix</a></span><span class="op">(</span></span>
<span>    countData <span class="op">=</span> <span class="va">covMat</span>, colData <span class="op">=</span> <span class="va">pheno</span>,</span>
<span>    design <span class="op">=</span> <span class="op">~</span> <span class="va">gene_target</span> <span class="op">+</span> <span class="va">group</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Perform differential expression analysis with DESeq2 at the ER-level</span></span>
<span><span class="va">dds_ers</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeq.html" class="external-link">DESeq</a></span><span class="op">(</span><span class="va">dds_ers</span>,</span>
<span>    test <span class="op">=</span> <span class="st">"LRT"</span>, reduced <span class="op">=</span> <span class="op">~</span><span class="va">gene_target</span>,</span>
<span>    fitType <span class="op">=</span> <span class="st">"local"</span></span>
<span><span class="op">)</span></span>
<span><span class="va">res_ers</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/results.html" class="external-link">results</a></span><span class="op">(</span><span class="va">dds_ers</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Explore results</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/plotMA.html" class="external-link">plotMA</a></span><span class="op">(</span><span class="va">res_ers</span>, main <span class="op">=</span> <span class="st">"DESeq2 results for SRP009615 (ER-level, chrY)"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Create a more extensive exploratory report</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/regionReport/man/DESeq2Report.html" class="external-link">DESeq2Report</a></span><span class="op">(</span><span class="va">dds_ers</span>,</span>
<span>    res <span class="op">=</span> <span class="va">res_ers</span>,</span>
<span>    project <span class="op">=</span> <span class="st">"SRP009615 (ER-level, chrY)"</span>,</span>
<span>    intgroup <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"group"</span>, <span class="st">"gene_target"</span><span class="op">)</span>, outdir <span class="op">=</span> <span class="st">"."</span>,</span>
<span>    output <span class="op">=</span> <span class="st">"SRP009615-results-ER-level-chrY"</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p><em><a href="https://bioconductor.org/packages/3.16/recount" class="external-link">recount</a></em>
is an R package that provides an interface to the <a href="https://jhubiostatistics.shinyapps.io/recount/" class="external-link">recount project
website</a>. This package allows you to download the files from the
recount project and has helper functions for getting you started with
differential expression analyses. This vignette will walk you through an
example.</p>
</div>
<div class="section level2">
<h2 id="sample-de-analysis">Sample DE analysis<a class="anchor" aria-label="anchor" href="#sample-de-analysis"></a>
</h2>
<p>This is a brief overview of what you can do with <em><a href="https://bioconductor.org/packages/3.16/recount" class="external-link">recount</a></em>.
In this particular example we will download data from the <a href="http://trace.ncbi.nlm.nih.gov/Traces/sra/?study=SRP009615" class="external-link">SRP009615</a>
study which sequenced 12 samples as described in the previous link.</p>
<p>If you don’t have <em><a href="https://bioconductor.org/packages/3.16/recount" class="external-link">recount</a></em>
installed, please do so with:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"recount"</span><span class="op">)</span></span></code></pre></div>
<p>Next we load the required packages. Loading <em><a href="https://bioconductor.org/packages/3.16/recount" class="external-link">recount</a></em>
will load the required dependencies.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Load recount R package</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/leekgroup/recount" class="external-link">"recount"</a></span><span class="op">)</span></span></code></pre></div>
<p>Lets say that we don’t know the actual SRA accession number for this
study but we do know a particular term which will help us identify it.
If that’s the case, we can use the <code><a href="../reference/abstract_search.html">abstract_search()</a></code>
function to identify the study of interest as shown below.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Find a project of interest</span></span>
<span><span class="va">project_info</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/abstract_search.html">abstract_search</a></span><span class="op">(</span><span class="st">"GSE32465"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Explore info</span></span>
<span><span class="va">project_info</span></span></code></pre></div>
<pre><code><span><span class="co">##     number_samples species</span></span>
<span><span class="co">## 340             12   human</span></span>
<span><span class="co">##                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               abstract</span></span>
<span><span class="co">## 340 Summary: K562-shX cells are made in an effort to validate TFBS data and ChIP-seq antibodies in Myers lab (GSE32465). K562 cells are transduced with lentiviral vector having Tet-inducible shRNA targeting a transcription factor gene. Cells with stable integration of shRNA constructs are selected using puromycin in growth media. Doxycycline is added to the growth media to induce the expression of shRNA and a red fluorescent protein marker.  A successful shRNA cell line shows at least a 70% reduction in expression of the target transcription factor as measured by qPCR. For identification, we designated these cell lines as K562-shX, where X is the transcription factor targeted by shRNA and K562 denotes the parent cell line.  For example, K562-shATF3 cells are K562 derived cells selected for stable integration of shRNA targeting the transcription factor ATF3 gene and showed at least a 70% reduction in the expression of ATF3 gene when measured by qPCR. Cells growing without doxycycline (uninduced) are used as a control to measure the change in expression of target transcription factor gene after induction of shRNA using doxycycline. For detailed growth and culturing protocols for these cells please refer to http://hudsonalpha.org/myers-lab/protocols . To identify the potential downstream targets of the candidate transcription factor, analyze the mRNA expression profile of the uninduced and induced K562-shX using RNA-seq.    For data usage terms and conditions, please refer to http://www.genome.gov/27528022 and http://www.genome.gov/Pages/Research/ENCODE/ENCODEDataReleasePolicyFinal2008.pdf    Overall Design: Make K562-shX cells as described in the http://hudsonalpha.org/myers-lab/protocols . Measure the mRNA expression levels in uninduced K562-shX and induced K562-shX cells in two biological replicates using RNA-seq. Identify the potential downstream targets of the candidate transcription factor.</span></span>
<span><span class="co">##       project</span></span>
<span><span class="co">## 340 SRP009615</span></span></code></pre>
<div class="section level3">
<h3 id="download-data">Download data<a class="anchor" aria-label="anchor" href="#download-data"></a>
</h3>
<p>Now that we have a study that we are interested in, we can download
the <em>RangedSummarizedExperiment</em> object (see <em><a href="https://bioconductor.org/packages/3.16/SummarizedExperiment" class="external-link">SummarizedExperiment</a></em>)
with the data summarized at the gene level. The function
<code><a href="../reference/download_study.html">download_study()</a></code> helps us do this. If you are interested on
how the annotation was defined, check <code><a href="../reference/reproduce_ranges.html">reproduce_ranges()</a></code>
described in the <em>Annotation</em> section further down.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Download the gene-level RangedSummarizedExperiment data</span></span>
<span><span class="fu"><a href="../reference/download_study.html">download_study</a></span><span class="op">(</span><span class="va">project_info</span><span class="op">$</span><span class="va">project</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 2023-05-07 01:07:06 downloading file rse_gene.Rdata to SRP009615</span></span></code></pre>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Load the data</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">project_info</span><span class="op">$</span><span class="va">project</span>, <span class="st">"rse_gene.Rdata"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Delete it if you don't need it anymore</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/unlink.html" class="external-link">unlink</a></span><span class="op">(</span><span class="va">project_info</span><span class="op">$</span><span class="va">project</span>, recursive <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p>We can explore a bit this <em>RangedSummarizedExperiment</em> as
shown below.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">rse_gene</span></span></code></pre></div>
<pre><code><span><span class="co">## class: RangedSummarizedExperiment </span></span>
<span><span class="co">## dim: 58037 12 </span></span>
<span><span class="co">## metadata(0):</span></span>
<span><span class="co">## assays(1): counts</span></span>
<span><span class="co">## rownames(58037): ENSG00000000003.14 ENSG00000000005.5 ...</span></span>
<span><span class="co">##   ENSG00000283698.1 ENSG00000283699.1</span></span>
<span><span class="co">## rowData names(3): gene_id bp_length symbol</span></span>
<span><span class="co">## colnames(12): SRR387777 SRR387778 ... SRR389083 SRR389084</span></span>
<span><span class="co">## colData names(21): project sample ... title characteristics</span></span></code></pre>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## This is the sample phenotype data provided by the recount project</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_gene</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## DataFrame with 12 rows and 21 columns</span></span>
<span><span class="co">##               project      sample  experiment         run</span></span>
<span><span class="co">##           &lt;character&gt; &lt;character&gt; &lt;character&gt; &lt;character&gt;</span></span>
<span><span class="co">## SRR387777   SRP009615   SRS281685   SRX110461   SRR387777</span></span>
<span><span class="co">## SRR387778   SRP009615   SRS281686   SRX110462   SRR387778</span></span>
<span><span class="co">## SRR387779   SRP009615   SRS281687   SRX110463   SRR387779</span></span>
<span><span class="co">## SRR387780   SRP009615   SRS281688   SRX110464   SRR387780</span></span>
<span><span class="co">## SRR389077   SRP009615   SRS282369   SRX111299   SRR389077</span></span>
<span><span class="co">## ...               ...         ...         ...         ...</span></span>
<span><span class="co">## SRR389080   SRP009615   SRS282372   SRX111302   SRR389080</span></span>
<span><span class="co">## SRR389081   SRP009615   SRS282373   SRX111303   SRR389081</span></span>
<span><span class="co">## SRR389082   SRP009615   SRS282374   SRX111304   SRR389082</span></span>
<span><span class="co">## SRR389083   SRP009615   SRS282375   SRX111305   SRR389083</span></span>
<span><span class="co">## SRR389084   SRP009615   SRS282376   SRX111306   SRR389084</span></span>
<span><span class="co">##           read_count_as_reported_by_sra reads_downloaded</span></span>
<span><span class="co">##                               &lt;integer&gt;        &lt;integer&gt;</span></span>
<span><span class="co">## SRR387777                      30631853         30631853</span></span>
<span><span class="co">## SRR387778                      37001306         37001306</span></span>
<span><span class="co">## SRR387779                      40552001         40552001</span></span>
<span><span class="co">## SRR387780                      32466352         32466352</span></span>
<span><span class="co">## SRR389077                      27819603         27819603</span></span>
<span><span class="co">## ...                                 ...              ...</span></span>
<span><span class="co">## SRR389080                      34856203         34856203</span></span>
<span><span class="co">## SRR389081                      23351679         23351679</span></span>
<span><span class="co">## SRR389082                      18144828         18144828</span></span>
<span><span class="co">## SRR389083                      24417368         24417368</span></span>
<span><span class="co">## SRR389084                      23060084         23060084</span></span>
<span><span class="co">##           proportion_of_reads_reported_by_sra_downloaded paired_end</span></span>
<span><span class="co">##                                                &lt;numeric&gt;  &lt;logical&gt;</span></span>
<span><span class="co">## SRR387777                                              1      FALSE</span></span>
<span><span class="co">## SRR387778                                              1      FALSE</span></span>
<span><span class="co">## SRR387779                                              1      FALSE</span></span>
<span><span class="co">## SRR387780                                              1      FALSE</span></span>
<span><span class="co">## SRR389077                                              1      FALSE</span></span>
<span><span class="co">## ...                                                  ...        ...</span></span>
<span><span class="co">## SRR389080                                              1      FALSE</span></span>
<span><span class="co">## SRR389081                                              1      FALSE</span></span>
<span><span class="co">## SRR389082                                              1      FALSE</span></span>
<span><span class="co">## SRR389083                                              1      FALSE</span></span>
<span><span class="co">## SRR389084                                              1      FALSE</span></span>
<span><span class="co">##           sra_misreported_paired_end mapped_read_count        auc</span></span>
<span><span class="co">##                            &lt;logical&gt;         &lt;integer&gt;  &lt;numeric&gt;</span></span>
<span><span class="co">## SRR387777                      FALSE          28798572 1029494445</span></span>
<span><span class="co">## SRR387778                      FALSE          33170281 1184877985</span></span>
<span><span class="co">## SRR387779                      FALSE          37322762 1336528969</span></span>
<span><span class="co">## SRR387780                      FALSE          29970735 1073178116</span></span>
<span><span class="co">## SRR389077                      FALSE          24966859  893978355</span></span>
<span><span class="co">## ...                              ...               ...        ...</span></span>
<span><span class="co">## SRR389080                      FALSE          32469994 1163527939</span></span>
<span><span class="co">## SRR389081                      FALSE          21904197  781685955</span></span>
<span><span class="co">## SRR389082                      FALSE          17199795  616048853</span></span>
<span><span class="co">## SRR389083                      FALSE          22499386  806323346</span></span>
<span><span class="co">## SRR389084                      FALSE          21957003  787795710</span></span>
<span><span class="co">##           sharq_beta_tissue sharq_beta_cell_type biosample_submission_date</span></span>
<span><span class="co">##                 &lt;character&gt;          &lt;character&gt;               &lt;character&gt;</span></span>
<span><span class="co">## SRR387777             blood                 k562    2011-12-05T15:40:03...</span></span>
<span><span class="co">## SRR387778             blood                 k562    2011-12-05T15:40:03...</span></span>
<span><span class="co">## SRR387779             blood                 k562    2011-12-05T15:40:03...</span></span>
<span><span class="co">## SRR387780             blood                 k562    2011-12-05T15:40:03...</span></span>
<span><span class="co">## SRR389077             blood                 k562    2011-12-13T11:26:05...</span></span>
<span><span class="co">## ...                     ...                  ...                       ...</span></span>
<span><span class="co">## SRR389080             blood                 k562    2011-12-13T11:26:05...</span></span>
<span><span class="co">## SRR389081             blood                 k562    2011-12-13T11:26:05...</span></span>
<span><span class="co">## SRR389082             blood                 k562    2011-12-13T11:26:05...</span></span>
<span><span class="co">## SRR389083             blood                 k562    2011-12-13T11:26:05...</span></span>
<span><span class="co">## SRR389084             blood                 k562    2011-12-13T11:26:05...</span></span>
<span><span class="co">##           biosample_publication_date  biosample_update_date avg_read_length</span></span>
<span><span class="co">##                          &lt;character&gt;            &lt;character&gt;       &lt;integer&gt;</span></span>
<span><span class="co">## SRR387777     2011-12-07T09:29:59... 2014-08-27T04:18:20...              36</span></span>
<span><span class="co">## SRR387778     2011-12-07T09:29:59... 2014-08-27T04:18:21...              36</span></span>
<span><span class="co">## SRR387779     2011-12-07T09:29:59... 2014-08-27T04:18:21...              36</span></span>
<span><span class="co">## SRR387780     2011-12-07T09:29:59... 2014-08-27T04:18:22...              36</span></span>
<span><span class="co">## SRR389077     2011-12-13T11:26:06... 2014-08-27T04:22:14...              36</span></span>
<span><span class="co">## ...                              ...                    ...             ...</span></span>
<span><span class="co">## SRR389080     2011-12-13T11:26:06... 2014-08-27T04:22:15...              36</span></span>
<span><span class="co">## SRR389081     2011-12-13T11:26:06... 2014-08-27T04:22:16...              36</span></span>
<span><span class="co">## SRR389082     2011-12-13T11:26:06... 2014-08-27T04:22:16...              36</span></span>
<span><span class="co">## SRR389083     2011-12-13T11:26:06... 2014-08-27T04:22:17...              36</span></span>
<span><span class="co">## SRR389084     2011-12-13T11:26:06... 2014-08-27T04:22:17...              36</span></span>
<span><span class="co">##           geo_accession  bigwig_file                  title</span></span>
<span><span class="co">##             &lt;character&gt;  &lt;character&gt;            &lt;character&gt;</span></span>
<span><span class="co">## SRR387777     GSM836270 SRR387777.bw K562 cells with shRN..</span></span>
<span><span class="co">## SRR387778     GSM836271 SRR387778.bw K562 cells with shRN..</span></span>
<span><span class="co">## SRR387779     GSM836272 SRR387779.bw K562 cells with shRN..</span></span>
<span><span class="co">## SRR387780     GSM836273 SRR387780.bw K562 cells with shRN..</span></span>
<span><span class="co">## SRR389077     GSM847561 SRR389077.bw K562 cells with shRN..</span></span>
<span><span class="co">## ...                 ...          ...                    ...</span></span>
<span><span class="co">## SRR389080     GSM847564 SRR389080.bw K562 cells with shRN..</span></span>
<span><span class="co">## SRR389081     GSM847565 SRR389081.bw K562 cells with shRN..</span></span>
<span><span class="co">## SRR389082     GSM847566 SRR389082.bw K562 cells with shRN..</span></span>
<span><span class="co">## SRR389083     GSM847567 SRR389083.bw K562 cells with shRN..</span></span>
<span><span class="co">## SRR389084     GSM847568 SRR389084.bw K562 cells with shRN..</span></span>
<span><span class="co">##                                                         characteristics</span></span>
<span><span class="co">##                                                         &lt;CharacterList&gt;</span></span>
<span><span class="co">## SRR387777         cells: K562,shRNA expression: no,treatment: Puromycin</span></span>
<span><span class="co">## SRR387778     cells: K562,shRNA expression: ye..,treatment: Puromycin..</span></span>
<span><span class="co">## SRR387779         cells: K562,shRNA expression: no,treatment: Puromycin</span></span>
<span><span class="co">## SRR387780     cells: K562,shRNA expression: ye..,treatment: Puromycin..</span></span>
<span><span class="co">## SRR389077   cell line: K562,shRNA expression: no..,treatment: Puromycin</span></span>
<span><span class="co">## ...                                                                 ...</span></span>
<span><span class="co">## SRR389080 cell line: K562,shRNA expression: ex..,treatment: Puromycin..</span></span>
<span><span class="co">## SRR389081   cell line: K562,shRNA expression: no..,treatment: Puromycin</span></span>
<span><span class="co">## SRR389082 cell line: K562,shRNA expression: ex..,treatment: Puromycin..</span></span>
<span><span class="co">## SRR389083   cell line: K562,shRNA expression: no..,treatment: Puromycin</span></span>
<span><span class="co">## SRR389084 cell line: K562,shRNA expression: ex..,treatment: Puromycin..</span></span></code></pre>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## At the gene level, the row data includes the gene Gencode ids, the gene</span></span>
<span><span class="co">## symbols and the sum of the disjoint exons widths, which can be used for</span></span>
<span><span class="co">## taking into account the gene length.</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">rse_gene</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## DataFrame with 58037 rows and 3 columns</span></span>
<span><span class="co">##                               gene_id bp_length          symbol</span></span>
<span><span class="co">##                           &lt;character&gt; &lt;integer&gt; &lt;CharacterList&gt;</span></span>
<span><span class="co">## ENSG00000000003.14 ENSG00000000003.14      4535          TSPAN6</span></span>
<span><span class="co">## ENSG00000000005.5   ENSG00000000005.5      1610            TNMD</span></span>
<span><span class="co">## ENSG00000000419.12 ENSG00000000419.12      1207            DPM1</span></span>
<span><span class="co">## ENSG00000000457.13 ENSG00000000457.13      6883           SCYL3</span></span>
<span><span class="co">## ENSG00000000460.16 ENSG00000000460.16      5967        C1orf112</span></span>
<span><span class="co">## ...                               ...       ...             ...</span></span>
<span><span class="co">## ENSG00000283695.1   ENSG00000283695.1        61              NA</span></span>
<span><span class="co">## ENSG00000283696.1   ENSG00000283696.1       997              NA</span></span>
<span><span class="co">## ENSG00000283697.1   ENSG00000283697.1      1184           HSFX3</span></span>
<span><span class="co">## ENSG00000283698.1   ENSG00000283698.1       940              NA</span></span>
<span><span class="co">## ENSG00000283699.1   ENSG00000283699.1        60         MIR4481</span></span></code></pre>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## At the exon level, you can get the gene Gencode ids from the names of:</span></span>
<span><span class="co"># rowRanges(rse_exon)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="finding-phenotype-information">Finding phenotype information<a class="anchor" aria-label="anchor" href="#finding-phenotype-information"></a>
</h3>
<p>Once we have identified the study of interest, we can use the
<code><a href="../reference/browse_study.html">browse_study()</a></code> function to browse the study at the SRA
website.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Browse the project at SRA</span></span>
<span><span class="fu"><a href="../reference/browse_study.html">browse_study</a></span><span class="op">(</span><span class="va">project_info</span><span class="op">$</span><span class="va">project</span><span class="op">)</span></span></code></pre></div>
<p>The SRA website includes an <em>Experiments</em> link which further
describes each of the samples. From the information available for <a href="http://www.ncbi.nlm.nih.gov/sra/?term=SRP009615" class="external-link">SRP009615 at
NCBI</a> we have some further sample information that we can save for
use in our differential expression analysis. We can get some of this
information from <a href="http://www.ncbi.nlm.nih.gov/geo/" class="external-link">GEO</a>. The
function <code><a href="../reference/find_geo.html">find_geo()</a></code> finds the <a href="http://www.ncbi.nlm.nih.gov/geo/" class="external-link">GEO</a> accession id for a given
SRA run accession id, which we can then use with <code><a href="../reference/geo_info.html">geo_info()</a></code>
and <code><a href="../reference/geo_characteristics.html">geo_characteristics()</a></code> to parse this information. The
<code>rse_gene</code> object already has some of this information.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## View GEO ids</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_gene</span><span class="op">)</span><span class="op">$</span><span class="va">geo_accession</span></span></code></pre></div>
<pre><code><span><span class="co">##  [1] "GSM836270" "GSM836271" "GSM836272" "GSM836273" "GSM847561" "GSM847562"</span></span>
<span><span class="co">##  [7] "GSM847563" "GSM847564" "GSM847565" "GSM847566" "GSM847567" "GSM847568"</span></span></code></pre>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Extract the sample characteristics</span></span>
<span><span class="va">geochar</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/splitAsList.html" class="external-link">split</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_gene</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_gene</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>, <span class="va">geo_characteristics</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Note that the information for this study is a little inconsistent, so we</span></span>
<span><span class="co">## have to fix it.</span></span>
<span><span class="va">geochar</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/do.call.html" class="external-link">do.call</a></span><span class="op">(</span><span class="va">rbind</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">geochar</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="kw">if</span> <span class="op">(</span><span class="st">"cells"</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>        <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">==</span> <span class="st">"cells"</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"cell.line"</span></span>
<span>        <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span>    <span class="op">}</span> <span class="kw">else</span> <span class="op">{</span></span>
<span>        <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span>    <span class="op">}</span></span>
<span><span class="op">}</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## We can now define some sample information to use</span></span>
<span><span class="va">sample_info</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span>
<span>    run <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_gene</span><span class="op">)</span><span class="op">$</span><span class="va">run</span>,</span>
<span>    group <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"uninduced"</span>, <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_gene</span><span class="op">)</span><span class="op">$</span><span class="va">title</span><span class="op">)</span>, <span class="st">"uninduced"</span>, <span class="st">"induced"</span><span class="op">)</span>,</span>
<span>    gene_target <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_gene</span><span class="op">)</span><span class="op">$</span><span class="va">title</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span></span>
<span>        <span class="fu"><a href="https://rdrr.io/r/base/strsplit.html" class="external-link">strsplit</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/strsplit.html" class="external-link">strsplit</a></span><span class="op">(</span></span>
<span>            <span class="va">x</span>,</span>
<span>            <span class="st">"targeting "</span></span>
<span>        <span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span>, <span class="st">" gene"</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span></span>
<span>    <span class="op">}</span><span class="op">)</span>,</span>
<span>    cell.line <span class="op">=</span> <span class="va">geochar</span><span class="op">$</span><span class="va">cell.line</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="predicted-phenotype-information">Predicted phenotype information<a class="anchor" aria-label="anchor" href="#predicted-phenotype-information"></a>
</h3>
<p>Shannon Ellis et at have predicted phenotypic information that can be
used with any data from the recount project thanks to
<code><a href="../reference/add_predictions.html">add_predictions()</a></code>. Check that function for more details
(Ellis, Collado-Torres, Jaffe, and Leek, 2018).</p>
</div>
<div class="section level3">
<h3 id="de-analysis">DE analysis<a class="anchor" aria-label="anchor" href="#de-analysis"></a>
</h3>
<p>The <a href="https://jhubiostatistics.shinyapps.io/recount/" class="external-link">recount
project</a> records the sum of the base level coverage for each gene (or
exon). These raw counts have to be scaled and there are several ways in
which you can choose to do so. The function <code><a href="../reference/scale_counts.html">scale_counts()</a></code>
helps you scale them in a way that is tailored to <a href="http://rail.bio" class="external-link">Rail-RNA</a> output. If you prefer read counts
without scaling, check the function <code><a href="../reference/read_counts.html">read_counts()</a></code>. Below we
show some of the differences.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Scale counts by taking into account the total coverage per sample</span></span>
<span><span class="va">rse</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/scale_counts.html">scale_counts</a></span><span class="op">(</span><span class="va">rse_gene</span><span class="op">)</span></span>
<span></span>
<span><span class="co">##### Details about counts #####</span></span>
<span></span>
<span><span class="co">## Scale counts to 40 million mapped reads. Not all mapped reads are in exonic</span></span>
<span><span class="co">## sequence, so the total is not necessarily 40 million.</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/colSums.html" class="external-link">colSums</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">rse</span><span class="op">)</span><span class="op">$</span><span class="va">counts</span><span class="op">)</span> <span class="op">/</span> <span class="fl">1e6</span></span></code></pre></div>
<pre><code><span><span class="co">## SRR387777 SRR387778 SRR387779 SRR387780 SRR389077 SRR389078 SRR389079 SRR389080 </span></span>
<span><span class="co">##  30.26702  29.07199  35.38355  34.65798  23.36050  22.62014  34.74629  35.20971 </span></span>
<span><span class="co">## SRR389081 SRR389082 SRR389083 SRR389084 </span></span>
<span><span class="co">##  33.41459  36.78133  34.05013  33.95469</span></span></code></pre>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Compute read counts</span></span>
<span><span class="va">rse_read_counts</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/read_counts.html">read_counts</a></span><span class="op">(</span><span class="va">rse_gene</span><span class="op">)</span></span>
<span><span class="co">## Difference between read counts and number of reads downloaded by Rail-RNA</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/colSums.html" class="external-link">colSums</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">rse_read_counts</span><span class="op">)</span><span class="op">$</span><span class="va">counts</span><span class="op">)</span> <span class="op">/</span> <span class="fl">1e6</span> <span class="op">-</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_read_counts</span><span class="op">)</span><span class="op">$</span><span class="va">reads_downloaded</span> <span class="op">/</span> <span class="fl">1e6</span></span></code></pre></div>
<pre><code><span><span class="co">##  SRR387777  SRR387778  SRR387779  SRR387780  SRR389077  SRR389078  SRR389079 </span></span>
<span><span class="co">##  -8.839994 -12.892541  -7.536284  -6.497571 -13.239233 -15.816168 -14.775568 </span></span>
<span><span class="co">##  SRR389080  SRR389081  SRR389082  SRR389083  SRR389084 </span></span>
<span><span class="co">##  -6.274001  -5.054220  -2.328694  -5.265160  -4.421728</span></span></code></pre>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Check the help page for read_counts() for more details</span></span></code></pre></div>
<p>We are almost ready to perform our differential expression analysis.
Lets just add the information we recovered <a href="http://www.ncbi.nlm.nih.gov/geo/" class="external-link">GEO</a> about these samples.</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Add sample information for DE analysis</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse</span><span class="op">)</span><span class="op">$</span><span class="va">group</span> <span class="op">&lt;-</span> <span class="va">sample_info</span><span class="op">$</span><span class="va">group</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse</span><span class="op">)</span><span class="op">$</span><span class="va">gene_target</span> <span class="op">&lt;-</span> <span class="va">sample_info</span><span class="op">$</span><span class="va">gene_target</span></span></code></pre></div>
<p>Now that the <em>RangedSummarizedExperiment</em> is complete, we can
use <em><a href="https://bioconductor.org/packages/3.16/DESeq2" class="external-link">DESeq2</a></em> or
another package to perform the differential expression test. Note that
you can use <em><a href="https://bioconductor.org/packages/3.16/DEFormats" class="external-link">DEFormats</a></em>
for switching between formats if you want to use another package, like
<em><a href="https://bioconductor.org/packages/3.16/edgeR" class="external-link">edgeR</a></em>.</p>
<p>In this particular analysis, we’ll test whether there is a group
difference adjusting for the gene target.</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Perform differential expression analysis with DESeq2</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/mikelove/DESeq2" class="external-link">"DESeq2"</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Specify design and switch to DESeq2 format</span></span>
<span><span class="va">dds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeqDataSet.html" class="external-link">DESeqDataSet</a></span><span class="op">(</span><span class="va">rse</span>, <span class="op">~</span> <span class="va">gene_target</span> <span class="op">+</span> <span class="va">group</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## converting counts to integer mode</span></span></code></pre>
<pre><code><span><span class="co">## Warning in DESeqDataSet(rse, ~gene_target + group): some variables in design</span></span>
<span><span class="co">## formula are characters, converting to factors</span></span></code></pre>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Perform DE analysis</span></span>
<span><span class="va">dds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeq.html" class="external-link">DESeq</a></span><span class="op">(</span><span class="va">dds</span>, test <span class="op">=</span> <span class="st">"LRT"</span>, reduced <span class="op">=</span> <span class="op">~</span><span class="va">gene_target</span>, fitType <span class="op">=</span> <span class="st">"local"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## estimating size factors</span></span></code></pre>
<pre><code><span><span class="co">## estimating dispersions</span></span></code></pre>
<pre><code><span><span class="co">## gene-wise dispersion estimates</span></span></code></pre>
<pre><code><span><span class="co">## mean-dispersion relationship</span></span></code></pre>
<pre><code><span><span class="co">## final dispersion estimates</span></span></code></pre>
<pre><code><span><span class="co">## fitting model and testing</span></span></code></pre>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/results.html" class="external-link">results</a></span><span class="op">(</span><span class="va">dds</span><span class="op">)</span></span></code></pre></div>
<p>We can now use functions from <em><a href="https://bioconductor.org/packages/3.16/DESeq2" class="external-link">DESeq2</a></em> to
explore the results. For more details check the <em><a href="https://bioconductor.org/packages/3.16/DESeq2" class="external-link">DESeq2</a></em>
vignette. For example, we can make a MA plot as shown in Figure
@ref(fig:maplot).</p>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Explore results</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/plotMA.html" class="external-link">plotMA</a></span><span class="op">(</span><span class="va">res</span>, main <span class="op">=</span> <span class="st">"DESeq2 results for SRP009615"</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="recount-quickstart_files/figure-html/maplot-1.png" alt="MA plot for group differences adjusted by gene target for SRP009615 using gene-level data." width="700"><p class="caption">
MA plot for group differences adjusted by gene target for SRP009615
using gene-level data.
</p>
</div>
</div>
<div class="section level3">
<h3 id="report-results">Report results<a class="anchor" aria-label="anchor" href="#report-results"></a>
</h3>
<p>We can also use the <em><a href="https://bioconductor.org/packages/3.16/regionReport" class="external-link">regionReport</a></em>
package to generate interactive HTML reports exploring the <em><a href="https://bioconductor.org/packages/3.16/DESeq2" class="external-link">DESeq2</a></em>
results (or <em><a href="https://bioconductor.org/packages/3.16/edgeR" class="external-link">edgeR</a></em>
results if you used that package).</p>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Make a report with the results</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/leekgroup/regionReport" class="external-link">"regionReport"</a></span><span class="op">)</span></span>
<span><span class="va">report</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/regionReport/man/DESeq2Report.html" class="external-link">DESeq2Report</a></span><span class="op">(</span><span class="va">dds</span>,</span>
<span>    res <span class="op">=</span> <span class="va">res</span>, project <span class="op">=</span> <span class="st">"SRP009615"</span>,</span>
<span>    intgroup <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"group"</span>, <span class="st">"gene_target"</span><span class="op">)</span>, outdir <span class="op">=</span> <span class="st">"."</span>,</span>
<span>    output <span class="op">=</span> <span class="st">"SRP009615-results"</span>, nBest <span class="op">=</span> <span class="fl">10</span>, nBestFeatures <span class="op">=</span> <span class="fl">2</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Loading required package: limma</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'limma'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:DESeq2':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     plotMA</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:BiocGenerics':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     plotMA</span></span></code></pre>
<pre><code><span><span class="co">## Writing 9 Bibtex entries ... OK</span></span>
<span><span class="co">## Results written to file './SRP009615-results.bib'</span></span>
<span><span class="co">## processing file: SRP009615-results.Rmd</span></span></code></pre>
<pre><code><span><span class="co">## Warning in doTryCatch(return(expr), name, parentenv, handler): "alpha" is not a graphical parameter</span></span>
<span></span>
<span><span class="co">## Warning in doTryCatch(return(expr), name, parentenv, handler): "alpha" is not a graphical parameter</span></span>
<span></span>
<span><span class="co">## Warning in doTryCatch(return(expr), name, parentenv, handler): "alpha" is not a graphical parameter</span></span>
<span></span>
<span><span class="co">## Warning in doTryCatch(return(expr), name, parentenv, handler): "alpha" is not a graphical parameter</span></span>
<span></span>
<span><span class="co">## Warning in doTryCatch(return(expr), name, parentenv, handler): "alpha" is not a graphical parameter</span></span>
<span></span>
<span><span class="co">## Warning in doTryCatch(return(expr), name, parentenv, handler): "alpha" is not a graphical parameter</span></span>
<span></span>
<span><span class="co">## Warning in doTryCatch(return(expr), name, parentenv, handler): "alpha" is not a graphical parameter</span></span>
<span></span>
<span><span class="co">## Warning in doTryCatch(return(expr), name, parentenv, handler): "alpha" is not a graphical parameter</span></span>
<span></span>
<span><span class="co">## Warning in doTryCatch(return(expr), name, parentenv, handler): "alpha" is not a graphical parameter</span></span>
<span></span>
<span><span class="co">## Warning in doTryCatch(return(expr), name, parentenv, handler): "alpha" is not a graphical parameter</span></span>
<span></span>
<span><span class="co">## Warning in doTryCatch(return(expr), name, parentenv, handler): "alpha" is not a graphical parameter</span></span>
<span></span>
<span><span class="co">## Warning in doTryCatch(return(expr), name, parentenv, handler): "alpha" is not a graphical parameter</span></span>
<span></span>
<span><span class="co">## Warning in doTryCatch(return(expr), name, parentenv, handler): "alpha" is not a graphical parameter</span></span>
<span></span>
<span><span class="co">## Warning in doTryCatch(return(expr), name, parentenv, handler): "alpha" is not a graphical parameter</span></span>
<span></span>
<span><span class="co">## Warning in doTryCatch(return(expr), name, parentenv, handler): "alpha" is not a graphical parameter</span></span>
<span></span>
<span><span class="co">## Warning in doTryCatch(return(expr), name, parentenv, handler): "alpha" is not a graphical parameter</span></span>
<span></span>
<span><span class="co">## Warning in doTryCatch(return(expr), name, parentenv, handler): "alpha" is not a graphical parameter</span></span>
<span></span>
<span><span class="co">## Warning in doTryCatch(return(expr), name, parentenv, handler): "alpha" is not a graphical parameter</span></span></code></pre>
<pre><code><span><span class="co">## output file: SRP009615-results.knit.md</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Output created: SRP009615-results.html</span></span></code></pre>
<p><img src="recount-quickstart_files/figure-html/make_report_real-1.png" width="700"></p>
<p>You can view the final report <a href="SRP009615-results.html">here</a>.</p>
</div>
<div class="section level3">
<h3 id="finding-gene-names">Finding gene names<a class="anchor" aria-label="anchor" href="#finding-gene-names"></a>
</h3>
<p>The gene Gencode ids are included in several objects in
<code>recount</code>. With these ids you can find the gene symbols, gene
names and other information. The <em><a href="https://bioconductor.org/packages/3.16/org.Hs.eg.db" class="external-link">org.Hs.eg.db</a></em>
is useful for finding this information and the following code shows how
to get the gene names and symbols.</p>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Load required library</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"org.Hs.eg.db"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Loading required package: AnnotationDbi</span></span></code></pre>
<pre><code><span><span class="co">## </span></span></code></pre>
<div class="sourceCode" id="cb53"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Extract Gencode gene ids</span></span>
<span><span class="va">gencode</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"\\..*"</span>, <span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">recount_genes</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Find the gene information we are interested in</span></span>
<span><span class="va">gene_info</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html" class="external-link">select</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span>, <span class="va">gencode</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>    <span class="st">"ENTREZID"</span>, <span class="st">"GENENAME"</span>, <span class="st">"SYMBOL"</span>,</span>
<span>    <span class="st">"ENSEMBL"</span></span>
<span><span class="op">)</span>, <span class="st">"ENSEMBL"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 'select()' returned many:many mapping between keys and columns</span></span></code></pre>
<div class="sourceCode" id="cb55"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Explore part of the results</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">gene_info</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 58290     4</span></span></code></pre>
<div class="sourceCode" id="cb57"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">gene_info</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##           ENSEMBL ENTREZID                                                    GENENAME   SYMBOL</span></span>
<span><span class="co">## 1 ENSG00000000003     7105                                               tetraspanin 6   TSPAN6</span></span>
<span><span class="co">## 2 ENSG00000000005    64102                                                 tenomodulin     TNMD</span></span>
<span><span class="co">## 3 ENSG00000000419     8813 dolichyl-phosphate mannosyltransferase subunit 1, catalytic     DPM1</span></span>
<span><span class="co">## 4 ENSG00000000457    57147                                    SCY1 like pseudokinase 3    SCYL3</span></span>
<span><span class="co">## 5 ENSG00000000460    55732                         chromosome 1 open reading frame 112 C1orf112</span></span>
<span><span class="co">## 6 ENSG00000000938     2268              FGR proto-oncogene, Src family tyrosine kinase      FGR</span></span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="sample-derfinder-analysis">Sample <em><a href="https://bioconductor.org/packages/3.16/derfinder" class="external-link">derfinder</a></em>
analysis<a class="anchor" aria-label="anchor" href="#sample-derfinder-analysis"></a>
</h2>
<p>The <a href="https://jhubiostatistics.shinyapps.io/recount/" class="external-link">recount
project</a> also hosts for each project sample coverage <a href="https://genome.ucsc.edu/goldenpath/help/bigWig.html" class="external-link">bigWig
files</a> created by <a href="http://rail.bio" class="external-link">Rail-RNA</a> and a mean
coverage bigWig file. For the mean coverage bigWig file, all samples
were normalized to libraries of 40 million reads, each a 100 base-pairs
long. <em><a href="https://bioconductor.org/packages/3.16/recount" class="external-link">recount</a></em>
can be used along with <em><a href="https://bioconductor.org/packages/3.16/derfinder" class="external-link">derfinder</a></em>
(Collado-Torres, Nellore, Frazee et al., 2017) to identify expressed
regions from the data. This type of analysis is annotation-agnostic
which can be advantageous. The following subsections illustrate this
type of analysis.</p>
<div class="section level3">
<h3 id="define-expressed-regions">Define expressed regions<a class="anchor" aria-label="anchor" href="#define-expressed-regions"></a>
</h3>
<p>For an annotation-agnostic differential expression analysis, we first
need to define the regions of interest. With <em><a href="https://bioconductor.org/packages/3.16/recount" class="external-link">recount</a></em>
we can do so using the <code><a href="../reference/expressed_regions.html">expressed_regions()</a></code> function as
shown below for the same study we studied earlier.</p>
<div class="sourceCode" id="cb59"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Define expressed regions for study SRP009615, only for chromosome Y</span></span>
<span><span class="va">regions</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/expressed_regions.html">expressed_regions</a></span><span class="op">(</span><span class="st">"SRP009615"</span>, <span class="st">"chrY"</span>,</span>
<span>    cutoff <span class="op">=</span> <span class="fl">5L</span>,</span>
<span>    maxClusterGap <span class="op">=</span> <span class="fl">3000L</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 2023-05-07 01:07:52 loadCoverage: loading BigWig file http://duffel.rail.bio/recount/SRP009615/bw/mean_SRP009615.bw</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:07:54 loadCoverage: applying the cutoff to the merged data</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:07:54 filterData: originally there were 57227415 rows, now there are 57227415 rows. Meaning that 0 percent was filtered.</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:07:54 findRegions: identifying potential segments</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:07:54 findRegions: segmenting information</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:07:54 .getSegmentsRle: segmenting with cutoff(s) 5</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:07:54 findRegions: identifying candidate regions</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:07:56 findRegions: identifying region clusters</span></span></code></pre>
<div class="sourceCode" id="cb68"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Briefly explore the resulting regions</span></span>
<span><span class="va">regions</span></span></code></pre></div>
<pre><code><span><span class="co">## GRanges object with 808 ranges and 6 metadata columns:</span></span>
<span><span class="co">##       seqnames            ranges strand |     value      area indexStart  indexEnd cluster clusterL</span></span>
<span><span class="co">##          &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt;  &lt;integer&gt; &lt;integer&gt;   &lt;Rle&gt;    &lt;Rle&gt;</span></span>
<span><span class="co">##     1     chrY   2929794-2929829      * |  14.72650   530.154    2929794   2929829       1       36</span></span>
<span><span class="co">##     2     chrY   2956678-2956701      * |  12.81063   307.455    2956678   2956701       2       24</span></span>
<span><span class="co">##     3     chrY   2977203-2977227      * |   5.34908   133.727    2977203   2977227       3     2750</span></span>
<span><span class="co">##     4     chrY   2977957-2977994      * |   6.46977   245.851    2977957   2977994       3     2750</span></span>
<span><span class="co">##     5     chrY   2978850-2978871      * |   5.79766   127.548    2978850   2978871       3     2750</span></span>
<span><span class="co">##   ...      ...               ...    ... .       ...       ...        ...       ...     ...      ...</span></span>
<span><span class="co">##   804     chrY 26614511-26614546      * |   7.28189  262.1482   26614511  26614546     224      298</span></span>
<span><span class="co">##   805     chrY 26614548-26614553      * |   5.48768   32.9261   26614548  26614553     224      298</span></span>
<span><span class="co">##   806     chrY 26614779-26614808      * |   6.64339  199.3018   26614779  26614808     224      298</span></span>
<span><span class="co">##   807     chrY 26626808-26626848      * |  12.60382  516.7564   26626808  26626848     225      221</span></span>
<span><span class="co">##   808     chrY 26626971-26627028      * |  14.16734  821.7055   26626971  26627028     225      221</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 1 sequence from an unspecified genome</span></span></code></pre>
<p>Once the regions have been defined, you can export them into a BED
file using <em><a href="https://bioconductor.org/packages/3.16/rtracklayer" class="external-link">rtracklayer</a></em>
or other file formats.</p>
</div>
<div class="section level3">
<h3 id="compute-coverage-matrix">Compute coverage matrix<a class="anchor" aria-label="anchor" href="#compute-coverage-matrix"></a>
</h3>
<p>Having defined the expressed regions, we can now compute a coverage
matrix for these regions. We can do so using the function
<code><a href="../reference/coverage_matrix.html">coverage_matrix()</a></code> from <em><a href="https://bioconductor.org/packages/3.16/recount" class="external-link">recount</a></em>
as shown below. It returns a <em>RangedSummarizedExperiment</em>
object.</p>
<div class="sourceCode" id="cb70"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Compute coverage matrix for study SRP009615, only for chromosome Y</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/system.time.html" class="external-link">system.time</a></span><span class="op">(</span><span class="va">rse_ER</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/coverage_matrix.html">coverage_matrix</a></span><span class="op">(</span><span class="st">"SRP009615"</span>, <span class="st">"chrY"</span>, <span class="va">regions</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 2023-05-07 01:07:58 railMatrix: processing regions 1 to 808</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:07:58 railMatrix: processing file http://duffel.rail.bio/recount/SRP009615/bw/SRR387777.bw</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:08:01 railMatrix: processing file http://duffel.rail.bio/recount/SRP009615/bw/SRR387778.bw</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:08:03 railMatrix: processing file http://duffel.rail.bio/recount/SRP009615/bw/SRR387779.bw</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:08:06 railMatrix: processing file http://duffel.rail.bio/recount/SRP009615/bw/SRR387780.bw</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:08:09 railMatrix: processing file http://duffel.rail.bio/recount/SRP009615/bw/SRR389077.bw</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:08:12 railMatrix: processing file http://duffel.rail.bio/recount/SRP009615/bw/SRR389078.bw</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:08:20 railMatrix: processing file http://duffel.rail.bio/recount/SRP009615/bw/SRR389079.bw</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:08:27 railMatrix: processing file http://duffel.rail.bio/recount/SRP009615/bw/SRR389080.bw</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:08:30 railMatrix: processing file http://duffel.rail.bio/recount/SRP009615/bw/SRR389081.bw</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:08:33 railMatrix: processing file http://duffel.rail.bio/recount/SRP009615/bw/SRR389082.bw</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:08:36 railMatrix: processing file http://duffel.rail.bio/recount/SRP009615/bw/SRR389083.bw</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:08:39 railMatrix: processing file http://duffel.rail.bio/recount/SRP009615/bw/SRR389084.bw</span></span></code></pre>
<pre><code><span><span class="co">##    user  system elapsed </span></span>
<span><span class="co">##   5.875   0.196  45.314</span></span></code></pre>
<div class="sourceCode" id="cb85"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Explore the RSE a bit</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">rse_ER</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 808  12</span></span></code></pre>
<div class="sourceCode" id="cb87"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">rse_ER</span></span></code></pre></div>
<pre><code><span><span class="co">## class: RangedSummarizedExperiment </span></span>
<span><span class="co">## dim: 808 12 </span></span>
<span><span class="co">## metadata(0):</span></span>
<span><span class="co">## assays(1): counts</span></span>
<span><span class="co">## rownames(808): 1 2 ... 807 808</span></span>
<span><span class="co">## rowData names(6): value area ... cluster clusterL</span></span>
<span><span class="co">## colnames(12): SRR387777 SRR387778 ... SRR389083 SRR389084</span></span>
<span><span class="co">## colData names(21): project sample ... title characteristics</span></span></code></pre>
<p>The resulting count matrix has one row per region and one column per
sample. The counts correspond to the number (or fraction) of reads
overlapping the regions and are scaled by default to a library size of
40 million reads. For some differential expression methods, you might
have to round this matrix into integers. We’ll use <em><a href="https://bioconductor.org/packages/3.16/DESeq2" class="external-link">DESeq2</a></em> to
identify which expressed regions are differentially expressed.</p>
<div class="sourceCode" id="cb89"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Round the coverage matrix to integers</span></span>
<span><span class="va">covMat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">rse_ER</span><span class="op">)</span><span class="op">$</span><span class="va">counts</span>, <span class="fl">0</span><span class="op">)</span></span></code></pre></div>
<p>You can use <code>scale = FALSE</code> if you want the raw base-pair
coverage counts and then scale them with <code><a href="../reference/scale_counts.html">scale_counts()</a></code>. If
you want integer counts, use <code>round = TRUE</code>.</p>
</div>
<div class="section level3">
<h3 id="construct-a-deseqdataset-object">Construct a <code>DESeqDataSet</code> object<a class="anchor" aria-label="anchor" href="#construct-a-deseqdataset-object"></a>
</h3>
<p>We first need to get some phenotype information for these samples
similar to the first analysis we did. We can get this data using
<code><a href="../reference/download_study.html">download_study()</a></code>.</p>
<div class="sourceCode" id="cb90"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Get phenotype data for study SRP009615</span></span>
<span><span class="va">pheno</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_ER</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Complete the phenotype table with the data we got from GEO</span></span>
<span><span class="va">m</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">match</a></span><span class="op">(</span><span class="va">pheno</span><span class="op">$</span><span class="va">run</span>, <span class="va">sample_info</span><span class="op">$</span><span class="va">run</span><span class="op">)</span></span>
<span><span class="va">pheno</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="va">pheno</span>, <span class="va">sample_info</span><span class="op">[</span><span class="va">m</span>, <span class="fl">2</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Explore the phenotype data a little bit</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">pheno</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## DataFrame with 6 rows and 23 columns</span></span>
<span><span class="co">##               project      sample  experiment         run read_count_as_reported_by_sra reads_downloaded</span></span>
<span><span class="co">##           &lt;character&gt; &lt;character&gt; &lt;character&gt; &lt;character&gt;                     &lt;integer&gt;        &lt;integer&gt;</span></span>
<span><span class="co">## SRR387777   SRP009615   SRS281685   SRX110461   SRR387777                      30631853         30631853</span></span>
<span><span class="co">## SRR387778   SRP009615   SRS281686   SRX110462   SRR387778                      37001306         37001306</span></span>
<span><span class="co">## SRR387779   SRP009615   SRS281687   SRX110463   SRR387779                      40552001         40552001</span></span>
<span><span class="co">## SRR387780   SRP009615   SRS281688   SRX110464   SRR387780                      32466352         32466352</span></span>
<span><span class="co">## SRR389077   SRP009615   SRS282369   SRX111299   SRR389077                      27819603         27819603</span></span>
<span><span class="co">## SRR389078   SRP009615   SRS282370   SRX111300   SRR389078                      31758658         31758658</span></span>
<span><span class="co">##           proportion_of_reads_reported_by_sra_downloaded paired_end sra_misreported_paired_end mapped_read_count</span></span>
<span><span class="co">##                                                &lt;integer&gt;  &lt;logical&gt;                  &lt;logical&gt;         &lt;integer&gt;</span></span>
<span><span class="co">## SRR387777                                              1      FALSE                      FALSE          28798572</span></span>
<span><span class="co">## SRR387778                                              1      FALSE                      FALSE          33170281</span></span>
<span><span class="co">## SRR387779                                              1      FALSE                      FALSE          37322762</span></span>
<span><span class="co">## SRR387780                                              1      FALSE                      FALSE          29970735</span></span>
<span><span class="co">## SRR389077                                              1      FALSE                      FALSE          24966859</span></span>
<span><span class="co">## SRR389078                                              1      FALSE                      FALSE          28190059</span></span>
<span><span class="co">##                  auc sharq_beta_tissue sharq_beta_cell_type biosample_submission_date biosample_publication_date</span></span>
<span><span class="co">##            &lt;integer&gt;       &lt;character&gt;          &lt;character&gt;               &lt;character&gt;                &lt;character&gt;</span></span>
<span><span class="co">## SRR387777 1029494445             blood                 k562    2011-12-05T15:40:03...     2011-12-07T09:29:59...</span></span>
<span><span class="co">## SRR387778 1184877985             blood                 k562    2011-12-05T15:40:03...     2011-12-07T09:29:59...</span></span>
<span><span class="co">## SRR387779 1336528969             blood                 k562    2011-12-05T15:40:03...     2011-12-07T09:29:59...</span></span>
<span><span class="co">## SRR387780 1073178116             blood                 k562    2011-12-05T15:40:03...     2011-12-07T09:29:59...</span></span>
<span><span class="co">## SRR389077  893978355             blood                 k562    2011-12-13T11:26:05...     2011-12-13T11:26:06...</span></span>
<span><span class="co">## SRR389078 1009216922             blood                 k562    2011-12-13T11:26:05...     2011-12-13T11:26:06...</span></span>
<span><span class="co">##            biosample_update_date avg_read_length geo_accession  bigwig_file                  title</span></span>
<span><span class="co">##                      &lt;character&gt;       &lt;integer&gt;   &lt;character&gt;  &lt;character&gt;            &lt;character&gt;</span></span>
<span><span class="co">## SRR387777 2014-08-27T04:18:20...              36     GSM836270 SRR387777.bw K562 cells with shRN..</span></span>
<span><span class="co">## SRR387778 2014-08-27T04:18:21...              36     GSM836271 SRR387778.bw K562 cells with shRN..</span></span>
<span><span class="co">## SRR387779 2014-08-27T04:18:21...              36     GSM836272 SRR387779.bw K562 cells with shRN..</span></span>
<span><span class="co">## SRR387780 2014-08-27T04:18:22...              36     GSM836273 SRR387780.bw K562 cells with shRN..</span></span>
<span><span class="co">## SRR389077 2014-08-27T04:22:14...              36     GSM847561 SRR389077.bw K562 cells with shRN..</span></span>
<span><span class="co">## SRR389078 2014-08-27T04:22:15...              36     GSM847562 SRR389078.bw K562 cells with shRN..</span></span>
<span><span class="co">##                  characteristics       group gene_target</span></span>
<span><span class="co">##                      &lt;character&gt; &lt;character&gt; &lt;character&gt;</span></span>
<span><span class="co">## SRR387777 c("cells: K562", "sh..   uninduced         SRF</span></span>
<span><span class="co">## SRR387778 c("cells: K562", "sh..     induced         SRF</span></span>
<span><span class="co">## SRR387779 c("cells: K562", "sh..   uninduced         SRF</span></span>
<span><span class="co">## SRR387780 c("cells: K562", "sh..     induced         SRF</span></span>
<span><span class="co">## SRR389077 c("cell line: K562",..   uninduced        EGR1</span></span>
<span><span class="co">## SRR389078 c("cell line: K562",..     induced        EGR1</span></span></code></pre>
<p>Now that we have the necessary data, we can construct a
<code>DESeqDataSet</code> object using the function
<code><a href="https://rdrr.io/pkg/DESeq2/man/DESeqDataSet.html" class="external-link">DESeqDataSetFromMatrix()</a></code> from <em><a href="https://bioconductor.org/packages/3.16/DESeq2" class="external-link">DESeq2</a></em>.</p>
<div class="sourceCode" id="cb92"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Build a DESeqDataSet</span></span>
<span><span class="va">dds_ers</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeqDataSet.html" class="external-link">DESeqDataSetFromMatrix</a></span><span class="op">(</span></span>
<span>    countData <span class="op">=</span> <span class="va">covMat</span>, colData <span class="op">=</span> <span class="va">pheno</span>,</span>
<span>    design <span class="op">=</span> <span class="op">~</span> <span class="va">gene_target</span> <span class="op">+</span> <span class="va">group</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## converting counts to integer mode</span></span></code></pre>
<pre><code><span><span class="co">## Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in design formula are characters, converting</span></span>
<span><span class="co">## to factors</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="deseq2-analysis">
<em><a href="https://bioconductor.org/packages/3.16/DESeq2" class="external-link">DESeq2</a></em>
analysis<a class="anchor" aria-label="anchor" href="#deseq2-analysis"></a>
</h3>
<p>With the <code>DESeqDataSet</code> object in place, we can then use
the function <code><a href="https://rdrr.io/pkg/DESeq2/man/DESeq.html" class="external-link">DESeq()</a></code> from <em><a href="https://bioconductor.org/packages/3.16/DESeq2" class="external-link">DESeq2</a></em> to
perform the differential expression analysis (between groups) as shown
below.</p>
<div class="sourceCode" id="cb95"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Perform differential expression analysis with DESeq2 at the ER-level</span></span>
<span><span class="va">dds_ers</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeq.html" class="external-link">DESeq</a></span><span class="op">(</span><span class="va">dds_ers</span>,</span>
<span>    test <span class="op">=</span> <span class="st">"LRT"</span>, reduced <span class="op">=</span> <span class="op">~</span><span class="va">gene_target</span>,</span>
<span>    fitType <span class="op">=</span> <span class="st">"local"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## estimating size factors</span></span></code></pre>
<pre><code><span><span class="co">## estimating dispersions</span></span></code></pre>
<pre><code><span><span class="co">## gene-wise dispersion estimates</span></span></code></pre>
<pre><code><span><span class="co">## mean-dispersion relationship</span></span></code></pre>
<pre><code><span><span class="co">## final dispersion estimates</span></span></code></pre>
<pre><code><span><span class="co">## fitting model and testing</span></span></code></pre>
<div class="sourceCode" id="cb102"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res_ers</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/results.html" class="external-link">results</a></span><span class="op">(</span><span class="va">dds_ers</span><span class="op">)</span></span></code></pre></div>
<p>We can then visually explore the results as shown in Figure
@ref(fig:maploters), like we did before in Figure @ref(fig:maplot).</p>
<div class="sourceCode" id="cb103"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Explore results</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/plotMA.html" class="external-link">plotMA</a></span><span class="op">(</span><span class="va">res_ers</span>, main <span class="op">=</span> <span class="st">"DESeq2 results for SRP009615 (ER-level, chrY)"</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="recount-quickstart_files/figure-html/maploters-1.png" alt="MA plot for group differences adjusted by gene target for SRP009615 using ER-level data (just chrY)." width="700"><p class="caption">
MA plot for group differences adjusted by gene target for SRP009615
using ER-level data (just chrY).
</p>
</div>
<p>We can also use <em><a href="https://bioconductor.org/packages/3.16/regionReport" class="external-link">regionReport</a></em>
to create a more extensive exploratory report.</p>
<div class="sourceCode" id="cb104"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Create the report</span></span>
<span><span class="va">report2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/regionReport/man/DESeq2Report.html" class="external-link">DESeq2Report</a></span><span class="op">(</span><span class="va">dds_ers</span>,</span>
<span>    res <span class="op">=</span> <span class="va">res_ers</span>,</span>
<span>    project <span class="op">=</span> <span class="st">"SRP009615 (ER-level, chrY)"</span>,</span>
<span>    intgroup <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"group"</span>, <span class="st">"gene_target"</span><span class="op">)</span>, outdir <span class="op">=</span> <span class="st">"."</span>,</span>
<span>    output <span class="op">=</span> <span class="st">"SRP009615-results-ER-level-chrY"</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="annotation-used">Annotation used<a class="anchor" aria-label="anchor" href="#annotation-used"></a>
</h2>
<p>This section describes the annotation used in
<code>recount</code>.</p>
<p>We used the comprehensive gene annotation for (<code>CHR</code>
regions) from <a href="https://www.gencodegenes.org/releases/25.html" class="external-link">Gencode v25</a>
(GRCh38.p7) to get the list of genes. Specifically <a href="ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_25/gencode.v25.annotation.gff3.gz">this
GFF3</a> file. We then used the <em><a href="https://bioconductor.org/packages/3.16/org.Hs.eg.db" class="external-link">org.Hs.eg.db</a></em>
package to get the gene symbols. For each gene, we also counted the
total length of exonic base pairs for that given gene and stored this
information in the <code>bp_length</code> column. You can see below this
information for the <code>rse_gene</code> object included in <em><a href="https://bioconductor.org/packages/3.16/recount" class="external-link">recount</a></em>.
The <code><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames()</a></code> are the Gencode <code>gene_id</code>.</p>
<div class="sourceCode" id="cb105"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Gene annotation information</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/MatrixGenerics/man/rowRanges.html" class="external-link">rowRanges</a></span><span class="op">(</span><span class="va">rse_gene_SRP009615</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## GRanges object with 58037 ranges and 3 metadata columns:</span></span>
<span><span class="co">##                      seqnames              ranges strand |            gene_id bp_length          symbol</span></span>
<span><span class="co">##                         &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; |        &lt;character&gt; &lt;integer&gt; &lt;CharacterList&gt;</span></span>
<span><span class="co">##   ENSG00000000003.14     chrX 100627109-100639991      - | ENSG00000000003.14      4535          TSPAN6</span></span>
<span><span class="co">##    ENSG00000000005.5     chrX 100584802-100599885      + |  ENSG00000000005.5      1610            TNMD</span></span>
<span><span class="co">##   ENSG00000000419.12    chr20   50934867-50958555      - | ENSG00000000419.12      1207            DPM1</span></span>
<span><span class="co">##   ENSG00000000457.13     chr1 169849631-169894267      - | ENSG00000000457.13      6883           SCYL3</span></span>
<span><span class="co">##   ENSG00000000460.16     chr1 169662007-169854080      + | ENSG00000000460.16      5967        C1orf112</span></span>
<span><span class="co">##                  ...      ...                 ...    ... .                ...       ...             ...</span></span>
<span><span class="co">##    ENSG00000283695.1    chr19   52865369-52865429      - |  ENSG00000283695.1        61            &lt;NA&gt;</span></span>
<span><span class="co">##    ENSG00000283696.1     chr1 161399409-161422424      + |  ENSG00000283696.1       997            &lt;NA&gt;</span></span>
<span><span class="co">##    ENSG00000283697.1     chrX 149548210-149549852      - |  ENSG00000283697.1      1184           HSFX3</span></span>
<span><span class="co">##    ENSG00000283698.1     chr2 112439312-112469687      - |  ENSG00000283698.1       940            &lt;NA&gt;</span></span>
<span><span class="co">##    ENSG00000283699.1    chr10   12653138-12653197      - |  ENSG00000283699.1        60         MIR4481</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 25 sequences (1 circular) from an unspecified genome; no seqlengths</span></span></code></pre>
<div class="sourceCode" id="cb107"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Also accessible via</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">rse_gene_SRP009615</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## DataFrame with 58037 rows and 3 columns</span></span>
<span><span class="co">##                               gene_id bp_length          symbol</span></span>
<span><span class="co">##                           &lt;character&gt; &lt;integer&gt; &lt;CharacterList&gt;</span></span>
<span><span class="co">## ENSG00000000003.14 ENSG00000000003.14      4535          TSPAN6</span></span>
<span><span class="co">## ENSG00000000005.5   ENSG00000000005.5      1610            TNMD</span></span>
<span><span class="co">## ENSG00000000419.12 ENSG00000000419.12      1207            DPM1</span></span>
<span><span class="co">## ENSG00000000457.13 ENSG00000000457.13      6883           SCYL3</span></span>
<span><span class="co">## ENSG00000000460.16 ENSG00000000460.16      5967        C1orf112</span></span>
<span><span class="co">## ...                               ...       ...             ...</span></span>
<span><span class="co">## ENSG00000283695.1   ENSG00000283695.1        61              NA</span></span>
<span><span class="co">## ENSG00000283696.1   ENSG00000283696.1       997              NA</span></span>
<span><span class="co">## ENSG00000283697.1   ENSG00000283697.1      1184           HSFX3</span></span>
<span><span class="co">## ENSG00000283698.1   ENSG00000283698.1       940              NA</span></span>
<span><span class="co">## ENSG00000283699.1   ENSG00000283699.1        60         MIR4481</span></span></code></pre>
<p>For an <code>rse_exon</code> object, the <code><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames()</a></code>
correspond to the Gencode <code>gene_id</code>. You can add a
<code>gene_id</code> column if you are interested in subsetting the
whole <code>rse_exon</code> object.</p>
<div class="sourceCode" id="cb109"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Get the rse_exon object for study SRP009615</span></span>
<span><span class="fu"><a href="../reference/download_study.html">download_study</a></span><span class="op">(</span><span class="st">"SRP009615"</span>, type <span class="op">=</span> <span class="st">"rse-exon"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 2023-05-07 01:08:45 downloading file rse_exon.Rdata to SRP009615</span></span></code></pre>
<div class="sourceCode" id="cb111"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="st">"SRP009615"</span>, <span class="st">"rse_exon.Rdata"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Delete it if you don't need it anymore</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/unlink.html" class="external-link">unlink</a></span><span class="op">(</span><span class="st">"SRP009615"</span>, recursive <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Annotation information</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/MatrixGenerics/man/rowRanges.html" class="external-link">rowRanges</a></span><span class="op">(</span><span class="va">rse_exon</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## GRanges object with 603139 ranges and 0 metadata columns:</span></span>
<span><span class="co">##                      seqnames              ranges strand</span></span>
<span><span class="co">##                         &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt;</span></span>
<span><span class="co">##   ENSG00000000003.14     chrX 100627109-100628669      -</span></span>
<span><span class="co">##   ENSG00000000003.14     chrX 100628670-100629986      -</span></span>
<span><span class="co">##   ENSG00000000003.14     chrX 100630759-100630866      -</span></span>
<span><span class="co">##   ENSG00000000003.14     chrX 100632063-100632068      -</span></span>
<span><span class="co">##   ENSG00000000003.14     chrX 100632485-100632540      -</span></span>
<span><span class="co">##                  ...      ...                 ...    ...</span></span>
<span><span class="co">##    ENSG00000283698.1     chr2 112440396-112440601      -</span></span>
<span><span class="co">##    ENSG00000283698.1     chr2 112440602-112440611      -</span></span>
<span><span class="co">##    ENSG00000283698.1     chr2 112463991-112464036      -</span></span>
<span><span class="co">##    ENSG00000283698.1     chr2 112469548-112469687      -</span></span>
<span><span class="co">##    ENSG00000283699.1    chr10   12653138-12653197      -</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 25 sequences (1 circular) from an unspecified genome; no seqlengths</span></span></code></pre>
<div class="sourceCode" id="cb113"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Add a gene_id column</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/MatrixGenerics/man/rowRanges.html" class="external-link">rowRanges</a></span><span class="op">(</span><span class="va">rse_exon</span><span class="op">)</span><span class="op">$</span><span class="va">gene_id</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">rse_exon</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Example subset</span></span>
<span><span class="va">rse_exon_subset</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">rse_exon</span>, subset <span class="op">=</span> <span class="va">gene_id</span> <span class="op">==</span> <span class="st">"ENSG00000000003.14"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/MatrixGenerics/man/rowRanges.html" class="external-link">rowRanges</a></span><span class="op">(</span><span class="va">rse_exon_subset</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## GRanges object with 17 ranges and 1 metadata column:</span></span>
<span><span class="co">##                      seqnames              ranges strand |            gene_id</span></span>
<span><span class="co">##                         &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; |        &lt;character&gt;</span></span>
<span><span class="co">##   ENSG00000000003.14     chrX 100627109-100628669      - | ENSG00000000003.14</span></span>
<span><span class="co">##   ENSG00000000003.14     chrX 100628670-100629986      - | ENSG00000000003.14</span></span>
<span><span class="co">##   ENSG00000000003.14     chrX 100630759-100630866      - | ENSG00000000003.14</span></span>
<span><span class="co">##   ENSG00000000003.14     chrX 100632063-100632068      - | ENSG00000000003.14</span></span>
<span><span class="co">##   ENSG00000000003.14     chrX 100632485-100632540      - | ENSG00000000003.14</span></span>
<span><span class="co">##                  ...      ...                 ...    ... .                ...</span></span>
<span><span class="co">##   ENSG00000000003.14     chrX 100636608-100636689      - | ENSG00000000003.14</span></span>
<span><span class="co">##   ENSG00000000003.14     chrX 100636690-100636792      - | ENSG00000000003.14</span></span>
<span><span class="co">##   ENSG00000000003.14     chrX 100636793-100636806      - | ENSG00000000003.14</span></span>
<span><span class="co">##   ENSG00000000003.14     chrX 100636807-100637104      - | ENSG00000000003.14</span></span>
<span><span class="co">##   ENSG00000000003.14     chrX 100639945-100639991      - | ENSG00000000003.14</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 25 sequences (1 circular) from an unspecified genome; no seqlengths</span></span></code></pre>
<p>The exons in a <code>rse_exon</code> object are those that have a
Gencode <code>gene_id</code> and that were disjointed as described in
the <em><a href="https://bioconductor.org/packages/3.16/GenomicRanges" class="external-link">GenomicRanges</a></em>
package:</p>
<pre><code><span><span class="op">?</span><span class="va">`inter-range-methods`</span></span></code></pre>
<p>You can reproduce the gene and exon information using the function
<code><a href="../reference/reproduce_ranges.html">reproduce_ranges()</a></code>.</p>
<div class="section level3">
<h3 id="using-anothernewer-annotation">Using another/newer annotation<a class="anchor" aria-label="anchor" href="#using-anothernewer-annotation"></a>
</h3>
<p>If you are interested in using another annotation based on hg38
coordinates you can do so using the function
<code><a href="../reference/coverage_matrix.html">coverage_matrix()</a></code> or by running <a href="https://github.com/CRG-Barcelona/bwtool/wiki" class="external-link">bwtool</a> thanks to
the BigWig files in <code>recount</code>, in which case <a href="https://github.com/LieberInstitute/recount.bwtool" class="external-link">recount.bwtool</a>
will be helpful.</p>
<p>If you are re-processing a small set of samples, it simply might be
easier to use <code><a href="../reference/coverage_matrix.html">coverage_matrix()</a></code> as shown below using the
annotation information provided by the <em><a href="https://bioconductor.org/packages/3.16/EnsDb.Hsapiens.v79" class="external-link">EnsDb.Hsapiens.v79</a></em>
package.</p>
<div class="sourceCode" id="cb116"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Get the disjoint exons based on EnsDb.Hsapiens.v79 which matches hg38</span></span>
<span><span class="va">exons</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/reproduce_ranges.html">reproduce_ranges</a></span><span class="op">(</span><span class="st">"exon"</span>, db <span class="op">=</span> <span class="st">"EnsDb.Hsapiens.v79"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Change the chromosome names to match those used in the BigWig files</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.org/packages/GenomeInfoDb" class="external-link">"GenomeInfoDb"</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqlevelsStyle.html" class="external-link">seqlevelsStyle</a></span><span class="op">(</span><span class="va">exons</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"UCSC"</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning in (function (seqlevels, genome, new_style) : cannot switch some GRCh38's seqlevels from NCBI to UCSC style</span></span></code></pre>
<div class="sourceCode" id="cb118"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Get the count matrix at the exon level for chrY</span></span>
<span><span class="va">exons_chrY</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqlevels-wrappers.html" class="external-link">keepSeqlevels</a></span><span class="op">(</span><span class="va">exons</span>, <span class="st">"chrY"</span>, pruning.mode <span class="op">=</span> <span class="st">"coarse"</span><span class="op">)</span></span>
<span><span class="va">exonRSE</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/coverage_matrix.html">coverage_matrix</a></span><span class="op">(</span><span class="st">"SRP009615"</span>, <span class="st">"chrY"</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="va">exons_chrY</span><span class="op">)</span>,</span>
<span>    chunksize <span class="op">=</span> <span class="fl">3000</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 2023-05-07 01:09:05 railMatrix: processing regions 1 to 2562</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:09:05 railMatrix: processing file http://duffel.rail.bio/recount/SRP009615/bw/SRR387777.bw</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:09:07 railMatrix: processing file http://duffel.rail.bio/recount/SRP009615/bw/SRR387778.bw</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:09:10 railMatrix: processing file http://duffel.rail.bio/recount/SRP009615/bw/SRR387779.bw</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:09:13 railMatrix: processing file http://duffel.rail.bio/recount/SRP009615/bw/SRR387780.bw</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:09:15 railMatrix: processing file http://duffel.rail.bio/recount/SRP009615/bw/SRR389077.bw</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:09:18 railMatrix: processing file http://duffel.rail.bio/recount/SRP009615/bw/SRR389078.bw</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:09:21 railMatrix: processing file http://duffel.rail.bio/recount/SRP009615/bw/SRR389079.bw</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:09:24 railMatrix: processing file http://duffel.rail.bio/recount/SRP009615/bw/SRR389080.bw</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:09:26 railMatrix: processing file http://duffel.rail.bio/recount/SRP009615/bw/SRR389081.bw</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:09:29 railMatrix: processing file http://duffel.rail.bio/recount/SRP009615/bw/SRR389082.bw</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:09:32 railMatrix: processing file http://duffel.rail.bio/recount/SRP009615/bw/SRR389083.bw</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:09:34 railMatrix: processing file http://duffel.rail.bio/recount/SRP009615/bw/SRR389084.bw</span></span></code></pre>
<div class="sourceCode" id="cb132"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">exonMatrix</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">exonRSE</span><span class="op">)</span><span class="op">$</span><span class="va">counts</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">exonMatrix</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 2562   12</span></span></code></pre>
<div class="sourceCode" id="cb134"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">exonMatrix</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##   SRR387777 SRR387778 SRR387779 SRR387780 SRR389077 SRR389078 SRR389079 SRR389080 SRR389081 SRR389082 SRR389083</span></span>
<span><span class="co">## 1         0         0         0         0         0         0 0.0000000 0.0000000         0         0         0</span></span>
<span><span class="co">## 2         0         0         0         0         0         0 0.0000000 0.0000000         0         0         0</span></span>
<span><span class="co">## 3         0         0         0         0         0         0 0.0000000 0.0000000         0         0         0</span></span>
<span><span class="co">## 4         0         0         0         0         0         0 0.0000000 0.0000000         0         0         0</span></span>
<span><span class="co">## 5         0         0         0         0         0         0 0.0000000 0.6875641         0         0         0</span></span>
<span><span class="co">## 6         0         0         0         0         0         0 0.3256991 0.0000000         0         0         0</span></span>
<span><span class="co">##   SRR389084</span></span>
<span><span class="co">## 1         0</span></span>
<span><span class="co">## 2         0</span></span>
<span><span class="co">## 3         0</span></span>
<span><span class="co">## 4         0</span></span>
<span><span class="co">## 5         0</span></span>
<span><span class="co">## 6         0</span></span></code></pre>
<div class="sourceCode" id="cb136"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Summary the information at the gene level for chrY</span></span>
<span><span class="va">exon_gene</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">exons_chrY</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/List-class.html" class="external-link">elementNROWS</a></span><span class="op">(</span><span class="va">exons_chrY</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">geneMatrix</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/do.call.html" class="external-link">do.call</a></span><span class="op">(</span><span class="va">rbind</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/splitAsList.html" class="external-link">split</a></span><span class="op">(</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">exonMatrix</span><span class="op">)</span>,</span>
<span>    <span class="va">exon_gene</span></span>
<span><span class="op">)</span>, <span class="va">colSums</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">geneMatrix</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 544  12</span></span></code></pre>
<div class="sourceCode" id="cb138"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">geneMatrix</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                 SRR387777 SRR387778  SRR387779  SRR387780 SRR389077 SRR389078  SRR389079  SRR389080 SRR389081</span></span>
<span><span class="co">## ENSG00000012817  9.791214 17.655826 17.8671773  6.7090447  4.832332  2.893332 12.7255290  8.1132560 16.988920</span></span>
<span><span class="co">## ENSG00000067048 77.902314 69.981889 41.6900803 83.0430650 52.797699 52.000713 52.3677621 53.9737791 80.441512</span></span>
<span><span class="co">## ENSG00000067646 44.371293 52.123510 22.6257722 57.2505152 44.251631 44.232314 31.5928125 56.0708495 38.480927</span></span>
<span><span class="co">## ENSG00000092377 10.568294  9.722520 13.8867173 16.0271624 11.364928  1.426849 17.0061457 24.2022551 33.773154</span></span>
<span><span class="co">## ENSG00000099715  3.613424  1.789214  0.6284937  2.6836179  2.997835  0.000000  0.8375120  0.0000000  1.023429</span></span>
<span><span class="co">## ENSG00000099721  0.000000  0.000000  0.0000000  0.7827219  1.342314  0.000000  0.4652844  0.7563205  0.000000</span></span>
<span><span class="co">##                  SRR389082 SRR389083 SRR389084</span></span>
<span><span class="co">## ENSG00000012817  28.049724  8.830204  9.139425</span></span>
<span><span class="co">## ENSG00000067048 106.485062 65.978494 34.679041</span></span>
<span><span class="co">## ENSG00000067646  45.905450 35.469641 28.992288</span></span>
<span><span class="co">## ENSG00000092377   9.349908  3.571768 16.450965</span></span>
<span><span class="co">## ENSG00000099715   2.337477  0.000000  0.000000</span></span>
<span><span class="co">## ENSG00000099721   0.000000  0.000000  0.000000</span></span></code></pre>
<p>The above solution works well for a small set of samples. However,
<code>bwtool</code> is faster for processing large sets. You can see how
we used <code>bwtool</code> at <a href="https://github.com/leekgroup/recount-website" class="external-link">leekgroup/recount-website</a>.
It’s much faster than R and the small package <a href="https://github.com/LieberInstitute/recount.bwtool" class="external-link">recount.bwtool</a>
will be helpful for such scenarios. Note that you will need to run
<code><a href="../reference/scale_counts.html">scale_counts()</a></code> after using
<code>coverage_matrix_bwtool()</code> and that will have to download the
BigWig files first unless you are working at <a href="https://jhpce.jhu.edu/" class="external-link">JHPCE</a> or <a href="http://www.sciserver.org/" class="external-link">SciServer</a>.
<code>coverage_matrix_bwtool()</code> will download the files for you,
but that till take time.</p>
<p>Finally, the BigWig files hosted by recount have the following
chromosome names and sizes as specified in <a href="https://github.com/nellore/runs/blob/master/gtex/hg38.sizes" class="external-link">hg38.sizes</a>.
You’ll have to make sure that you match these names when using
alternative annotations.</p>
</div>
</div>
<div class="section level2">
<h2 id="candidate-gene-fusions">Candidate gene fusions<a class="anchor" aria-label="anchor" href="#candidate-gene-fusions"></a>
</h2>
<p>If you are interested in finding possible gene fusion events in a
particular study, you can download the
<em>RangedSummarizedExperiment</em> object at the exon-exon junctions
level for that study. The objects we provide classify exon-exon junction
events as shown below.</p>
<div class="sourceCode" id="cb140"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/leekgroup/recount" class="external-link">"recount"</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Download and load RSE object for SRP009615 study</span></span>
<span><span class="fu"><a href="../reference/download_study.html">download_study</a></span><span class="op">(</span><span class="st">"SRP009615"</span>, type <span class="op">=</span> <span class="st">"rse-jx"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 2023-05-07 01:09:37 downloading file rse_jx.Rdata to SRP009615</span></span></code></pre>
<div class="sourceCode" id="cb142"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="st">"SRP009615"</span>, <span class="st">"rse_jx.Rdata"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Delete it if you don't need it anymore</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/unlink.html" class="external-link">unlink</a></span><span class="op">(</span><span class="st">"SRP009615"</span>, recursive <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Exon-exon junctions by class</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/MatrixGenerics/man/rowRanges.html" class="external-link">rowRanges</a></span><span class="op">(</span><span class="va">rse_jx</span><span class="op">)</span><span class="op">$</span><span class="va">class</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## alternative_end       annotated       exon_skip          fusion           novel </span></span>
<span><span class="co">##           35601          140251            9513             489           26475</span></span></code></pre>
<div class="sourceCode" id="cb144"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Potential gene fusions by chromosome</span></span>
<span><span class="va">fusions</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MatrixGenerics/man/rowRanges.html" class="external-link">rowRanges</a></span><span class="op">(</span><span class="va">rse_jx</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/MatrixGenerics/man/rowRanges.html" class="external-link">rowRanges</a></span><span class="op">(</span><span class="va">rse_jx</span><span class="op">)</span><span class="op">$</span><span class="va">class</span> <span class="op">==</span> <span class="st">"fusion"</span><span class="op">]</span></span>
<span><span class="va">fusions_by_chr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">fusions</span><span class="op">)</span><span class="op">)</span>, decreasing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">fusions_by_chr</span><span class="op">[</span><span class="va">fusions_by_chr</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">##  chr1 chr19  chr6 chr12 chr22 chr17  chr5 chr16 chr11  chr3 chr10  chr2  chr7  chrX chr14  chr4  chr9  chr8 chr15 chr20 </span></span>
<span><span class="co">##    61    54    37    31    27    26    26    24    22    22    21    20    20    16    13    13    12    10     9     9 </span></span>
<span><span class="co">## chr13 chr21 chr18 </span></span>
<span><span class="co">##     6     6     4</span></span></code></pre>
<div class="sourceCode" id="cb146"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Genes with the most fusions</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="va">fusions</span><span class="op">$</span><span class="va">symbol_proposed</span><span class="op">)</span><span class="op">)</span>, decreasing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">##      TMEM14B       SYCP2L LOC100129924        MATR3        ACSM3         BMI1 </span></span>
<span><span class="co">##            6            5            4            4            3            3</span></span></code></pre>
<p>If you are interested in checking the frequency of a particular
exon-exon junction then check the <code><a href="../reference/snaptron_query.html">snaptron_query()</a></code> function
described in the following section.</p>
</div>
<div class="section level2">
<h2 id="snaptron">Snaptron<a class="anchor" aria-label="anchor" href="#snaptron"></a>
</h2>
<p>Our collaborators built <em>SnaptronUI</em> to query the <a href="https://github.com/nellore/intropolis" class="external-link">Intropolis</a> database of
the exon-exon junctions found in SRA. <em>SnaptronUI</em> allows you to
do several types of queries manually and perform analyses. <em><a href="https://bioconductor.org/packages/3.16/recount" class="external-link">recount</a></em>
has the function <code><a href="../reference/snaptron_query.html">snaptron_query()</a></code> which uses <a href="https://github.com/ChristopherWilks/snaptron" class="external-link">Snaptron</a> to
check if particular exon-exon junctions are present in
<em>Intropolis</em>. For example, here we use
<code><a href="../reference/snaptron_query.html">snaptron_query()</a></code> to get in <code>R</code>-friendly format
the information for 3 exon-exon junctions using <em>Intropolis</em>
version 1 which uses hg19 coordinates. Version 2 uses hg38 coordinates
and the exon-exon junctions were derived from a larger data set: about
50,000 samples vs 21,000 in version 1. <code><a href="../reference/snaptron_query.html">snaptron_query()</a></code> can
also be used to access the 30 million and 36 million exon-exon junctions
derived from the GTEx and TCGA consortiums respectively (about 10 and 11
thousand samples respectively).</p>
<div class="sourceCode" id="cb148"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.org/packages/GenomicRanges" class="external-link">"GenomicRanges"</a></span><span class="op">)</span></span>
<span><span class="va">junctions</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span>seqnames <span class="op">=</span> <span class="st">"chr2"</span>, <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html" class="external-link">IRanges</a></span><span class="op">(</span></span>
<span>    start <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">28971711</span>, <span class="fl">29555082</span>, <span class="fl">29754983</span><span class="op">)</span>,</span>
<span>    end <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">29462418</span>, <span class="fl">29923339</span>, <span class="fl">29917715</span><span class="op">)</span></span>
<span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/snaptron_query.html">snaptron_query</a></span><span class="op">(</span><span class="va">junctions</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 2023-05-07 01:09:41 querying Snaptron</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:09:42 processing results</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:09:42 found no exon-exon junctions in Intropolis version srav1 matching your query: this version uses hg19 coordinates.</span></span></code></pre>
<pre><code><span><span class="co">## NULL</span></span></code></pre>
<p>If you use <code><a href="../reference/snaptron_query.html">snaptron_query()</a></code> please cite <a href="http://snaptron.cs.jhu.edu" class="external-link">snaptron.cs.jhu.edu</a>.
<em>Snaptron</em> is separate from <em><a href="https://bioconductor.org/packages/3.16/recount" class="external-link">recount</a></em>.
Thank you!</p>
</div>
<div class="section level2">
<h2 id="fantom-cat-annotation">FANTOM-CAT annotation<a class="anchor" aria-label="anchor" href="#fantom-cat-annotation"></a>
</h2>
<p>Thanks to <a href="https://www.biorxiv.org/content/10.1101/659490v1" class="external-link">Imada, Sanchez,
et al., bioRxiv, 2019</a> you can now use the <code>recount</code>
package to access the FANTOM-CAT annotation that was merged with
<code>recount2</code>. You can use the main <code>recount2</code>
website or the <code><a href="../reference/download_study.html">recount::download_study()</a></code> function by
specifying <code>type = 'rse-fc'</code>. See their manuscript for
further details on how this annotation was derived.</p>
<div class="sourceCode" id="cb153"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Example use of download_study()</span></span>
<span><span class="co">## for downloading the FANTOM-CAT `rse_fc`</span></span>
<span><span class="co">## file for a given study</span></span>
<span><span class="fu"><a href="../reference/download_study.html">download_study</a></span><span class="op">(</span><span class="st">"DRP000366"</span>, type <span class="op">=</span> <span class="st">"rse-fc"</span>, download <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="recount-brain">
<code>recount-brain</code><a class="anchor" aria-label="anchor" href="#recount-brain"></a>
</h2>
<p>The <code>recount-brain</code> project by <a href="https://www.biorxiv.org/content/10.1101/618025v1" class="external-link">Ramzara et al.,
bioRxiv, 2019</a> contains curated metadata for brain studies present in
<code>recount2</code> that you can access via
<code>recount::add_metadata(source = 'recount_brain_v2')</code>. The
manuscript by Razmara et al describes a curation process that can be
used for curating other tissues or projects. If you are interested in
doing such a project and/or contributing curated metadata to the
<code>recount</code> package via the <code><a href="../reference/add_metadata.html">add_metadata()</a></code>
function please let us know.</p>
<div class="sourceCode" id="cb154"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## recount-brain citation details</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/citation.html" class="external-link">citation</a></span><span class="op">(</span><span class="st">"recount"</span><span class="op">)</span><span class="op">[</span><span class="fl">5</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Frazee AC, Langmead B, Leek JT (2011). "ReCount: A multi-experiment resource of analysis-ready RNA-seq gene</span></span>
<span><span class="co">## count datasets." _BMC Bioinformatics_. doi:10.1186/1471-2105-12-449</span></span>
<span><span class="co">## &lt;https://doi.org/10.1186/1471-2105-12-449&gt;, &lt;https://doi.org/10.1186/1471-2105-12-449&gt;.</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## A BibTeX entry for LaTeX users is</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##   @Article{,</span></span>
<span><span class="co">##     title = {ReCount: A multi-experiment resource of analysis-ready RNA-seq gene count datasets},</span></span>
<span><span class="co">##     author = {Alyssa C. Frazee and Ben Langmead and Jeffrey T. Leek},</span></span>
<span><span class="co">##     year = {2011},</span></span>
<span><span class="co">##     journal = {BMC Bioinformatics},</span></span>
<span><span class="co">##     doi = {10.1186/1471-2105-12-449},</span></span>
<span><span class="co">##     url = {https://doi.org/10.1186/1471-2105-12-449},</span></span>
<span><span class="co">##   }</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="download-all-the-data">Download all the data<a class="anchor" aria-label="anchor" href="#download-all-the-data"></a>
</h2>
<p>If you are interested in downloading all the data you can do so using
the <code>recount_url</code> data.frame included in the package. That
is:</p>
<div class="sourceCode" id="cb156"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Get data.frame with all the URLs</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/leekgroup/recount" class="external-link">"recount"</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">recount_url</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 97229     6</span></span></code></pre>
<div class="sourceCode" id="cb158"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Explore URLs</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">recount_url</span><span class="op">$</span><span class="va">url</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "http://duffel.rail.bio/recount/v2/DRP000366/counts_exon.tsv.gz"</span></span>
<span><span class="co">## [2] "http://duffel.rail.bio/recount/v2/DRP000366/counts_gene.tsv.gz"</span></span>
<span><span class="co">## [3] "http://duffel.rail.bio/recount/v2/DRP000366/rse_exon.Rdata"    </span></span>
<span><span class="co">## [4] "http://duffel.rail.bio/recount/v2/DRP000366/rse_gene.Rdata"    </span></span>
<span><span class="co">## [5] "http://duffel.rail.bio/recount/DRP000366/counts_jx.tsv.gz"     </span></span>
<span><span class="co">## [6] "http://duffel.rail.bio/recount/DRP000366/rse_jx.Rdata"</span></span></code></pre>
<p>You can then download all the data using the
<code><a href="../reference/download_study.html">download_study()</a></code> function or however you prefer to do so
with <code>recount_url</code>.</p>
<div class="sourceCode" id="cb160"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Download all files (will take a while! It's over 8 TB of data)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">recount_url</span><span class="op">$</span><span class="va">project</span><span class="op">)</span>, <span class="va">download_study</span>, type <span class="op">=</span> <span class="st">"all"</span><span class="op">)</span></span></code></pre></div>
<p>You can find the code for the website at <a href="https://github.com/leekgroup/recount-website/tree/master/website" class="external-link">leekgroup/recount-website</a>
if you want to deploy your own local mirror. Please let us know if you
choose to do deploy a mirror.</p>
</div>
<div class="section level2">
<h2 id="accessing-recount-via-sciserver">Accessing <em><a href="https://bioconductor.org/packages/3.16/recount" class="external-link">recount</a></em>
via <em>SciServer</em><a class="anchor" aria-label="anchor" href="#accessing-recount-via-sciserver"></a>
</h2>
<p>Not everyone has over 8 terabytes of disk space available to download
all the data from the <code>recount</code> project. However, thanks to
<a href="http://www.sciserver.org/" class="external-link">SciServer</a> you can access it
locally via a Jupyter Notebook. If you do so and want to share your
work, please let the <a href="http://www.sciserver.org/" class="external-link">SciServer</a>
maintainers know via Twitter at <a href="https://twitter.com/IDIESJHU" class="external-link">IDIESJHU</a>.</p>
<div class="section level3">
<h3 id="step-by-step">Step by step<a class="anchor" aria-label="anchor" href="#step-by-step"></a>
</h3>
<p>To use <em>SciServer</em>, you first have to go to <a href="http://www.sciserver.org/" class="external-link uri">http://www.sciserver.org/</a> then click on “Login to
SciServer”. If you are a first time user, click on “Register New
Account” and enter the information they request as shown on Figure
@ref(fig:Figure1).</p>
<div class="figure" style="text-align: center">
<img src="https://raw.githubusercontent.com/leekgroup/recount-website/master/sciserver_demo/register.png" alt="SIgn up to SciServer" width="100%"><p class="caption">
SIgn up to SciServer
</p>
</div>
<p>Once you’ve registered, open the <a href="http://compute.sciserver.org/dashboard/" class="external-link">SciServer compute
tool</a> (see Figure @ref(fig:Figure2)).</p>
<div class="figure" style="text-align: center">
<img src="https://raw.githubusercontent.com/leekgroup/recount-website/master/sciserver_demo/sciserver-compute.png" alt="Click on SciServer Compute," width="100%"><p class="caption">
Click on SciServer Compute,
</p>
</div>
<p>Then click on “Create container”, choose a name of your preference
and make sure that you choose to load R 3.3.x and the
<code>recount</code> public volume (see Figure @ref(fig:Figure3)). This
will enable you to access all of the <code>recount</code> data as if it
was on your computer. Click “Create” once you are ready.</p>
<div class="figure" style="text-align: center">
<img src="https://raw.githubusercontent.com/leekgroup/recount-website/master/sciserver_demo/new-container.png" alt="Select the appropriate image and load the recount public volume." width="100%"><p class="caption">
Select the appropriate image and load the recount public volume.
</p>
</div>
<p>Once you have a container, create a new R <em>Notebook</em> as shown
in Figure @ref(fig:Figure4).</p>
<div class="figure" style="text-align: center">
<img src="https://raw.githubusercontent.com/leekgroup/recount-website/master/sciserver_demo/new-R.png" alt="Create a R notebook." width="100%"><p class="caption">
Create a R notebook.
</p>
</div>
<p>In this R <em>Notebook</em> you can insert new code <em>cells</em>
where you can type R code. For example, you can install <em><a href="https://bioconductor.org/packages/3.16/recount" class="external-link">recount</a></em>
and <em><a href="https://bioconductor.org/packages/3.16/DESeq2" class="external-link">DESeq2</a></em> to
run the code example from this vignette. You first have to install the
packages as shown in Figure @ref(fig:Figure5).</p>
<div class="figure" style="text-align: center">
<img src="https://raw.githubusercontent.com/leekgroup/recount-website/master/sciserver_demo/demo1.png" alt="Install recount and DESeq2 on SciServer." width="100%"><p class="caption">
Install recount and DESeq2 on SciServer.
</p>
</div>
<p>Then, the main part when using <em>SciServer</em> is to use to
remember that all the <code>recount</code> data is available locally. So
instead of using <code><a href="../reference/download_study.html">download_study()</a></code>, you can simply use
<code><a href="https://rdrr.io/r/base/load.html" class="external-link">load()</a></code> with the correct path as shown in Figure
@ref(fig:Figure6).</p>
<div class="figure" style="text-align: center">
<img src="https://raw.githubusercontent.com/leekgroup/recount-website/master/sciserver_demo/demo2.png" alt="recount files are available locally so remmeber to use the local data when working on SciServer." width="100%"><p class="caption">
recount files are available locally so remmeber to use the local data
when working on SciServer.
</p>
</div>
<p>Several functions in the <em><a href="https://bioconductor.org/packages/3.16/recount" class="external-link">recount</a></em>
package have the <code>outdir</code> argument, which you can specify to
use the data locally. Try using <code><a href="../reference/expressed_regions.html">expressed_regions()</a></code> in
<em>SciServer</em> to get started with the following code:</p>
<div class="sourceCode" id="cb161"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Expressed-regions SciServer example</span></span>
<span><span class="va">regions</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/expressed_regions.html">expressed_regions</a></span><span class="op">(</span><span class="st">"SRP009615"</span>, <span class="st">"chrY"</span>,</span>
<span>    cutoff <span class="op">=</span> <span class="fl">5L</span>,</span>
<span>    maxClusterGap <span class="op">=</span> <span class="fl">3000L</span>, outdir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">scipath</span>, <span class="st">"SRP009615"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="va">regions</span></span></code></pre></div>
<p>You can find the R code for a full <em>SciServer</em> demo <a href="https://github.com/leekgroup/recount-website/blob/master/sciserver_demo/sciserver_demo.R" class="external-link">here</a>.
You can copy and paste it into a cell and run it all to see how
<em>SciServer compute</em> works.</p>
</div>
<div class="section level3">
<h3 id="sciserver-help">SciServer help<a class="anchor" aria-label="anchor" href="#sciserver-help"></a>
</h3>
<p>If you encounter problems when using SciServer please check their <a href="http://www.sciserver.org/support/help/" class="external-link">support page</a> and the
<a href="http://compute.sciserver.org/dashboard/Home/Help" class="external-link">SciServer
compute help section</a>.</p>
</div>
<div class="section level3">
<h3 id="citing-sciserver">Citing <em>SciServer</em><a class="anchor" aria-label="anchor" href="#citing-sciserver"></a>
</h3>
<p>If you use <a href="http://www.sciserver.org/" class="external-link">SciServer</a> for your
published work, please cite it accordingly. <em>SciServer</em> is
administered by the <a href="http://idies.jhu.edu/" class="external-link">Institute for Data
Intensive Engineering and Science</a> at <a href="http://www.jhu.edu/" class="external-link">Johns Hopkins University</a> and is funded by
National Science Foundation award ACI-1261715.</p>
</div>
</div>
<div class="section level2">
<h2 id="reproducibility">Reproducibility<a class="anchor" aria-label="anchor" href="#reproducibility"></a>
</h2>
<p>The <em><a href="https://bioconductor.org/packages/3.16/recount" class="external-link">recount</a></em>
package (Collado-Torres, Nellore, Kammers et al., 2017) was made
possible thanks to:</p>
<ul>
<li>R (R Core Team, 2023)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.16/AnnotationDbi" class="external-link">AnnotationDbi</a></em>
(Pagès, Carlson, Falcon, and Li, 2017)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.16/BiocParallel" class="external-link">BiocParallel</a></em>
(Morgan, Wang, Obenchain, Lang, Thompson, and Turaga, 2023)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.16/BiocStyle" class="external-link">BiocStyle</a></em>
(Oleś, 2022)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.16/derfinder" class="external-link">derfinder</a></em>
(Collado-Torres, Nellore, Frazee et al., 2017)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=sessioninfo" class="external-link">sessioninfo</a></em>
(Wickham, Chang, Flight, Müller, and Hester, 2021)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=downloader" class="external-link">downloader</a></em>
(Chang, 2015)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.16/EnsDb.Hsapiens.v79" class="external-link">EnsDb.Hsapiens.v79</a></em>
(Rainer, 2017)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.16/GEOquery" class="external-link">GEOquery</a></em>
(Davis and Meltzer, 2007)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.16/GenomeInfoDb" class="external-link">GenomeInfoDb</a></em>
(Arora, Morgan, Carlson, and Pagès, 2017)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.16/GenomicFeatures" class="external-link">GenomicFeatures</a></em>
(Lawrence, Huber, Pagès, Aboyoun, Carlson, Gentleman, Morgan, and Carey,
2013)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.16/GenomicRanges" class="external-link">GenomicRanges</a></em>
(Lawrence, Huber, Pagès et al., 2013)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=knitr" class="external-link">knitr</a></em> (Xie,
2014)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.16/org.Hs.eg.db" class="external-link">org.Hs.eg.db</a></em>
(Carlson, 2022)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=RCurl" class="external-link">RCurl</a></em> (Temple
Lang, 2023)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=rentrez" class="external-link">rentrez</a></em>
(Winter, 2017)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=RefManageR" class="external-link">RefManageR</a></em>
(McLean, 2017)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.16/rtracklayer" class="external-link">rtracklayer</a></em>
(Lawrence, Gentleman, and Carey, 2009)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=rmarkdown" class="external-link">rmarkdown</a></em>
(Allaire, Xie, Dervieux et al., 2023)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.16/S4Vectors" class="external-link">S4Vectors</a></em>
(Pagès, Lawrence, and Aboyoun, 2017)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.16/SummarizedExperiment" class="external-link">SummarizedExperiment</a></em>
(Morgan, Obenchain, Hester et al., 2017)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=testthat" class="external-link">testthat</a></em>
(Wickham, 2011)</li>
</ul>
<p>Code for creating the vignette</p>
<div class="sourceCode" id="cb162"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Create the vignette</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/rstudio/rmarkdown" class="external-link">"rmarkdown"</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/system.time.html" class="external-link">system.time</a></span><span class="op">(</span><span class="fu"><a href="https://pkgs.rstudio.com/rmarkdown/reference/render.html" class="external-link">render</a></span><span class="op">(</span><span class="st">"recount-quickstart.Rmd"</span>, <span class="st">"BiocStyle::html_document"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Extract the R code</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://yihui.org/knitr/" class="external-link">"knitr"</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/knit.html" class="external-link">knit</a></span><span class="op">(</span><span class="st">"recount-quickstart.Rmd"</span>, tangle <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p>Date the vignette was generated.</p>
<pre><code><span><span class="co">## [1] "2023-05-07 01:09:43 UTC"</span></span></code></pre>
<p>Wallclock time spent generating the vignette.</p>
<pre><code><span><span class="co">## Time difference of 2.826 mins</span></span></code></pre>
<p><code>R</code> session information.</p>
<pre><code><span><span class="co">## <span style="color: #00BBBB; font-weight: bold;">─ Session info ───────────────────────────────────────────────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">##  <span style="color: #555555; font-style: italic;">setting </span> <span style="color: #555555; font-style: italic;">value</span></span></span>
<span><span class="co">##  version  R version 4.2.3 (2023-03-15)</span></span>
<span><span class="co">##  os       Ubuntu 22.04.2 LTS</span></span>
<span><span class="co">##  system   x86_64, linux-gnu</span></span>
<span><span class="co">##  ui       X11</span></span>
<span><span class="co">##  language en</span></span>
<span><span class="co">##  collate  en_US.UTF-8</span></span>
<span><span class="co">##  ctype    en_US.UTF-8</span></span>
<span><span class="co">##  tz       UTC</span></span>
<span><span class="co">##  date     2023-05-07</span></span>
<span><span class="co">##  pandoc   2.19.2 @ /usr/local/bin/ (via rmarkdown)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## <span style="color: #00BBBB; font-weight: bold;">─ Packages ───────────────────────────────────────────────────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">##  <span style="color: #555555; font-style: italic;">package             </span> <span style="color: #555555; font-style: italic;">*</span> <span style="color: #555555; font-style: italic;">version  </span> <span style="color: #555555; font-style: italic;">date (UTC)</span> <span style="color: #555555; font-style: italic;">lib</span> <span style="color: #555555; font-style: italic;">source</span></span></span>
<span><span class="co">##  annotate               1.76.0    <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  AnnotationDbi        * 1.60.2    <span style="color: #555555;">2023-03-10</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  AnnotationFilter       1.22.0    <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  backports              1.4.1     <span style="color: #555555;">2021-12-13</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">##  base64enc              0.1-3     <span style="color: #555555;">2015-07-28</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  bibtex                 0.5.1     <span style="color: #555555;">2023-01-26</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  Biobase              * 2.58.0    <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  BiocFileCache          2.6.1     <span style="color: #555555;">2023-02-17</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  BiocGenerics         * 0.44.0    <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  BiocIO                 1.8.0     <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  BiocManager            1.30.20   <span style="color: #555555;">2023-02-24</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.3)</span></span></span>
<span><span class="co">##  BiocParallel           1.32.6    <span style="color: #555555;">2023-03-17</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  BiocStyle            * 2.26.0    <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  biomaRt                2.54.1    <span style="color: #555555;">2023-03-20</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  Biostrings             2.66.0    <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  bit                    4.0.5     <span style="color: #555555;">2022-11-15</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">##  bit64                  4.0.5     <span style="color: #555555;">2020-08-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">##  bitops                 1.0-7     <span style="color: #555555;">2021-04-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">##  blob                   1.2.4     <span style="color: #555555;">2023-03-17</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  bookdown               0.33      <span style="color: #555555;">2023-03-06</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  BSgenome               1.66.3    <span style="color: #555555;">2023-02-16</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  bslib                  0.4.2     <span style="color: #555555;">2022-12-16</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  bumphunter             1.40.0    <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  cachem                 1.0.7     <span style="color: #555555;">2023-02-24</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  checkmate              2.1.0     <span style="color: #555555;">2022-04-21</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">##  cli                    3.6.1     <span style="color: #555555;">2023-03-23</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  cluster                2.1.4     <span style="color: #555555;">2022-08-22</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.2.3)</span></span></span>
<span><span class="co">##  codetools              0.2-19    <span style="color: #555555;">2023-02-01</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.2.3)</span></span></span>
<span><span class="co">##  colorspace             2.1-0     <span style="color: #555555;">2023-01-23</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  crayon                 1.5.2     <span style="color: #555555;">2022-09-29</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  crosstalk              1.2.0     <span style="color: #555555;">2021-11-04</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">##  curl                   5.0.0     <span style="color: #555555;">2023-01-12</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  data.table             1.14.8    <span style="color: #555555;">2023-02-17</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  DBI                    1.1.3     <span style="color: #555555;">2022-06-18</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">##  dbplyr                 2.3.2     <span style="color: #555555;">2023-03-21</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  DEFormats              1.26.0    <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  DelayedArray           0.24.0    <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  derfinder              1.32.0    <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  derfinderHelper        1.32.0    <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  desc                   1.4.2     <span style="color: #555555;">2022-09-08</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  DESeq2               * 1.38.3    <span style="color: #555555;">2023-01-19</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  digest                 0.6.31    <span style="color: #555555;">2022-12-11</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  doRNG                  1.8.6     <span style="color: #555555;">2023-01-16</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  downloader             0.4       <span style="color: #555555;">2015-07-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">##  dplyr                  1.1.1     <span style="color: #555555;">2023-03-22</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  DT                   * 0.27      <span style="color: #555555;">2023-01-17</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  edgeR                * 3.40.2    <span style="color: #555555;">2023-01-19</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  ellipsis               0.3.2     <span style="color: #555555;">2021-04-29</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  EnsDb.Hsapiens.v79     2.99.0    <span style="color: #555555;">2023-05-07</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  ensembldb              2.22.0    <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  evaluate               0.20      <span style="color: #555555;">2023-01-17</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  fansi                  1.0.4     <span style="color: #555555;">2023-01-22</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  farver                 2.1.1     <span style="color: #555555;">2022-07-06</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">##  fastmap                1.1.1     <span style="color: #555555;">2023-02-24</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  filelock               1.0.2     <span style="color: #555555;">2018-10-05</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">##  foreach                1.5.2     <span style="color: #555555;">2022-02-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">##  foreign                0.8-84    <span style="color: #555555;">2022-12-06</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.2.3)</span></span></span>
<span><span class="co">##  Formula                1.2-5     <span style="color: #555555;">2023-02-24</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  fs                     1.6.1     <span style="color: #555555;">2023-02-06</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  geneplotter            1.76.0    <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  generics               0.1.3     <span style="color: #555555;">2022-07-05</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">##  GenomeInfoDb         * 1.34.9    <span style="color: #555555;">2023-02-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  GenomeInfoDbData       1.2.9     <span style="color: #555555;">2022-12-06</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  GenomicAlignments      1.34.1    <span style="color: #555555;">2023-03-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  GenomicFeatures        1.50.4    <span style="color: #555555;">2023-01-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  GenomicFiles           1.34.0    <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  GenomicRanges        * 1.50.2    <span style="color: #555555;">2022-12-16</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  GEOquery               2.66.0    <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  ggplot2              * 3.4.2     <span style="color: #555555;">2023-04-03</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  glue                   1.6.2     <span style="color: #555555;">2022-02-24</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  gridExtra              2.3       <span style="color: #555555;">2017-09-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">##  gtable                 0.3.3     <span style="color: #555555;">2023-03-21</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  highr                  0.10      <span style="color: #555555;">2022-12-22</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  Hmisc                  5.0-1     <span style="color: #555555;">2023-03-08</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  hms                    1.1.3     <span style="color: #555555;">2023-03-21</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  htmlTable              2.4.1     <span style="color: #555555;">2022-07-07</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">##  htmltools              0.5.5     <span style="color: #555555;">2023-03-23</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  htmlwidgets            1.6.2     <span style="color: #555555;">2023-03-17</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  httr                   1.4.5     <span style="color: #555555;">2023-02-24</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  IRanges              * 2.32.0    <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  iterators              1.0.14    <span style="color: #555555;">2022-02-05</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">##  jquerylib              0.1.4     <span style="color: #555555;">2021-04-26</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  jsonlite               1.8.4     <span style="color: #555555;">2022-12-06</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  KEGGREST               1.38.0    <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  knitr                * 1.42      <span style="color: #555555;">2023-01-25</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  knitrBootstrap         1.0.2     <span style="color: #555555;">2018-05-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">##  labeling               0.4.2     <span style="color: #555555;">2020-10-20</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">##  lattice                0.21-8    <span style="color: #555555;">2023-04-05</span> <span style="color: #555555;">[3]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  lazyeval               0.2.2     <span style="color: #555555;">2019-03-15</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">##  lifecycle              1.0.3     <span style="color: #555555;">2022-10-07</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  limma                * 3.54.2    <span style="color: #555555;">2023-02-28</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  locfit                 1.5-9.7   <span style="color: #555555;">2023-01-02</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  lubridate              1.9.2     <span style="color: #555555;">2023-02-10</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  magrittr               2.0.3     <span style="color: #555555;">2022-03-30</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  markdown               1.6       <span style="color: #555555;">2023-04-07</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  Matrix                 1.5-4     <span style="color: #555555;">2023-04-04</span> <span style="color: #555555;">[3]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  MatrixGenerics       * 1.10.0    <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  matrixStats          * 0.63.0    <span style="color: #555555;">2022-11-18</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">##  memoise                2.0.1     <span style="color: #555555;">2021-11-26</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  munsell                0.5.0     <span style="color: #555555;">2018-06-12</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">##  nnet                   7.3-18    <span style="color: #555555;">2022-09-28</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.2.3)</span></span></span>
<span><span class="co">##  org.Hs.eg.db         * 3.16.0    <span style="color: #555555;">2022-12-06</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  pheatmap             * 1.0.12    <span style="color: #555555;">2019-01-04</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">##  pillar                 1.9.0     <span style="color: #555555;">2023-03-22</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  pkgconfig              2.0.3     <span style="color: #555555;">2019-09-22</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  pkgdown                2.0.7     <span style="color: #555555;">2022-12-14</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  plyr                   1.8.8     <span style="color: #555555;">2022-11-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">##  png                    0.1-8     <span style="color: #555555;">2022-11-29</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">##  prettyunits            1.1.1     <span style="color: #555555;">2020-01-24</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  progress               1.2.2     <span style="color: #555555;">2019-05-16</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">##  ProtGenerics           1.30.0    <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  purrr                  1.0.1     <span style="color: #555555;">2023-01-10</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  qvalue                 2.30.0    <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  R6                     2.5.1     <span style="color: #555555;">2021-08-19</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  ragg                   1.2.5     <span style="color: #555555;">2023-01-12</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  rappdirs               0.3.3     <span style="color: #555555;">2021-01-31</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  RColorBrewer         * 1.1-3     <span style="color: #555555;">2022-04-03</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">##  Rcpp                   1.0.10    <span style="color: #555555;">2023-01-22</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  RCurl                  1.98-1.12 <span style="color: #555555;">2023-03-27</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  readr                  2.1.4     <span style="color: #555555;">2023-02-10</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  recount              * 1.27.0    <span style="color: #555555;">2023-05-07</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  RefManageR           * 1.4.0     <span style="color: #555555;">2022-09-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">##  regionReport         * 1.32.0    <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  rentrez                1.2.3     <span style="color: #555555;">2020-11-10</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">##  reshape2               1.4.4     <span style="color: #555555;">2020-04-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">##  restfulr               0.0.15    <span style="color: #555555;">2022-06-16</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">##  rjson                  0.2.21    <span style="color: #555555;">2022-01-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">##  rlang                  1.1.0     <span style="color: #555555;">2023-03-14</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  rmarkdown              2.21      <span style="color: #555555;">2023-03-26</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  rngtools               1.5.2     <span style="color: #555555;">2021-09-20</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">##  rpart                  4.1.19    <span style="color: #555555;">2022-10-21</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.2.3)</span></span></span>
<span><span class="co">##  rprojroot              2.0.3     <span style="color: #555555;">2022-04-02</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  Rsamtools              2.14.0    <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  RSQLite                2.3.1     <span style="color: #555555;">2023-04-03</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  rstudioapi             0.14      <span style="color: #555555;">2022-08-22</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  rtracklayer            1.58.0    <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  S4Vectors            * 0.36.2    <span style="color: #555555;">2023-02-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  sass                   0.4.5     <span style="color: #555555;">2023-01-24</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  scales                 1.2.1     <span style="color: #555555;">2022-08-20</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">##  sessioninfo          * 1.2.2     <span style="color: #555555;">2021-12-06</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  stringi                1.7.12    <span style="color: #555555;">2023-01-11</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  stringr                1.5.0     <span style="color: #555555;">2022-12-02</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  SummarizedExperiment * 1.28.0    <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  systemfonts            1.0.4     <span style="color: #555555;">2022-02-11</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  textshaping            0.3.6     <span style="color: #555555;">2021-10-13</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  tibble                 3.2.1     <span style="color: #555555;">2023-03-20</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  tidyr                  1.3.0     <span style="color: #555555;">2023-01-24</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  tidyselect             1.2.0     <span style="color: #555555;">2022-10-10</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">##  timechange             0.2.0     <span style="color: #555555;">2023-01-11</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  tzdb                   0.3.0     <span style="color: #555555;">2022-03-28</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">##  utf8                   1.2.3     <span style="color: #555555;">2023-01-31</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  VariantAnnotation      1.44.1    <span style="color: #555555;">2023-02-15</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  vctrs                  0.6.2     <span style="color: #555555;">2023-04-19</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  withr                  2.5.0     <span style="color: #555555;">2022-03-03</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  xfun                   0.38      <span style="color: #555555;">2023-03-24</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  XML                    3.99-0.14 <span style="color: #555555;">2023-03-19</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  xml2                   1.3.3     <span style="color: #555555;">2021-11-30</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  xtable                 1.8-4     <span style="color: #555555;">2019-04-21</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  XVector                0.38.0    <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  yaml                   2.3.7     <span style="color: #555555;">2023-01-23</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">##  zlibbioc               1.44.0    <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">## </span></span>
<span><span class="co">## <span style="color: #555555;"> [1] /__w/_temp/Library</span></span></span>
<span><span class="co">## <span style="color: #555555;"> [2] /usr/local/lib/R/site-library</span></span></span>
<span><span class="co">## <span style="color: #555555;"> [3] /usr/local/lib/R/library</span></span></span>
<span><span class="co">## </span></span>
<span><span class="co">## <span style="color: #00BBBB; font-weight: bold;">──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────</span></span></span></code></pre>
</div>
<div class="section level2">
<h2 id="bibliography">Bibliography<a class="anchor" aria-label="anchor" href="#bibliography"></a>
</h2>
<p>This vignette was generated using <em><a href="https://bioconductor.org/packages/3.16/BiocStyle" class="external-link">BiocStyle</a></em>
(Oleś, 2022) with <em><a href="https://CRAN.R-project.org/package=knitr" class="external-link">knitr</a></em> (Xie,
2014) and <em><a href="https://CRAN.R-project.org/package=rmarkdown" class="external-link">rmarkdown</a></em>
(Allaire, Xie, Dervieux et al., 2023) running behind the scenes.</p>
<p>Citations made with <em><a href="https://CRAN.R-project.org/package=RefManageR" class="external-link">RefManageR</a></em>
(McLean, 2017).</p>
<p>
<a id="bib-allaire2023rmarkdown"></a><a href="#cite-allaire2023rmarkdown">[1]</a><cite>
J. Allaire, Y. Xie, C. Dervieux, et al. <em>rmarkdown: Dynamic Documents
for R</em>. R package version 2.21. 2023. URL:
<a href="https://github.com/rstudio/rmarkdown" class="external-link">https://github.com/rstudio/rmarkdown</a>.</cite>
</p>
<p>
<a id="bib-GenomeInfoDb"></a><a href="#cite-GenomeInfoDb">[2]</a><cite>
S. Arora, M. Morgan, M. Carlson, et al. <em>GenomeInfoDb: Utilities for
manipulating chromosome and other ‘seqname’ identifiers</em>. 2017. DOI:
<a href="https://doi.org/10.18129/B9.bioc.GenomeInfoDb" class="external-link">10.18129/B9.bioc.GenomeInfoDb</a>.</cite>
</p>
<p>
<a id="bib-carlson2022genome"></a><a href="#cite-carlson2022genome">[3]</a><cite>
M. Carlson. <em>org.Hs.eg.db: Genome wide annotation for Human</em>. R
package version 3.16.0. 2022.</cite>
</p>
<p>
<a id="bib-chang2015downloader"></a><a href="#cite-chang2015downloader">[4]</a><cite>
W. Chang. <em>downloader: Download Files over HTTP and HTTPS</em>. R
package version 0.4. 2015. URL:
<a href="https://CRAN.R-project.org/package=downloader" class="external-link">https://CRAN.R-project.org/package=downloader</a>.</cite>
</p>
<p>
<a id="bib-colladotorres2016regionreport"></a><a href="#cite-colladotorres2016regionreport">[5]</a><cite>
L. Collado-Torres, A. E. Jaffe, and J. T. Leek. “regionReport:
Interactive reports for region-level and feature-level genomic analyses
[version2; referees: 2 approved, 1 approved with reservations]”. In:
<em>F1000Research</em> 4 (2016), p. 105. DOI:
<a href="https://doi.org/10.12688/f1000research.6379.2" class="external-link">10.12688/f1000research.6379.2</a>.
URL:
<a href="http://f1000research.com/articles/4-105/v2" class="external-link">http://f1000research.com/articles/4-105/v2</a>.</cite>
</p>
<p>
<a id="bib-colladotorres2017flexible"></a><a href="#cite-colladotorres2017flexible">[6]</a><cite>
L. Collado-Torres, A. Nellore, A. C. Frazee, et al. “Flexible expressed
region analysis for RNA-seq with derfinder”. In: <em>Nucl. Acids
Res.</em> (2017). DOI:
<a href="https://doi.org/10.1093/nar/gkw852" class="external-link">10.1093/nar/gkw852</a>.
URL:
<a href="http://nar.oxfordjournals.org/content/early/2016/09/29/nar.gkw852" class="external-link">http://nar.oxfordjournals.org/content/early/2016/09/29/nar.gkw852</a>.</cite>
</p>
<p>
<a id="bib-colladotorres2017recount"></a><a href="#cite-colladotorres2017recount">[7]</a><cite>
L. Collado-Torres, A. Nellore, and A. E. Jaffe. “recount workflow:
Accessing over 70,000 human RNA-seq samples with Bioconductor [version
1; referees: 1 approved, 2 approved with reservations]”. In:
<em>F1000Research</em> (2017). DOI:
<a href="https://doi.org/10.12688/f1000research.12223.1" class="external-link">10.12688/f1000research.12223.1</a>.
URL:
<a href="https://f1000research.com/articles/6-1558/v1" class="external-link">https://f1000research.com/articles/6-1558/v1</a>.</cite>
</p>
<p>
<a id="bib-colladotorres2017reproducible"></a><a href="#cite-colladotorres2017reproducible">[8]</a><cite>
L. Collado-Torres, A. Nellore, K. Kammers, et al. “Reproducible RNA-seq
analysis using recount2”. In: <em>Nature Biotechnology</em> (2017). DOI:
<a href="https://doi.org/10.1038/nbt.3838" class="external-link">10.1038/nbt.3838</a>. URL:
<a href="http://www.nature.com/nbt/journal/v35/n4/full/nbt.3838.html" class="external-link">http://www.nature.com/nbt/journal/v35/n4/full/nbt.3838.html</a>.</cite>
</p>
<p>
<a id="bib-davis2007geoquery"></a><a href="#cite-davis2007geoquery">[9]</a><cite>
S. Davis and P. Meltzer. “GEOquery: a bridge between the Gene Expression
Omnibus (GEO) and BioConductor”. In: <em>Bioinformatics</em> 14 (2007),
pp. 1846–1847.</cite>
</p>
<p>
<a id="bib-ellis2018improving"></a><a href="#cite-ellis2018improving">[10]</a><cite>
S. E. Ellis, L. Collado-Torres, A. E. Jaffe, et al. “Improving the value
of public RNA-seq expression data by phenotype prediction”. In:
<em>Nucl. Acids Res.</em> (2018). DOI:
<a href="https://doi.org/10.1093/nar/gky102" class="external-link">10.1093/nar/gky102</a>.
URL:
<a href="https://doi.org/10.1093/nar/gky102" class="external-link">https://doi.org/10.1093/nar/gky102</a>.</cite>
</p>
<p>
<a id="bib-lawrence2009rtracklayer"></a><a href="#cite-lawrence2009rtracklayer">[11]</a><cite>
M. Lawrence, R. Gentleman, and V. Carey. “rtracklayer: an R package for
interfacing with genome browsers”. In: <em>Bioinformatics</em> 25
(2009), pp. 1841-1842. DOI:
<a href="https://doi.org/10.1093/bioinformatics/btp328" class="external-link">10.1093/bioinformatics/btp328</a>.
URL:
<a href="http://bioinformatics.oxfordjournals.org/content/25/14/1841.abstract" class="external-link">http://bioinformatics.oxfordjournals.org/content/25/14/1841.abstract</a>.</cite>
</p>
<p>
<a id="bib-lawrence2013software"></a><a href="#cite-lawrence2013software">[12]</a><cite>
M. Lawrence, W. Huber, H. Pagès, et al. “Software for Computing and
Annotating Genomic Ranges”. In: <em>PLoS Computational Biology</em> 9 (8
2013). DOI:
<a href="https://doi.org/10.1371/journal.pcbi.1003118" class="external-link">10.1371/journal.pcbi.1003118</a>.
URL:
<a href="http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003118%7D." class="external-link"></a><a href="http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003118" class="external-link uri">http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003118</a>}.
</cite></p>


<p>
<a id="bib-love2014moderated"></a><a href="#cite-love2014moderated">[13]</a><cite>
M. I. Love, W. Huber, and S. Anders. “Moderated estimation of fold
change and dispersion for RNA-seq data with DESeq2”. In: <em>Genome
Biology</em> 15 (12 2014), p. 550. DOI:
<a href="https://doi.org/10.1186/s13059-014-0550-8" class="external-link">10.1186/s13059-014-0550-8</a>.</cite>
</p>
<p>
<a id="bib-mclean2017refmanager"></a><a href="#cite-mclean2017refmanager">[14]</a><cite>
M. W. McLean. “RefManageR: Import and Manage BibTeX and BibLaTeX
References in R”. In: <em>The Journal of Open Source Software</em>
(2017). DOI:
<a href="https://doi.org/10.21105/joss.00338" class="external-link">10.21105/joss.00338</a>.</cite>
</p>
<p>
<a id="bib-SummarizedExperiment"></a><a href="#cite-SummarizedExperiment">[15]</a><cite>
M. Morgan, V. Obenchain, J. Hester, et al. <em>SummarizedExperiment:
SummarizedExperiment container</em>. 2017. DOI:
<a href="https://doi.org/10.18129/B9.bioc.SummarizedExperiment" class="external-link">10.18129/B9.bioc.SummarizedExperiment</a>.</cite>
</p>
<p>
<a id="bib-morgan2023biocparallel"></a><a href="#cite-morgan2023biocparallel">[16]</a><cite>
M. Morgan, J. Wang, V. Obenchain, et al. <em>BiocParallel: Bioconductor
facilities for parallel evaluation</em>. R package version 1.32.6. 2023.
URL:
<a href="https://github.com/Bioconductor/BiocParallel" class="external-link">https://github.com/Bioconductor/BiocParallel</a>.</cite>
</p>
<p>
<a id="bib-ole2022biocstyle"></a><a href="#cite-ole2022biocstyle">[17]</a><cite>
A. Oleś. <em>BiocStyle: Standard styles for vignettes and other
Bioconductor documents</em>. R package version 2.26.0. 2022. URL:
<a href="https://github.com/Bioconductor/BiocStyle" class="external-link">https://github.com/Bioconductor/BiocStyle</a>.</cite>
</p>
<p>
<a id="bib-AnnotationDbi"></a><a href="#cite-AnnotationDbi">[18]</a><cite>
H. Pagès, M. Carlson, S. Falcon, et al. <em>AnnotationDbi: Annotation
Database Interface</em>. 2017. DOI:
<a href="https://doi.org/10.18129/B9.bioc.AnnotationDbi" class="external-link">10.18129/B9.bioc.AnnotationDbi</a>.</cite>
</p>
<p>
<a id="bib-S4Vectors"></a><a href="#cite-S4Vectors">[19]</a><cite> H.
Pagès, M. Lawrence, and P. Aboyoun. <em>S4Vectors: S4 implementation of
vector-like and list-like objects</em>. 2017. DOI:
<a href="https://doi.org/10.18129/B9.bioc.S4Vectors" class="external-link">10.18129/B9.bioc.S4Vectors</a>.</cite>
</p>
<p>
<a id="bib-2023language"></a><a href="#cite-2023language">[20]</a><cite>
R Core Team. <em>R: A Language and Environment for Statistical
Computing</em>. R Foundation for Statistical Computing. Vienna, Austria,
2023. URL:
<a href="https://www.R-project.org/" class="external-link">https://www.R-project.org/</a>.</cite>
</p>
<p>
<a id="bib-rainer2017ensdb"></a><a href="#cite-rainer2017ensdb">[21]</a><cite>
J. Rainer. <em>EnsDb.Hsapiens.v79: Ensembl based annotation
package</em>. R package version 2.99.0. 2017.</cite>
</p>
<p>
<a id="bib-templelang2023rcurl"></a><a href="#cite-templelang2023rcurl">[22]</a><cite>
D. Temple Lang. <em>RCurl: General Network (HTTP/FTP/…) Client Interface
for R</em>. R package version 1.98-1.12. 2023.</cite>
</p>
<p>
<a id="bib-wickham2011testthat"></a><a href="#cite-wickham2011testthat">[23]</a><cite>
H. Wickham. “testthat: Get Started with Testing”. In: <em>The R
Journal</em> 3 (2011), pp. 5–10. URL:
<a href="https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf" class="external-link">https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf</a>.</cite>
</p>
<p>
<a id="bib-wickham2021sessioninfo"></a><a href="#cite-wickham2021sessioninfo">[24]</a><cite>
H. Wickham, W. Chang, R. Flight, et al. <em>sessioninfo: R Session
Information</em>. <a href="https://github.com/r-lib/sessioninfo#readme" class="external-link uri">https://github.com/r-lib/sessioninfo#readme</a>, <a href="https://r-lib.github.io/sessioninfo/" class="external-link uri">https://r-lib.github.io/sessioninfo/</a>. 2021.</cite>
</p>
<p>
<a id="bib-winter2017rentrez"></a><a href="#cite-winter2017rentrez">[25]</a><cite>
D. J. Winter. “rentrez: an R package for the NCBI eUtils API”. In:
<em>The R Journal</em> 9 (2 2017), pp. 520–526.</cite>
</p>
<p>
<a id="bib-xie2014knitr"></a><a href="#cite-xie2014knitr">[26]</a><cite>
Y. Xie. “knitr: A Comprehensive Tool for Reproducible Research in R”.
In: <em>Implementing Reproducible Computational Research</em>. Ed. by V.
Stodden, F. Leisch and R. D. Peng. ISBN 978-1466561595. Chapman and
Hall/CRC, 2014.</cite>
</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Leonardo Collado-Torres, Jeffrey T. Leek.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
