<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>recount quick start guide • recount</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="recount quick start guide">
<meta property="og:description" content="recount">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">recount</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.15.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/fakeVignettes/SRP009615-results.html">UNKNOWN TITLE</a>
    </li>
    <li>
      <a href="../articles/recount-quickstart.html">recount quick start guide</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/leekgroup/recount/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>recount quick start guide</h1>
                        <h4 class="author">Leonardo Collado-Torres</h4>
            <address class="author_afil">
      Lieber Institute for Brain Development, Johns Hopkins Medical Campus<br><a class="author_email" href="mailto:#"></a><a href="mailto:lcolladotor@gmail.com" class="email">lcolladotor@gmail.com</a>
      </address>
                  
            <h4 class="date">13 May 2020</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/leekgroup/recount/blob/master/vignettes/recount-quickstart.Rmd"><code>vignettes/recount-quickstart.Rmd</code></a></small>
      <div class="hidden name"><code>recount-quickstart.Rmd</code></div>

    </div>

    
    
<div id="basics" class="section level1">
<h1 class="hasAnchor">
<a href="#basics" class="anchor"></a>Basics</h1>
<div id="install-recount" class="section level2">
<h2 class="hasAnchor">
<a href="#install-recount" class="anchor"></a>Install <em><a href="https://bioconductor.org/packages/3.12/recount">recount</a></em>
</h2>
<p><code>R</code> is an open-source statistical environment which can be easily modified to enhance its functionality via packages. <em><a href="https://bioconductor.org/packages/3.12/recount">recount</a></em> is a <code>R</code> package available via the <a href="http://bioconductor/packages/recount">Bioconductor</a> repository for packages. <code>R</code> can be installed on any operating system from <a href="https://cran.r-project.org/">CRAN</a> after which you can install <em><a href="https://bioconductor.org/packages/3.12/recount">recount</a></em> by using the following commands in your <code>R</code> session:</p>
<div class="sourceCode" id="cb1"><html><body><pre class="r"><span class="kw">if</span> (!<span class="fu"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span>(<span class="st">"BiocManager"</span>, <span class="kw">quietly</span> <span class="kw">=</span> <span class="fl">TRUE</span>)) {
      <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span>(<span class="st">"BiocManager"</span>)
  }

<span class="kw pkg">BiocManager</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span>(<span class="st">"recount"</span>)

<span class="co">## Check that you have a valid Bioconductor installation</span>
<span class="kw pkg">BiocManager</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/valid.html">valid</a></span>()</pre></body></html></div>
</div>
<div id="required-knowledge" class="section level2">
<h2 class="hasAnchor">
<a href="#required-knowledge" class="anchor"></a>Required knowledge</h2>
<p><em><a href="https://bioconductor.org/packages/3.12/recount">recount</a></em> is based on many other packages and in particular in those that have implemented the infrastructure needed for dealing with RNA-seq data. That is, packages like <em><a href="https://bioconductor.org/packages/3.12/GenomicFeatures">GenomicFeatures</a></em> and <em><a href="https://bioconductor.org/packages/3.12/rtracklayer">rtracklayer</a></em> that allow you to import the data. A <em><a href="https://bioconductor.org/packages/3.12/recount">recount</a></em> user is not expected to deal with those packages directly but will need to be familiar with <em><a href="https://bioconductor.org/packages/3.12/SummarizedExperiment">SummarizedExperiment</a></em> to understand the results <em><a href="https://bioconductor.org/packages/3.12/recount">recount</a></em> generates. It might also prove to be highly beneficial to check the</p>
<ul>
<li>
<em><a href="https://bioconductor.org/packages/3.12/DESeq2">DESeq2</a></em> package for performing differential expression analysis with the <em>RangedSummarizedExperiment</em> objects,</li>
<li>
<em><a href="https://bioconductor.org/packages/3.12/DEFormats">DEFormats</a></em> package for switching the objects to those used by other differential expression packages such as <em><a href="https://bioconductor.org/packages/3.12/edgeR">edgeR</a></em>,</li>
<li>
<em><a href="https://bioconductor.org/packages/3.12/derfinder">derfinder</a></em> package for performing annotation-agnostic differential expression analysis.</li>
</ul>
<p>If you are asking yourself the question “Where do I start using Bioconductor?” you might be interested in <a href="http://lcolladotor.github.io/2014/10/16/startbioc/#.VkOKbq6rRuU">this blog post</a>.</p>
</div>
<div id="asking-for-help" class="section level2">
<h2 class="hasAnchor">
<a href="#asking-for-help" class="anchor"></a>Asking for help</h2>
<p>As package developers, we try to explain clearly how to use our packages and in which order to use the functions. But <code>R</code> and <code>Bioconductor</code> have a steep learning curve so it is critical to learn where to ask for help. The blog post quoted above mentions some but we would like to highlight the <a href="https://support.bioconductor.org/">Bioconductor support site</a> as the main resource for getting help: remember to use the <code>recount</code> tag and check <a href="https://support.bioconductor.org/t/recount/">the older posts</a>. Other alternatives are available such as creating GitHub issues and tweeting. However, please note that if you want to receive help you should adhere to the <a href="http://www.bioconductor.org/help/support/posting-guide/">posting guidelines</a>. It is particularly critical that you provide a small reproducible example and your session information so package developers can track down the source of the error.</p>
</div>
<div id="workflow-using-recount" class="section level2">
<h2 class="hasAnchor">
<a href="#workflow-using-recount" class="anchor"></a>Workflow using <em><a href="https://bioconductor.org/packages/3.12/recount">recount</a></em>
</h2>
<p>We have written a workflow on how to use <em><a href="https://bioconductor.org/packages/3.12/recount">recount</a></em> that explains more details on how to use this package with other Bioconductor packages as well as the details on the actual counts provided by <em><a href="https://bioconductor.org/packages/3.12/recount">recount</a></em>. Check it at <a href="https://f1000research.com/articles/6-1558/v1">f1000research.com/articles/6-1558/v1</a> or <a href="http://bioconductor.org/help/workflows/recountWorkflow/">bioconductor.org/help/workflows/recountWorkflow/</a> <a id="cite-Collado-Torres_2017"></a>(<a href="https://f1000research.com/articles/6-1558/v1">Collado-Torres, Nellore, and Jaffe, 2017</a>).</p>
</div>
<div id="citing-recount" class="section level2">
<h2 class="hasAnchor">
<a href="#citing-recount" class="anchor"></a>Citing <em><a href="https://bioconductor.org/packages/3.12/recount">recount</a></em>
</h2>
<p>We hope that <em><a href="https://bioconductor.org/packages/3.12/recount">recount</a></em> will be useful for your research. Please use the following information to cite the package and the overall approach. Thank you!</p>
<div class="sourceCode" id="cb2"><html><body><pre class="r"><span class="co">## Citation info</span>
<span class="fu"><a href="https://rdrr.io/r/utils/citation.html">citation</a></span>(<span class="st">"recount"</span>)</pre></body></html></div>
<pre><code>## 
## Collado-Torres L, Nellore A, Kammers K, Ellis SE, Taub MA, Hansen KD,
## Jaffe AE, Langmead B, Leek JT (2017). "Reproducible RNA-seq analysis
## using recount2." _Nature Biotechnology_. doi: 10.1038/nbt.3838 (URL:
## https://doi.org/10.1038/nbt.3838), &lt;URL:
## http://www.nature.com/nbt/journal/v35/n4/full/nbt.3838.html&gt;.
## 
## Collado-Torres L, Nellore A, Jaffe AE (2017). "recount workflow:
## Accessing over 70,000 human RNA-seq samples with Bioconductor [version
## 1; referees: 1 approved, 2 approved with reservations]."
## _F1000Research_. doi: 10.12688/f1000research.12223.1 (URL:
## https://doi.org/10.12688/f1000research.12223.1), &lt;URL:
## https://f1000research.com/articles/6-1558/v1&gt;.
## 
## Ellis SE, Collado-Torres L, Jaffe AE, Leek JT (2018). "Improving the
## value of public RNA-seq expression data by phenotype prediction."
## _Nucl. Acids Res._. doi: 10.1093/nar/gky102 (URL:
## https://doi.org/10.1093/nar/gky102), &lt;URL:
## https://doi.org/10.1093/nar/gky102&gt;.
## 
## Collado-Torres L, Nellore A, Kammers K, Ellis SE, Taub MA, Hansen KD,
## Jaffe AE, Langmead B, Leek JT (2020). _Explore and download data from
## the recount project_. doi: 10.18129/B9.bioc.recount (URL:
## https://doi.org/10.18129/B9.bioc.recount),
## https://github.com/leekgroup/recount - R package version 1.15.1, &lt;URL:
## http://www.bioconductor.org/packages/recount&gt;.
## 
## Frazee AC, Langmead B, Leek JT (2011). "ReCount: A multi-experiment
## resource of analysis-ready RNA-seq gene count datasets." _BMC
## Bioinformatics_. doi: 10.1186/1471-2105-12-449 (URL:
## https://doi.org/10.1186/1471-2105-12-449), &lt;URL:
## https://doi.org/10.1186/1471-2105-12-449&gt;.
## 
## Razmara A, Ellis SE, Sokolowski DJ, Davis S, Wilson MD, Leek JT, Jaffe
## AE, Collado-Torres L (2019). "recount-brain: a curated repository of
## human brain RNA-seq datasets metadata." _bioRxiv_. doi: 10.1101/618025
## (URL: https://doi.org/10.1101/618025), &lt;URL:
## https://doi.org/10.1101/618025&gt;.
## 
## Imada E, Sanchez DF, Collado-Torres L, Wilks C, Matam T, Dinalankara W,
## Stupnikov A, Lobo-Pereira F, Yip C, Yasuzawa K, Kondo N, Itoh M, Suzuki
## H, Kasukawa T, Hon CC, de Hoon MJ, Shin JW, Carninci P, Jaffe AE, Leek
## JT, Favorov A, Franco GR, Langmead B, Marchionni L (2020). "Recounting
## the FANTOM CAGE–Associated Transcriptome." _Genome Research_. doi:
## 10.1101/gr.254656.119 (URL: https://doi.org/10.1101/gr.254656.119),
## &lt;URL: https://doi.org/10.1101/gr.254656.119&gt;.
## 
## To see these entries in BibTeX format, use 'print(&lt;citation&gt;,
## bibtex=TRUE)', 'toBibtex(.)', or set
## 'options(citation.bibtex.max=999)'.</code></pre>
</div>
</div>
<div id="quick-start-to-using-to-recount" class="section level1">
<h1 class="hasAnchor">
<a href="#quick-start-to-using-to-recount" class="anchor"></a>Quick start to using to <em><a href="https://bioconductor.org/packages/3.12/recount">recount</a></em>
</h1>
<p>Main updates:</p>
<ul>
<li><strong>As of January 30, 2017 the annotation used for the exon and gene counts is Gencode v25.</strong></li>
<li><strong>As of January 12, 2018 transcripts counts are available via recount2 thanks to the work of <a href="https://www.biorxiv.org/content/10.1101/247346v2">Fu et al, bioRxiv, 2018</a>. Disjoint exon counts (version 2) were also released as described in detail in the <a href="https://jhubiostatistics.shinyapps.io/recount/">recount website</a> documentation tab.</strong></li>
<li><strong>As of April 29, 2019 FANTOM-CAT/recount2 annotation quantifications area available via recount2 thanks to the work by <a href="https://www.biorxiv.org/content/10.1101/659490v1">Imada, Sanchez, et al., bioRxiv, 2019</a>.</strong></li>
</ul>
<p><a href="https://jhubiostatistics.shinyapps.io/recount/"><img src="https://raw.githubusercontent.com/leekgroup/recount-website/master/website/recount.png" align="center" width="600"></a></p>
<p>Here is a very quick example of how to download a <code>RangedSummarizedExperiment</code> object with the gene counts for a 2 groups project (12 samples) with SRA study id <a href="http://trace.ncbi.nlm.nih.gov/Traces/sra/?study=SRP009615">SRP009615</a> using the <em><a href="https://bioconductor.org/packages/3.12/recount">recount</a></em> package <a id="cite-Collado-Torres_2017a"></a>(<a href="http://www.nature.com/nbt/journal/v35/n4/full/nbt.3838.html">Collado-Torres, Nellore, Kammers, Ellis, et al., 2017</a>). The <code>RangedSummarizedExperiment</code> object is defined in the <em><a href="https://bioconductor.org/packages/3.12/SummarizedExperiment">SummarizedExperiment</a></em> <a id="cite-SummarizedExperiment"></a>(<a href="https://doi.org/10.18129/B9.bioc.SummarizedExperiment">Morgan, Obenchain, Hester, and Pagès, 2017</a>) package and can be used for differential expression analysis with different packages. Here we show how to use <em><a href="https://bioconductor.org/packages/3.12/DESeq2">DESeq2</a></em> <a id="cite-Love_2014"></a>(<a href="https://doi.org/10.1186/s13059-014-0550-8">Love, Huber, and Anders, 2014</a>) to perform the differential expresion analysis.</p>
<p>This quick analysis is explained in more detail later on in this document. Further information about the recount project can be found in the <a href="https://www.nature.com/articles/nbt.3838">main publication</a>. Check the <a href="https://jhubiostatistics.shinyapps.io/recount/">recount website</a> for related publications.</p>
<div class="sourceCode" id="cb4"><html><body><pre class="r"><span class="co">## Load library</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"recount"</span>)

<span class="co">## Find a project of interest</span>
<span class="no">project_info</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/abstract_search.html">abstract_search</a></span>(<span class="st">"GSE32465"</span>)

<span class="co">## Download the gene-level RangedSummarizedExperiment data</span>
<span class="fu"><a href="../reference/download_study.html">download_study</a></span>(<span class="no">project_info</span>$<span class="no">project</span>)

<span class="co">## Load the data</span>
<span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span>(<span class="no">project_info</span>$<span class="no">project</span>, <span class="st">"rse_gene.Rdata"</span>))

<span class="co">## Browse the project at SRA</span>
<span class="fu"><a href="../reference/browse_study.html">browse_study</a></span>(<span class="no">project_info</span>$<span class="no">project</span>)

<span class="co">## View GEO ids</span>
<span class="fu">colData</span>(<span class="no">rse_gene</span>)$<span class="no">geo_accession</span>

<span class="co">## Extract the sample characteristics</span>
<span class="no">geochar</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/split.html">split</a></span>(<span class="fu">colData</span>(<span class="no">rse_gene</span>), <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq_len</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span>(<span class="fu">colData</span>(<span class="no">rse_gene</span>)))), <span class="no">geo_characteristics</span>)

<span class="co">## Note that the information for this study is a little inconsistent, so we</span>
<span class="co">## have to fix it.</span>
<span class="no">geochar</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/do.call.html">do.call</a></span>(<span class="no">rbind</span>, <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span>(<span class="no">geochar</span>, <span class="kw">function</span>(<span class="no">x</span>) {
    <span class="kw">if</span> (<span class="st">"cells"</span> <span class="kw">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="no">x</span>)) {
        <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="no">x</span>)[<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="no">x</span>) <span class="kw">==</span> <span class="st">"cells"</span>] <span class="kw">&lt;-</span> <span class="st">"cell.line"</span>
        <span class="fu"><a href="https://rdrr.io/r/base/function.html">return</a></span>(<span class="no">x</span>)
    } <span class="kw">else</span> {
        <span class="fu"><a href="https://rdrr.io/r/base/function.html">return</a></span>(<span class="no">x</span>)
    }
}))

<span class="co">## We can now define some sample information to use</span>
<span class="no">sample_info</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span>(
    <span class="kw">run</span> <span class="kw">=</span> <span class="fu">colData</span>(<span class="no">rse_gene</span>)$<span class="no">run</span>,
    <span class="kw">group</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/grep.html">grepl</a></span>(<span class="st">"uninduced"</span>, <span class="fu">colData</span>(<span class="no">rse_gene</span>)$<span class="no">title</span>), <span class="st">"uninduced"</span>, <span class="st">"induced"</span>),
    <span class="kw">gene_target</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span>(<span class="fu">colData</span>(<span class="no">rse_gene</span>)$<span class="no">title</span>, <span class="kw">function</span>(<span class="no">x</span>) {
        <span class="fu"><a href="https://rdrr.io/r/base/strsplit.html">strsplit</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/strsplit.html">strsplit</a></span>(
            <span class="no">x</span>,
            <span class="st">"targeting "</span>
        )<span class="kw">[[</span><span class="fl">1</span>]][<span class="fl">2</span>], <span class="st">" gene"</span>)<span class="kw">[[</span><span class="fl">1</span>]][<span class="fl">1</span>]
    }),
    <span class="kw">cell.line</span> <span class="kw">=</span> <span class="no">geochar</span>$<span class="no">cell.line</span>
)

<span class="co">## Scale counts by taking into account the total coverage per sample</span>
<span class="no">rse</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/scale_counts.html">scale_counts</a></span>(<span class="no">rse_gene</span>)

<span class="co">## Add sample information for DE analysis</span>
<span class="fu">colData</span>(<span class="no">rse</span>)$<span class="no">group</span> <span class="kw">&lt;-</span> <span class="no">sample_info</span>$<span class="no">group</span>
<span class="fu">colData</span>(<span class="no">rse</span>)$<span class="no">gene_target</span> <span class="kw">&lt;-</span> <span class="no">sample_info</span>$<span class="no">gene_target</span>

<span class="co">## Perform differential expression analysis with DESeq2</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"DESeq2"</span>)

<span class="co">## Specify design and switch to DESeq2 format</span>
<span class="no">dds</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeqDataSet.html">DESeqDataSet</a></span>(<span class="no">rse</span>, ~ <span class="no">gene_target</span> + <span class="no">group</span>)

<span class="co">## Perform DE analysis</span>
<span class="no">dds</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeq.html">DESeq</a></span>(<span class="no">dds</span>, <span class="kw">test</span> <span class="kw">=</span> <span class="st">"LRT"</span>, <span class="kw">reduced</span> <span class="kw">=</span> ~<span class="no">gene_target</span>, <span class="kw">fitType</span> <span class="kw">=</span> <span class="st">"local"</span>)
<span class="no">res</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/results.html">results</a></span>(<span class="no">dds</span>)

<span class="co">## Explore results</span>
<span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/plotMA.html">plotMA</a></span>(<span class="no">res</span>, <span class="kw">main</span> <span class="kw">=</span> <span class="st">"DESeq2 results for SRP009615"</span>)

<span class="co">## Make a report with the results</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"regionReport"</span>)
<span class="fu"><a href="https://rdrr.io/pkg/regionReport/man/DESeq2Report.html">DESeq2Report</a></span>(<span class="no">dds</span>,
    <span class="kw">res</span> <span class="kw">=</span> <span class="no">res</span>, <span class="kw">project</span> <span class="kw">=</span> <span class="st">"SRP009615"</span>,
    <span class="kw">intgroup</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"group"</span>, <span class="st">"gene_target"</span>), <span class="kw">outdir</span> <span class="kw">=</span> <span class="st">"."</span>,
    <span class="kw">output</span> <span class="kw">=</span> <span class="st">"SRP009615-results"</span>
)</pre></body></html></div>
<p>The <a href="https://jhubiostatistics.shinyapps.io/recount/">recount project</a> also hosts the necessary data to perform annotation-agnostic differential expression analyses with <em><a href="https://bioconductor.org/packages/3.12/derfinder">derfinder</a></em> <a id="cite-Collado-Torres_2017ab"></a>(<a href="http://nar.oxfordjournals.org/content/early/2016/09/29/nar.gkw852">Collado-Torres, Nellore, Frazee, Wilks, et al., 2017</a>). An example analysis would like this:</p>
<div class="sourceCode" id="cb5"><html><body><pre class="r"><span class="co">## Define expressed regions for study SRP009615, only for chromosome Y</span>
<span class="no">regions</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/expressed_regions.html">expressed_regions</a></span>(<span class="st">"SRP009615"</span>, <span class="st">"chrY"</span>,
    <span class="kw">cutoff</span> <span class="kw">=</span> <span class="fl">5L</span>,
    <span class="kw">maxClusterGap</span> <span class="kw">=</span> <span class="fl">3000L</span>
)

<span class="co">## Compute coverage matrix for study SRP009615, only for chromosome Y</span>
<span class="fu"><a href="https://rdrr.io/r/base/system.time.html">system.time</a></span>(<span class="no">rse_ER</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/coverage_matrix.html">coverage_matrix</a></span>(<span class="st">"SRP009615"</span>, <span class="st">"chrY"</span>, <span class="no">regions</span>))

<span class="co">## Round the coverage matrix to integers</span>
<span class="no">covMat</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span>(<span class="fu">assays</span>(<span class="no">rse_ER</span>)$<span class="no">counts</span>, <span class="fl">0</span>)

<span class="co">## Get phenotype data for study SRP009615</span>
<span class="no">pheno</span> <span class="kw">&lt;-</span> <span class="fu">colData</span>(<span class="no">rse_ER</span>)

<span class="co">## Complete the phenotype table with the data we got from GEO</span>
<span class="no">m</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/match.html">match</a></span>(<span class="no">pheno</span>$<span class="no">run</span>, <span class="no">sample_info</span>$<span class="no">run</span>)
<span class="no">pheno</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html">cbind</a></span>(<span class="no">pheno</span>, <span class="no">sample_info</span>[<span class="no">m</span>, <span class="fl">2</span>:<span class="fl">3</span>])

<span class="co">## Build a DESeqDataSet</span>
<span class="no">dds_ers</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeqDataSet.html">DESeqDataSetFromMatrix</a></span>(
    <span class="kw">countData</span> <span class="kw">=</span> <span class="no">covMat</span>, <span class="kw">colData</span> <span class="kw">=</span> <span class="no">pheno</span>,
    <span class="kw">design</span> <span class="kw">=</span> ~ <span class="no">gene_target</span> + <span class="no">group</span>
)

<span class="co">## Perform differential expression analysis with DESeq2 at the ER-level</span>
<span class="no">dds_ers</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeq.html">DESeq</a></span>(<span class="no">dds_ers</span>,
    <span class="kw">test</span> <span class="kw">=</span> <span class="st">"LRT"</span>, <span class="kw">reduced</span> <span class="kw">=</span> ~<span class="no">gene_target</span>,
    <span class="kw">fitType</span> <span class="kw">=</span> <span class="st">"local"</span>
)
<span class="no">res_ers</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/results.html">results</a></span>(<span class="no">dds_ers</span>)

<span class="co">## Explore results</span>
<span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/plotMA.html">plotMA</a></span>(<span class="no">res_ers</span>, <span class="kw">main</span> <span class="kw">=</span> <span class="st">"DESeq2 results for SRP009615 (ER-level, chrY)"</span>)

<span class="co">## Create a more extensive exploratory report</span>
<span class="fu"><a href="https://rdrr.io/pkg/regionReport/man/DESeq2Report.html">DESeq2Report</a></span>(<span class="no">dds_ers</span>,
    <span class="kw">res</span> <span class="kw">=</span> <span class="no">res_ers</span>,
    <span class="kw">project</span> <span class="kw">=</span> <span class="st">"SRP009615 (ER-level, chrY)"</span>,
    <span class="kw">intgroup</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"group"</span>, <span class="st">"gene_target"</span>), <span class="kw">outdir</span> <span class="kw">=</span> <span class="st">"."</span>,
    <span class="kw">output</span> <span class="kw">=</span> <span class="st">"SRP009615-results-ER-level-chrY"</span>
)</pre></body></html></div>
</div>
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p><em><a href="https://bioconductor.org/packages/3.12/recount">recount</a></em> is an R package that provides an interface to the <a href="https://jhubiostatistics.shinyapps.io/recount/">recount project website</a>. This package allows you to download the files from the recount project and has helper functions for getting you started with differential expression analyses. This vignette will walk you through an example.</p>
</div>
<div id="sample-de-analysis" class="section level1">
<h1 class="hasAnchor">
<a href="#sample-de-analysis" class="anchor"></a>Sample DE analysis</h1>
<p>This is a brief overview of what you can do with <em><a href="https://bioconductor.org/packages/3.12/recount">recount</a></em>. In this particular example we will download data from the <a href="http://trace.ncbi.nlm.nih.gov/Traces/sra/?study=SRP009615">SRP009615</a> study which sequenced 12 samples as described in the previous link.</p>
<p>If you don’t have <em><a href="https://bioconductor.org/packages/3.12/recount">recount</a></em> installed, please do so with:</p>
<div class="sourceCode" id="cb6"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span>(<span class="st">"BiocManager"</span>)
<span class="kw pkg">BiocManager</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span>(<span class="st">"recount"</span>)</pre></body></html></div>
<p>Next we load the required packages. Loading <em><a href="https://bioconductor.org/packages/3.12/recount">recount</a></em> will load the required dependencies.</p>
<div class="sourceCode" id="cb7"><html><body><pre class="r"><span class="co">## Load recount R package</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"recount"</span>)</pre></body></html></div>
<p>Lets say that we don’t know the actual SRA accession number for this study but we do know a particular term which will help us identify it. If that’s the case, we can use the <code><a href="../reference/abstract_search.html">abstract_search()</a></code> function to identify the study of interest as shown below.</p>
<div class="sourceCode" id="cb8"><html><body><pre class="r"><span class="co">## Find a project of interest</span>
<span class="no">project_info</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/abstract_search.html">abstract_search</a></span>(<span class="st">"GSE32465"</span>)

<span class="co">## Explore info</span>
<span class="no">project_info</span></pre></body></html></div>
<pre><code>##     number_samples species
## 340             12   human
##                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               abstract
## 340 Summary: K562-shX cells are made in an effort to validate TFBS data and ChIP-seq antibodies in Myers lab (GSE32465). K562 cells are transduced with lentiviral vector having Tet-inducible shRNA targeting a transcription factor gene. Cells with stable integration of shRNA constructs are selected using puromycin in growth media. Doxycycline is added to the growth media to induce the expression of shRNA and a red fluorescent protein marker.  A successful shRNA cell line shows at least a 70% reduction in expression of the target transcription factor as measured by qPCR. For identification, we designated these cell lines as K562-shX, where X is the transcription factor targeted by shRNA and K562 denotes the parent cell line.  For example, K562-shATF3 cells are K562 derived cells selected for stable integration of shRNA targeting the transcription factor ATF3 gene and showed at least a 70% reduction in the expression of ATF3 gene when measured by qPCR. Cells growing without doxycycline (uninduced) are used as a control to measure the change in expression of target transcription factor gene after induction of shRNA using doxycycline. For detailed growth and culturing protocols for these cells please refer to http://hudsonalpha.org/myers-lab/protocols . To identify the potential downstream targets of the candidate transcription factor, analyze the mRNA expression profile of the uninduced and induced K562-shX using RNA-seq.    For data usage terms and conditions, please refer to http://www.genome.gov/27528022 and http://www.genome.gov/Pages/Research/ENCODE/ENCODEDataReleasePolicyFinal2008.pdf    Overall Design: Make K562-shX cells as described in the http://hudsonalpha.org/myers-lab/protocols . Measure the mRNA expression levels in uninduced K562-shX and induced K562-shX cells in two biological replicates using RNA-seq. Identify the potential downstream targets of the candidate transcription factor.
##       project
## 340 SRP009615</code></pre>
<div id="download-data" class="section level2">
<h2 class="hasAnchor">
<a href="#download-data" class="anchor"></a>Download data</h2>
<p>Now that we have a study that we are interested in, we can download the <em>RangedSummarizedExperiment</em> object (see <em><a href="https://bioconductor.org/packages/3.12/SummarizedExperiment">SummarizedExperiment</a></em>) with the data summarized at the gene level. The function <code><a href="../reference/download_study.html">download_study()</a></code> helps us do this. If you are interested on how the annotation was defined, check <code><a href="../reference/reproduce_ranges.html">reproduce_ranges()</a></code> described in the <em>Annotation</em> section further down.</p>
<div class="sourceCode" id="cb10"><html><body><pre class="r"><span class="co">## Download the gene-level RangedSummarizedExperiment data</span>
<span class="fu"><a href="../reference/download_study.html">download_study</a></span>(<span class="no">project_info</span>$<span class="no">project</span>)</pre></body></html></div>
<pre><code>## 2020-05-13 16:51:29 downloading file rse_gene.Rdata to SRP009615</code></pre>
<div class="sourceCode" id="cb12"><html><body><pre class="r"><span class="co">## Load the data</span>
<span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span>(<span class="no">project_info</span>$<span class="no">project</span>, <span class="st">"rse_gene.Rdata"</span>))

<span class="co">## Delete it if you don't need it anymore</span>
<span class="fu"><a href="https://rdrr.io/r/base/unlink.html">unlink</a></span>(<span class="no">project_info</span>$<span class="no">project</span>, <span class="kw">recursive</span> <span class="kw">=</span> <span class="fl">TRUE</span>)</pre></body></html></div>
<p>We can explore a bit this <em>RangedSummarizedExperiment</em> as shown below.</p>
<div class="sourceCode" id="cb13"><html><body><pre class="r"><span class="no">rse_gene</span></pre></body></html></div>
<pre><code>## class: RangedSummarizedExperiment 
## dim: 58037 12 
## metadata(0):
## assays(1): counts
## rownames(58037): ENSG00000000003.14 ENSG00000000005.5 ...
##   ENSG00000283698.1 ENSG00000283699.1
## rowData names(3): gene_id bp_length symbol
## colnames(12): SRR387777 SRR387778 ... SRR389083 SRR389084
## colData names(21): project sample ... title characteristics</code></pre>
<div class="sourceCode" id="cb15"><html><body><pre class="r"><span class="co">## This is the sample phenotype data provided by the recount project</span>
<span class="fu">colData</span>(<span class="no">rse_gene</span>)</pre></body></html></div>
<pre><code>## DataFrame with 12 rows and 21 columns
##               project      sample  experiment         run
##           &lt;character&gt; &lt;character&gt; &lt;character&gt; &lt;character&gt;
## SRR387777   SRP009615   SRS281685   SRX110461   SRR387777
## SRR387778   SRP009615   SRS281686   SRX110462   SRR387778
## SRR387779   SRP009615   SRS281687   SRX110463   SRR387779
## SRR387780   SRP009615   SRS281688   SRX110464   SRR387780
## SRR389077   SRP009615   SRS282369   SRX111299   SRR389077
## ...               ...         ...         ...         ...
## SRR389080   SRP009615   SRS282372   SRX111302   SRR389080
## SRR389081   SRP009615   SRS282373   SRX111303   SRR389081
## SRR389082   SRP009615   SRS282374   SRX111304   SRR389082
## SRR389083   SRP009615   SRS282375   SRX111305   SRR389083
## SRR389084   SRP009615   SRS282376   SRX111306   SRR389084
##           read_count_as_reported_by_sra reads_downloaded
##                               &lt;integer&gt;        &lt;integer&gt;
## SRR387777                      30631853         30631853
## SRR387778                      37001306         37001306
## SRR387779                      40552001         40552001
## SRR387780                      32466352         32466352
## SRR389077                      27819603         27819603
## ...                                 ...              ...
## SRR389080                      34856203         34856203
## SRR389081                      23351679         23351679
## SRR389082                      18144828         18144828
## SRR389083                      24417368         24417368
## SRR389084                      23060084         23060084
##           proportion_of_reads_reported_by_sra_downloaded paired_end
##                                                &lt;numeric&gt;  &lt;logical&gt;
## SRR387777                                              1      FALSE
## SRR387778                                              1      FALSE
## SRR387779                                              1      FALSE
## SRR387780                                              1      FALSE
## SRR389077                                              1      FALSE
## ...                                                  ...        ...
## SRR389080                                              1      FALSE
## SRR389081                                              1      FALSE
## SRR389082                                              1      FALSE
## SRR389083                                              1      FALSE
## SRR389084                                              1      FALSE
##           sra_misreported_paired_end mapped_read_count        auc
##                            &lt;logical&gt;         &lt;integer&gt;  &lt;numeric&gt;
## SRR387777                      FALSE          28798572 1029494445
## SRR387778                      FALSE          33170281 1184877985
## SRR387779                      FALSE          37322762 1336528969
## SRR387780                      FALSE          29970735 1073178116
## SRR389077                      FALSE          24966859  893978355
## ...                              ...               ...        ...
## SRR389080                      FALSE          32469994 1163527939
## SRR389081                      FALSE          21904197  781685955
## SRR389082                      FALSE          17199795  616048853
## SRR389083                      FALSE          22499386  806323346
## SRR389084                      FALSE          21957003  787795710
##           sharq_beta_tissue sharq_beta_cell_type biosample_submission_date
##                 &lt;character&gt;          &lt;character&gt;               &lt;character&gt;
## SRR387777             blood                 k562   2011-12-05T15:40:03.870
## SRR387778             blood                 k562   2011-12-05T15:40:03.897
## SRR387779             blood                 k562   2011-12-05T15:40:03.910
## SRR387780             blood                 k562   2011-12-05T15:40:03.923
## SRR389077             blood                 k562   2011-12-13T11:26:05.720
## ...                     ...                  ...                       ...
## SRR389080             blood                 k562   2011-12-13T11:26:05.787
## SRR389081             blood                 k562   2011-12-13T11:26:05.800
## SRR389082             blood                 k562   2011-12-13T11:26:05.817
## SRR389083             blood                 k562   2011-12-13T11:26:05.830
## SRR389084             blood                 k562   2011-12-13T11:26:05.847
##           biosample_publication_date   biosample_update_date avg_read_length
##                          &lt;character&gt;             &lt;character&gt;       &lt;integer&gt;
## SRR387777    2011-12-07T09:29:59.890 2014-08-27T04:18:20.530              36
## SRR387778    2011-12-07T09:29:59.890 2014-08-27T04:18:21.053              36
## SRR387779    2011-12-07T09:29:59.890 2014-08-27T04:18:21.800              36
## SRR387780    2011-12-07T09:29:59.890 2014-08-27T04:18:22.320              36
## SRR389077    2011-12-13T11:26:06.663 2014-08-27T04:22:14.857              36
## ...                              ...                     ...             ...
## SRR389080    2011-12-13T11:26:06.663 2014-08-27T04:22:15.933              36
## SRR389081    2011-12-13T11:26:06.663 2014-08-27T04:22:16.290              36
## SRR389082    2011-12-13T11:26:06.663 2014-08-27T04:22:16.647              36
## SRR389083    2011-12-13T11:26:06.663 2014-08-27T04:22:17.037              36
## SRR389084    2011-12-13T11:26:06.663 2014-08-27T04:22:17.473              36
##           geo_accession  bigwig_file
##             &lt;character&gt;  &lt;character&gt;
## SRR387777     GSM836270 SRR387777.bw
## SRR387778     GSM836271 SRR387778.bw
## SRR387779     GSM836272 SRR387779.bw
## SRR387780     GSM836273 SRR387780.bw
## SRR389077     GSM847561 SRR389077.bw
## ...                 ...          ...
## SRR389080     GSM847564 SRR389080.bw
## SRR389081     GSM847565 SRR389081.bw
## SRR389082     GSM847566 SRR389082.bw
## SRR389083     GSM847567 SRR389083.bw
## SRR389084     GSM847568 SRR389084.bw
##                                                                                                     title
##                                                                                               &lt;character&gt;
## SRR387777   K562 cells with shRNA targeting SRF gene cultured with no doxycycline (uninduced - UI), rep1.
## SRR387778  K562 cells with shRNA targeting SRF gene cultured with doxycycline for 48 hours (48 hr), rep1.
## SRR387779   K562 cells with shRNA targeting SRF gene cultured with no doxycycline (uninduced - UI), rep2.
## SRR387780  K562 cells with shRNA targeting SRF gene cultured with doxycycline for 48 hours (48 hr), rep2.
## SRR389077  K562 cells with shRNA targeting EGR1 gene cultured with no doxycycline (uninduced - UI), rep1.
## ...                                                                                                   ...
## SRR389080 K562 cells with shRNA targeting EGR1 gene cultured with doxycycline for 72 hours (72 hr), rep2.
## SRR389081  K562 cells with shRNA targeting ATF3 gene cultured with no doxycycline (uninduced - UI), rep1.
## SRR389082 K562 cells with shRNA targeting ATF3 gene cultured with doxycycline for 24 hours (24 hr), rep1.
## SRR389083  K562 cells with shRNA targeting ATF3 gene cultured with no doxycycline (uninduced - UI), rep2.
## SRR389084 K562 cells with shRNA targeting ATF3 gene cultured with doxycycline for 24 hours (24 hr), rep2.
##                                                                                               characteristics
##                                                                                               &lt;CharacterList&gt;
## SRR387777                                               cells: K562,shRNA expression: no,treatment: Puromycin
## SRR387778                  cells: K562,shRNA expression: yes, targeting SRF,treatment: Puromycin, doxycycline
## SRR387779                                               cells: K562,shRNA expression: no,treatment: Puromycin
## SRR387780                   cells: K562,shRNA expression: yes targeting SRF,treatment: Puromycin, doxycycline
## SRR389077                          cell line: K562,shRNA expression: no shRNA expression,treatment: Puromycin
## ...                                                                                                       ...
## SRR389080 cell line: K562,shRNA expression: expressing shRNA targeting EGR1,treatment: Puromycin, doxycycline
## SRR389081                          cell line: K562,shRNA expression: no shRNA expression,treatment: Puromycin
## SRR389082 cell line: K562,shRNA expression: expressing shRNA targeting ATF3,treatment: Puromycin, doxycycline
## SRR389083                          cell line: K562,shRNA expression: no shRNA expression,treatment: Puromycin
## SRR389084 cell line: K562,shRNA expression: expressing shRNA targeting ATF3,treatment: Puromycin, doxycycline</code></pre>
<div class="sourceCode" id="cb17"><html><body><pre class="r"><span class="co">## At the gene level, the row data includes the gene Gencode ids, the gene</span>
<span class="co">## symbols and the sum of the disjoint exons widths, which can be used for</span>
<span class="co">## taking into account the gene length.</span>
<span class="fu">rowData</span>(<span class="no">rse_gene</span>)</pre></body></html></div>
<pre><code>## DataFrame with 58037 rows and 3 columns
##                               gene_id bp_length          symbol
##                           &lt;character&gt; &lt;integer&gt; &lt;CharacterList&gt;
## ENSG00000000003.14 ENSG00000000003.14      4535          TSPAN6
## ENSG00000000005.5   ENSG00000000005.5      1610            TNMD
## ENSG00000000419.12 ENSG00000000419.12      1207            DPM1
## ENSG00000000457.13 ENSG00000000457.13      6883           SCYL3
## ENSG00000000460.16 ENSG00000000460.16      5967        C1orf112
## ...                               ...       ...             ...
## ENSG00000283695.1   ENSG00000283695.1        61              NA
## ENSG00000283696.1   ENSG00000283696.1       997              NA
## ENSG00000283697.1   ENSG00000283697.1      1184           HSFX3
## ENSG00000283698.1   ENSG00000283698.1       940              NA
## ENSG00000283699.1   ENSG00000283699.1        60         MIR4481</code></pre>
<div class="sourceCode" id="cb19"><html><body><pre class="r">## At the exon level, you can get the gene Gencode ids from the names of:
# rowRanges(rse_exon)</pre></body></html></div>
</div>
<div id="finding-phenotype-information" class="section level2">
<h2 class="hasAnchor">
<a href="#finding-phenotype-information" class="anchor"></a>Finding phenotype information</h2>
<p>Once we have identified the study of interest, we can use the <code><a href="../reference/browse_study.html">browse_study()</a></code> function to browse the study at the SRA website.</p>
<div class="sourceCode" id="cb20"><html><body><pre class="r"><span class="co">## Browse the project at SRA</span>
<span class="fu"><a href="../reference/browse_study.html">browse_study</a></span>(<span class="no">project_info</span>$<span class="no">project</span>)</pre></body></html></div>
<p>The SRA website includes an <em>Experiments</em> link which further describes each of the samples. From the information available for <a href="http://www.ncbi.nlm.nih.gov/sra/?term=SRP009615">SRP009615 at NCBI</a> we have some further sample information that we can save for use in our differential expression analysis. We can get some of this information from <a href="http://www.ncbi.nlm.nih.gov/geo/">GEO</a>. The function <code><a href="../reference/find_geo.html">find_geo()</a></code> finds the <a href="http://www.ncbi.nlm.nih.gov/geo/">GEO</a> accession id for a given SRA run accession id, which we can then use with <code><a href="../reference/geo_info.html">geo_info()</a></code> and <code><a href="../reference/geo_characteristics.html">geo_characteristics()</a></code> to parse this information. The <code>rse_gene</code> object already has some of this information.</p>
<div class="sourceCode" id="cb21"><html><body><pre class="r"><span class="co">## View GEO ids</span>
<span class="fu">colData</span>(<span class="no">rse_gene</span>)$<span class="no">geo_accession</span></pre></body></html></div>
<pre><code>##  [1] "GSM836270" "GSM836271" "GSM836272" "GSM836273" "GSM847561" "GSM847562"
##  [7] "GSM847563" "GSM847564" "GSM847565" "GSM847566" "GSM847567" "GSM847568"</code></pre>
<div class="sourceCode" id="cb23"><html><body><pre class="r"><span class="co">## Extract the sample characteristics</span>
<span class="no">geochar</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/split.html">split</a></span>(<span class="fu">colData</span>(<span class="no">rse_gene</span>), <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq_len</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span>(<span class="fu">colData</span>(<span class="no">rse_gene</span>)))), <span class="no">geo_characteristics</span>)

<span class="co">## Note that the information for this study is a little inconsistent, so we</span>
<span class="co">## have to fix it.</span>
<span class="no">geochar</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/do.call.html">do.call</a></span>(<span class="no">rbind</span>, <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span>(<span class="no">geochar</span>, <span class="kw">function</span>(<span class="no">x</span>) {
    <span class="kw">if</span> (<span class="st">"cells"</span> <span class="kw">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="no">x</span>)) {
        <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="no">x</span>)[<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="no">x</span>) <span class="kw">==</span> <span class="st">"cells"</span>] <span class="kw">&lt;-</span> <span class="st">"cell.line"</span>
        <span class="fu"><a href="https://rdrr.io/r/base/function.html">return</a></span>(<span class="no">x</span>)
    } <span class="kw">else</span> {
        <span class="fu"><a href="https://rdrr.io/r/base/function.html">return</a></span>(<span class="no">x</span>)
    }
}))

<span class="co">## We can now define some sample information to use</span>
<span class="no">sample_info</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span>(
    <span class="kw">run</span> <span class="kw">=</span> <span class="fu">colData</span>(<span class="no">rse_gene</span>)$<span class="no">run</span>,
    <span class="kw">group</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/grep.html">grepl</a></span>(<span class="st">"uninduced"</span>, <span class="fu">colData</span>(<span class="no">rse_gene</span>)$<span class="no">title</span>), <span class="st">"uninduced"</span>, <span class="st">"induced"</span>),
    <span class="kw">gene_target</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span>(<span class="fu">colData</span>(<span class="no">rse_gene</span>)$<span class="no">title</span>, <span class="kw">function</span>(<span class="no">x</span>) {
        <span class="fu"><a href="https://rdrr.io/r/base/strsplit.html">strsplit</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/strsplit.html">strsplit</a></span>(
            <span class="no">x</span>,
            <span class="st">"targeting "</span>
        )<span class="kw">[[</span><span class="fl">1</span>]][<span class="fl">2</span>], <span class="st">" gene"</span>)<span class="kw">[[</span><span class="fl">1</span>]][<span class="fl">1</span>]
    }),
    <span class="kw">cell.line</span> <span class="kw">=</span> <span class="no">geochar</span>$<span class="no">cell.line</span>
)</pre></body></html></div>
</div>
<div id="predicted-phenotype-information" class="section level2">
<h2 class="hasAnchor">
<a href="#predicted-phenotype-information" class="anchor"></a>Predicted phenotype information</h2>
<p>Shannon Ellis et at have predicted phenotypic information that can be used with any data from the recount project thanks to <code><a href="../reference/add_predictions.html">add_predictions()</a></code>. Check that function for more details <a id="cite-Ellis_2018"></a>(<a href="https://doi.org/10.1093/nar/gky102">Ellis, Collado-Torres, Jaffe, and Leek, 2018</a>).</p>
</div>
<div id="de-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#de-analysis" class="anchor"></a>DE analysis</h2>
<p>The <a href="https://jhubiostatistics.shinyapps.io/recount/">recount project</a> records the sum of the base level coverage for each gene (or exon). These raw counts have to be scaled and there are several ways in which you can choose to do so. The function <code><a href="../reference/scale_counts.html">scale_counts()</a></code> helps you scale them in a way that is tailored to <a href="http://rail.bio">Rail-RNA</a> output. If you prefer read counts without scaling, check the function <code><a href="../reference/read_counts.html">read_counts()</a></code>. Below we show some of the differences.</p>
<div class="sourceCode" id="cb24"><html><body><pre class="r"><span class="co">## Scale counts by taking into account the total coverage per sample</span>
<span class="no">rse</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/scale_counts.html">scale_counts</a></span>(<span class="no">rse_gene</span>)

<span class="co">##### Details about counts #####</span>

<span class="co">## Scale counts to 40 million mapped reads. Not all mapped reads are in exonic</span>
<span class="co">## sequence, so the total is not necessarily 40 million.</span>
<span class="fu"><a href="https://rdrr.io/r/base/colSums.html">colSums</a></span>(<span class="fu">assays</span>(<span class="no">rse</span>)$<span class="no">counts</span>) / <span class="fl">1e6</span></pre></body></html></div>
<pre><code>## SRR387777 SRR387778 SRR387779 SRR387780 SRR389077 SRR389078 SRR389079 SRR389080 
##  30.26702  29.07199  35.38355  34.65798  23.36050  22.62014  34.74629  35.20971 
## SRR389081 SRR389082 SRR389083 SRR389084 
##  33.41459  36.78133  34.05013  33.95469</code></pre>
<div class="sourceCode" id="cb26"><html><body><pre class="r"><span class="co">## Compute read counts</span>
<span class="no">rse_read_counts</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/read_counts.html">read_counts</a></span>(<span class="no">rse_gene</span>)
<span class="co">## Difference between read counts and number of reads downloaded by Rail-RNA</span>
<span class="fu"><a href="https://rdrr.io/r/base/colSums.html">colSums</a></span>(<span class="fu">assays</span>(<span class="no">rse_read_counts</span>)$<span class="no">counts</span>) / <span class="fl">1e6</span> -
    <span class="fu">colData</span>(<span class="no">rse_read_counts</span>)$<span class="no">reads_downloaded</span> / <span class="fl">1e6</span></pre></body></html></div>
<pre><code>##  SRR387777  SRR387778  SRR387779  SRR387780  SRR389077  SRR389078  SRR389079 
##  -8.839994 -12.892541  -7.536284  -6.497571 -13.239233 -15.816168 -14.775568 
##  SRR389080  SRR389081  SRR389082  SRR389083  SRR389084 
##  -6.274001  -5.054220  -2.328694  -5.265160  -4.421728</code></pre>
<div class="sourceCode" id="cb28"><html><body><pre class="r">## Check the help page for read_counts() for more details</pre></body></html></div>
<p>We are almost ready to perform our differential expression analysis. Lets just add the information we recovered <a href="http://www.ncbi.nlm.nih.gov/geo/">GEO</a> about these samples.</p>
<div class="sourceCode" id="cb29"><html><body><pre class="r"><span class="co">## Add sample information for DE analysis</span>
<span class="fu">colData</span>(<span class="no">rse</span>)$<span class="no">group</span> <span class="kw">&lt;-</span> <span class="no">sample_info</span>$<span class="no">group</span>
<span class="fu">colData</span>(<span class="no">rse</span>)$<span class="no">gene_target</span> <span class="kw">&lt;-</span> <span class="no">sample_info</span>$<span class="no">gene_target</span></pre></body></html></div>
<p>Now that the <em>RangedSummarizedExperiment</em> is complete, we can use <em><a href="https://bioconductor.org/packages/3.12/DESeq2">DESeq2</a></em> or another package to perform the differential expression test. Note that you can use <em><a href="https://bioconductor.org/packages/3.12/DEFormats">DEFormats</a></em> for switching between formats if you want to use another package, like <em><a href="https://bioconductor.org/packages/3.12/edgeR">edgeR</a></em>.</p>
<p>In this particular analysis, we’ll test whether there is a group difference adjusting for the gene target.</p>
<div class="sourceCode" id="cb30"><html><body><pre class="r"><span class="co">## Perform differential expression analysis with DESeq2</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"DESeq2"</span>)

<span class="co">## Specify design and switch to DESeq2 format</span>
<span class="no">dds</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeqDataSet.html">DESeqDataSet</a></span>(<span class="no">rse</span>, ~ <span class="no">gene_target</span> + <span class="no">group</span>)</pre></body></html></div>
<pre><code>## converting counts to integer mode</code></pre>
<pre><code>## Warning in DESeqDataSet(rse, ~gene_target + group): some variables in design
## formula are characters, converting to factors</code></pre>
<div class="sourceCode" id="cb33"><html><body><pre class="r"><span class="co">## Perform DE analysis</span>
<span class="no">dds</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeq.html">DESeq</a></span>(<span class="no">dds</span>, <span class="kw">test</span> <span class="kw">=</span> <span class="st">"LRT"</span>, <span class="kw">reduced</span> <span class="kw">=</span> ~<span class="no">gene_target</span>, <span class="kw">fitType</span> <span class="kw">=</span> <span class="st">"local"</span>)</pre></body></html></div>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates</code></pre>
<pre><code>## fitting model and testing</code></pre>
<div class="sourceCode" id="cb40"><html><body><pre class="r"><span class="no">res</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/results.html">results</a></span>(<span class="no">dds</span>)</pre></body></html></div>
<p>We can now use functions from <em><a href="https://bioconductor.org/packages/3.12/DESeq2">DESeq2</a></em> to explore the results. For more details check the <em><a href="https://bioconductor.org/packages/3.12/DESeq2">DESeq2</a></em> vignette. For example, we can make a MA plot as shown in Figure @ref(fig:maplot).</p>
<div class="sourceCode" id="cb41"><html><body><pre class="r"><span class="co">## Explore results</span>
<span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/plotMA.html">plotMA</a></span>(<span class="no">res</span>, <span class="kw">main</span> <span class="kw">=</span> <span class="st">"DESeq2 results for SRP009615"</span>)</pre></body></html></div>
<div class="figure">
<img src="recount-quickstart_files/figure-html/maplot-1.png" alt="MA plot for group differences adjusted by gene target for SRP009615 using gene-level data." width="700"><p class="caption">
MA plot for group differences adjusted by gene target for SRP009615 using gene-level data.
</p>
</div>
</div>
<div id="report-results" class="section level2">
<h2 class="hasAnchor">
<a href="#report-results" class="anchor"></a>Report results</h2>
<p>We can also use the <em><a href="https://bioconductor.org/packages/3.12/regionReport">regionReport</a></em> package to generate interactive HTML reports exploring the <em><a href="https://bioconductor.org/packages/3.12/DESeq2">DESeq2</a></em> results (or <em><a href="https://bioconductor.org/packages/3.12/edgeR">edgeR</a></em> results if you used that package).</p>
<div class="sourceCode" id="cb42"><html><body><pre class="r"><span class="co">## Make a report with the results</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"regionReport"</span>)
<span class="no">report</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/regionReport/man/DESeq2Report.html">DESeq2Report</a></span>(<span class="no">dds</span>,
    <span class="kw">res</span> <span class="kw">=</span> <span class="no">res</span>, <span class="kw">project</span> <span class="kw">=</span> <span class="st">"SRP009615"</span>,
    <span class="kw">intgroup</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"group"</span>, <span class="st">"gene_target"</span>), <span class="kw">outdir</span> <span class="kw">=</span> <span class="st">"."</span>,
    <span class="kw">output</span> <span class="kw">=</span> <span class="st">"SRP009615-results"</span>, <span class="kw">nBest</span> <span class="kw">=</span> <span class="fl">10</span>, <span class="kw">nBestFeatures</span> <span class="kw">=</span> <span class="fl">2</span>
)</pre></body></html></div>
<pre><code>## Warning in c.bibentry(knitcitations = citation("knitcitations"), regionReport = citation("regionReport")[1], : method is
## only applicable to 'bibentry' objects</code></pre>
<pre><code>## Writing 9 Bibtex entries ... OK
## Results written to file './SRP009615-results.bib'
## processing file: SRP009615-results.Rmd
## output file: SRP009615-results.knit.md
## 
## Output created: SRP009615-results.html</code></pre>
<p><img src="recount-quickstart_files/figure-html/make_report_real-1.png" width="700"></p>
<p>You can view the final report <a href="SRP009615-results.html">here</a>.</p>
</div>
<div id="finding-gene-names" class="section level2">
<h2 class="hasAnchor">
<a href="#finding-gene-names" class="anchor"></a>Finding gene names</h2>
<p>The gene Gencode ids are included in several objects in <code>recount</code>. With these ids you can find the gene symbols, gene names and other information. The <em><a href="https://bioconductor.org/packages/3.12/org.Hs.eg.db">org.Hs.eg.db</a></em> is useful for finding this information and the following code shows how to get the gene names and symbols.</p>
<div class="sourceCode" id="cb45"><html><body><pre class="r"><span class="co">## Load required library</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"org.Hs.eg.db"</span>)</pre></body></html></div>
<pre><code>## Loading required package: AnnotationDbi</code></pre>
<pre><code>## </code></pre>
<div class="sourceCode" id="cb48"><html><body><pre class="r"><span class="co">## Extract Gencode gene ids</span>
<span class="no">gencode</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span>(<span class="st">"\\..*"</span>, <span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="no">recount_genes</span>))

<span class="co">## Find the gene information we are interested in</span>
<span class="no">gene_info</span> <span class="kw">&lt;-</span> <span class="fu">select</span>(<span class="no">org.Hs.eg.db</span>, <span class="no">gencode</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(
    <span class="st">"ENTREZID"</span>, <span class="st">"GENENAME"</span>, <span class="st">"SYMBOL"</span>,
    <span class="st">"ENSEMBL"</span>
), <span class="st">"ENSEMBL"</span>)</pre></body></html></div>
<pre><code>## 'select()' returned many:many mapping between keys and columns</code></pre>
<div class="sourceCode" id="cb50"><html><body><pre class="r"><span class="co">## Explore part of the results</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span>(<span class="no">gene_info</span>)</pre></body></html></div>
<pre><code>## [1] 58265     4</code></pre>
<div class="sourceCode" id="cb52"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="no">gene_info</span>)</pre></body></html></div>
<pre><code>##           ENSEMBL ENTREZID                                                    GENENAME   SYMBOL
## 1 ENSG00000000003     7105                                               tetraspanin 6   TSPAN6
## 2 ENSG00000000005    64102                                                 tenomodulin     TNMD
## 3 ENSG00000000419     8813 dolichyl-phosphate mannosyltransferase subunit 1, catalytic     DPM1
## 4 ENSG00000000457    57147                                    SCY1 like pseudokinase 3    SCYL3
## 5 ENSG00000000460    55732                         chromosome 1 open reading frame 112 C1orf112
## 6 ENSG00000000938     2268              FGR proto-oncogene, Src family tyrosine kinase      FGR</code></pre>
</div>
</div>
<div id="sample-derfinder-analysis" class="section level1">
<h1 class="hasAnchor">
<a href="#sample-derfinder-analysis" class="anchor"></a>Sample <em><a href="https://bioconductor.org/packages/3.12/derfinder">derfinder</a></em> analysis</h1>
<p>The <a href="https://jhubiostatistics.shinyapps.io/recount/">recount project</a> also hosts for each project sample coverage <a href="https://genome.ucsc.edu/goldenpath/help/bigWig.html">bigWig files</a> created by <a href="http://rail.bio">Rail-RNA</a> and a mean coverage bigWig file. For the mean coverage bigWig file, all samples were normalized to libraries of 40 million reads, each a 100 base-pairs long. <em><a href="https://bioconductor.org/packages/3.12/recount">recount</a></em> can be used along with <em><a href="https://bioconductor.org/packages/3.12/derfinder">derfinder</a></em> (<a href="http://nar.oxfordjournals.org/content/early/2016/09/29/nar.gkw852">Collado-Torres, Nellore, Frazee, Wilks, et al., 2017</a>) to identify expressed regions from the data. This type of analysis is annotation-agnostic which can be advantageous. The following subsections illustrate this type of analysis.</p>
<div id="define-expressed-regions" class="section level2">
<h2 class="hasAnchor">
<a href="#define-expressed-regions" class="anchor"></a>Define expressed regions</h2>
<p>For an annotation-agnostic differential expression analysis, we first need to define the regions of interest. With <em><a href="https://bioconductor.org/packages/3.12/recount">recount</a></em> we can do so using the <code><a href="../reference/expressed_regions.html">expressed_regions()</a></code> function as shown below for the same study we studied earlier.</p>
<div class="sourceCode" id="cb54"><html><body><pre class="r"><span class="co">## Define expressed regions for study SRP009615, only for chromosome Y</span>
<span class="no">regions</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/expressed_regions.html">expressed_regions</a></span>(<span class="st">"SRP009615"</span>, <span class="st">"chrY"</span>,
    <span class="kw">cutoff</span> <span class="kw">=</span> <span class="fl">5L</span>,
    <span class="kw">maxClusterGap</span> <span class="kw">=</span> <span class="fl">3000L</span>
)</pre></body></html></div>
<pre><code>## 2020-05-13 16:52:26 loadCoverage: loading BigWig file http://duffel.rail.bio/recount/SRP009615/bw/mean_SRP009615.bw</code></pre>
<pre><code>## 2020-05-13 16:52:28 loadCoverage: applying the cutoff to the merged data</code></pre>
<pre><code>## 2020-05-13 16:52:28 filterData: originally there were 57227415 rows, now there are 57227415 rows. Meaning that 0 percent was filtered.</code></pre>
<pre><code>## 2020-05-13 16:52:28 findRegions: identifying potential segments</code></pre>
<pre><code>## 2020-05-13 16:52:28 findRegions: segmenting information</code></pre>
<pre><code>## 2020-05-13 16:52:28 .getSegmentsRle: segmenting with cutoff(s) 5</code></pre>
<pre><code>## 2020-05-13 16:52:29 findRegions: identifying candidate regions</code></pre>
<pre><code>## 2020-05-13 16:52:30 findRegions: identifying region clusters</code></pre>
<div class="sourceCode" id="cb63"><html><body><pre class="r"><span class="co">## Briefly explore the resulting regions</span>
<span class="no">regions</span></pre></body></html></div>
<pre><code>## GRanges object with 808 ranges and 6 metadata columns:
##       seqnames            ranges strand |     value      area indexStart  indexEnd cluster clusterL
##          &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt;  &lt;integer&gt; &lt;integer&gt;   &lt;Rle&gt;    &lt;Rle&gt;
##     1     chrY   2929794-2929829      * |  14.72650   530.154    2929794   2929829       1       36
##     2     chrY   2956678-2956701      * |  12.81063   307.455    2956678   2956701       2       24
##     3     chrY   2977203-2977227      * |   5.34908   133.727    2977203   2977227       3     2750
##     4     chrY   2977957-2977994      * |   6.46977   245.851    2977957   2977994       3     2750
##     5     chrY   2978850-2978871      * |   5.79766   127.548    2978850   2978871       3     2750
##   ...      ...               ...    ... .       ...       ...        ...       ...     ...      ...
##   804     chrY 26614511-26614546      * |   7.28189  262.1482   26614511  26614546     224      298
##   805     chrY 26614548-26614553      * |   5.48768   32.9261   26614548  26614553     224      298
##   806     chrY 26614779-26614808      * |   6.64339  199.3018   26614779  26614808     224      298
##   807     chrY 26626808-26626848      * |  12.60382  516.7564   26626808  26626848     225      221
##   808     chrY 26626971-26627028      * |  14.16734  821.7055   26626971  26627028     225      221
##   -------
##   seqinfo: 1 sequence from an unspecified genome</code></pre>
<p>Once the regions have been defined, you can export them into a BED file using <em><a href="https://bioconductor.org/packages/3.12/rtracklayer">rtracklayer</a></em> or other file formats.</p>
</div>
<div id="compute-coverage-matrix" class="section level2">
<h2 class="hasAnchor">
<a href="#compute-coverage-matrix" class="anchor"></a>Compute coverage matrix</h2>
<p>Having defined the expressed regions, we can now compute a coverage matrix for these regions. We can do so using the function <code><a href="../reference/coverage_matrix.html">coverage_matrix()</a></code> from <em><a href="https://bioconductor.org/packages/3.12/recount">recount</a></em> as shown below. It returns a <em>RangedSummarizedExperiment</em> object.</p>
<div class="sourceCode" id="cb65"><html><body><pre class="r"><span class="co">## Compute coverage matrix for study SRP009615, only for chromosome Y</span>
<span class="fu"><a href="https://rdrr.io/r/base/system.time.html">system.time</a></span>(<span class="no">rse_ER</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/coverage_matrix.html">coverage_matrix</a></span>(<span class="st">"SRP009615"</span>, <span class="st">"chrY"</span>, <span class="no">regions</span>))</pre></body></html></div>
<pre><code>## 2020-05-13 16:52:32 railMatrix: processing regions 1 to 808</code></pre>
<pre><code>## 2020-05-13 16:52:32 railMatrix: processing file http://duffel.rail.bio/recount/SRP009615/bw/SRR387777.bw</code></pre>
<pre><code>## 2020-05-13 16:52:33 railMatrix: processing file http://duffel.rail.bio/recount/SRP009615/bw/SRR387778.bw</code></pre>
<pre><code>## 2020-05-13 16:52:34 railMatrix: processing file http://duffel.rail.bio/recount/SRP009615/bw/SRR387779.bw</code></pre>
<pre><code>## 2020-05-13 16:52:35 railMatrix: processing file http://duffel.rail.bio/recount/SRP009615/bw/SRR387780.bw</code></pre>
<pre><code>## 2020-05-13 16:52:36 railMatrix: processing file http://duffel.rail.bio/recount/SRP009615/bw/SRR389077.bw</code></pre>
<pre><code>## 2020-05-13 16:52:37 railMatrix: processing file http://duffel.rail.bio/recount/SRP009615/bw/SRR389078.bw</code></pre>
<pre><code>## 2020-05-13 16:52:38 railMatrix: processing file http://duffel.rail.bio/recount/SRP009615/bw/SRR389079.bw</code></pre>
<pre><code>## 2020-05-13 16:52:39 railMatrix: processing file http://duffel.rail.bio/recount/SRP009615/bw/SRR389080.bw</code></pre>
<pre><code>## 2020-05-13 16:52:39 railMatrix: processing file http://duffel.rail.bio/recount/SRP009615/bw/SRR389081.bw</code></pre>
<pre><code>## 2020-05-13 16:52:40 railMatrix: processing file http://duffel.rail.bio/recount/SRP009615/bw/SRR389082.bw</code></pre>
<pre><code>## 2020-05-13 16:52:41 railMatrix: processing file http://duffel.rail.bio/recount/SRP009615/bw/SRR389083.bw</code></pre>
<pre><code>## 2020-05-13 16:52:42 railMatrix: processing file http://duffel.rail.bio/recount/SRP009615/bw/SRR389084.bw</code></pre>
<pre><code>##    user  system elapsed 
##   2.208   0.202  10.988</code></pre>
<div class="sourceCode" id="cb80"><html><body><pre class="r"><span class="co">## Explore the RSE a bit</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span>(<span class="no">rse_ER</span>)</pre></body></html></div>
<pre><code>## [1] 808  12</code></pre>
<div class="sourceCode" id="cb82"><html><body><pre class="r"><span class="no">rse_ER</span></pre></body></html></div>
<pre><code>## class: RangedSummarizedExperiment 
## dim: 808 12 
## metadata(0):
## assays(1): counts
## rownames(808): 1 2 ... 807 808
## rowData names(6): value area ... cluster clusterL
## colnames(12): SRR387777 SRR387778 ... SRR389083 SRR389084
## colData names(21): project sample ... title characteristics</code></pre>
<p>The resulting count matrix has one row per region and one column per sample. The counts correspond to the number (or fraction) of reads overlapping the regions and are scaled by default to a library size of 40 million reads. For some differential expression methods, you might have to round this matrix into integers. We’ll use <em><a href="https://bioconductor.org/packages/3.12/DESeq2">DESeq2</a></em> to identify which expressed regions are differentially expressed.</p>
<div class="sourceCode" id="cb84"><html><body><pre class="r"><span class="co">## Round the coverage matrix to integers</span>
<span class="no">covMat</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span>(<span class="fu">assays</span>(<span class="no">rse_ER</span>)$<span class="no">counts</span>, <span class="fl">0</span>)</pre></body></html></div>
<p>You can use <code>scale = FALSE</code> if you want the raw base-pair coverage counts and then scale them with <code><a href="../reference/scale_counts.html">scale_counts()</a></code>. If you want integer counts, use <code>round = TRUE</code>.</p>
</div>
<div id="construct-a-deseqdataset-object" class="section level2">
<h2 class="hasAnchor">
<a href="#construct-a-deseqdataset-object" class="anchor"></a>Construct a <code>DESeqDataSet</code> object</h2>
<p>We first need to get some phenotype information for these samples similar to the first analysis we did. We can get this data using <code><a href="../reference/download_study.html">download_study()</a></code>.</p>
<div class="sourceCode" id="cb85"><html><body><pre class="r"><span class="co">## Get phenotype data for study SRP009615</span>
<span class="no">pheno</span> <span class="kw">&lt;-</span> <span class="fu">colData</span>(<span class="no">rse_ER</span>)

<span class="co">## Complete the phenotype table with the data we got from GEO</span>
<span class="no">m</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/match.html">match</a></span>(<span class="no">pheno</span>$<span class="no">run</span>, <span class="no">sample_info</span>$<span class="no">run</span>)
<span class="no">pheno</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html">cbind</a></span>(<span class="no">pheno</span>, <span class="no">sample_info</span>[<span class="no">m</span>, <span class="fl">2</span>:<span class="fl">3</span>])

<span class="co">## Explore the phenotype data a little bit</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="no">pheno</span>)</pre></body></html></div>
<pre><code>## DataFrame with 6 rows and 23 columns
##               project      sample  experiment         run read_count_as_reported_by_sra reads_downloaded
##           &lt;character&gt; &lt;character&gt; &lt;character&gt; &lt;character&gt;                     &lt;integer&gt;        &lt;integer&gt;
## SRR387777   SRP009615   SRS281685   SRX110461   SRR387777                      30631853         30631853
## SRR387778   SRP009615   SRS281686   SRX110462   SRR387778                      37001306         37001306
## SRR387779   SRP009615   SRS281687   SRX110463   SRR387779                      40552001         40552001
## SRR387780   SRP009615   SRS281688   SRX110464   SRR387780                      32466352         32466352
## SRR389077   SRP009615   SRS282369   SRX111299   SRR389077                      27819603         27819603
## SRR389078   SRP009615   SRS282370   SRX111300   SRR389078                      31758658         31758658
##           proportion_of_reads_reported_by_sra_downloaded paired_end sra_misreported_paired_end mapped_read_count
##                                                &lt;integer&gt;  &lt;logical&gt;                  &lt;logical&gt;         &lt;integer&gt;
## SRR387777                                              1      FALSE                      FALSE          28798572
## SRR387778                                              1      FALSE                      FALSE          33170281
## SRR387779                                              1      FALSE                      FALSE          37322762
## SRR387780                                              1      FALSE                      FALSE          29970735
## SRR389077                                              1      FALSE                      FALSE          24966859
## SRR389078                                              1      FALSE                      FALSE          28190059
##                  auc sharq_beta_tissue sharq_beta_cell_type biosample_submission_date biosample_publication_date
##            &lt;integer&gt;       &lt;character&gt;          &lt;character&gt;               &lt;character&gt;                &lt;character&gt;
## SRR387777 1029494445             blood                 k562   2011-12-05T15:40:03.870    2011-12-07T09:29:59.890
## SRR387778 1184877985             blood                 k562   2011-12-05T15:40:03.897    2011-12-07T09:29:59.890
## SRR387779 1336528969             blood                 k562   2011-12-05T15:40:03.910    2011-12-07T09:29:59.890
## SRR387780 1073178116             blood                 k562   2011-12-05T15:40:03.923    2011-12-07T09:29:59.890
## SRR389077  893978355             blood                 k562   2011-12-13T11:26:05.720    2011-12-13T11:26:06.663
## SRR389078 1009216922             blood                 k562   2011-12-13T11:26:05.760    2011-12-13T11:26:06.663
##             biosample_update_date avg_read_length geo_accession  bigwig_file
##                       &lt;character&gt;       &lt;integer&gt;   &lt;character&gt;  &lt;character&gt;
## SRR387777 2014-08-27T04:18:20.530              36     GSM836270 SRR387777.bw
## SRR387778 2014-08-27T04:18:21.053              36     GSM836271 SRR387778.bw
## SRR387779 2014-08-27T04:18:21.800              36     GSM836272 SRR387779.bw
## SRR387780 2014-08-27T04:18:22.320              36     GSM836273 SRR387780.bw
## SRR389077 2014-08-27T04:22:14.857              36     GSM847561 SRR389077.bw
## SRR389078 2014-08-27T04:22:15.227              36     GSM847562 SRR389078.bw
##                                                                                                     title
##                                                                                               &lt;character&gt;
## SRR387777   K562 cells with shRNA targeting SRF gene cultured with no doxycycline (uninduced - UI), rep1.
## SRR387778  K562 cells with shRNA targeting SRF gene cultured with doxycycline for 48 hours (48 hr), rep1.
## SRR387779   K562 cells with shRNA targeting SRF gene cultured with no doxycycline (uninduced - UI), rep2.
## SRR387780  K562 cells with shRNA targeting SRF gene cultured with doxycycline for 48 hours (48 hr), rep2.
## SRR389077  K562 cells with shRNA targeting EGR1 gene cultured with no doxycycline (uninduced - UI), rep1.
## SRR389078 K562 cells with shRNA targeting EGR1 gene cultured with doxycycline for 72 hours (72 hr), rep1.
##                                                                                                          characteristics
##                                                                                                              &lt;character&gt;
## SRR387777                                               c("cells: K562", "shRNA expression: no", "treatment: Puromycin")
## SRR387778                  c("cells: K562", "shRNA expression: yes, targeting SRF", "treatment: Puromycin, doxycycline")
## SRR387779                                               c("cells: K562", "shRNA expression: no", "treatment: Puromycin")
## SRR387780                   c("cells: K562", "shRNA expression: yes targeting SRF", "treatment: Puromycin, doxycycline")
## SRR389077                          c("cell line: K562", "shRNA expression: no shRNA expression", "treatment: Puromycin")
## SRR389078 c("cell line: K562", "shRNA expression: expressing shRNA targeting EGR1", "treatment: Puromycin, doxycycline")
##                 group gene_target
##           &lt;character&gt; &lt;character&gt;
## SRR387777   uninduced         SRF
## SRR387778     induced         SRF
## SRR387779   uninduced         SRF
## SRR387780     induced         SRF
## SRR389077   uninduced        EGR1
## SRR389078     induced        EGR1</code></pre>
<p>Now that we have the necessary data, we can construct a <code>DESeqDataSet</code> object using the function <code><a href="https://rdrr.io/pkg/DESeq2/man/DESeqDataSet.html">DESeqDataSetFromMatrix()</a></code> from <em><a href="https://bioconductor.org/packages/3.12/DESeq2">DESeq2</a></em>.</p>
<div class="sourceCode" id="cb87"><html><body><pre class="r"><span class="co">## Build a DESeqDataSet</span>
<span class="no">dds_ers</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeqDataSet.html">DESeqDataSetFromMatrix</a></span>(
    <span class="kw">countData</span> <span class="kw">=</span> <span class="no">covMat</span>, <span class="kw">colData</span> <span class="kw">=</span> <span class="no">pheno</span>,
    <span class="kw">design</span> <span class="kw">=</span> ~ <span class="no">gene_target</span> + <span class="no">group</span>
)</pre></body></html></div>
<pre><code>## converting counts to integer mode</code></pre>
<pre><code>## Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in design formula are characters, converting to
## factors</code></pre>
</div>
<div id="deseq2-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#deseq2-analysis" class="anchor"></a><em><a href="https://bioconductor.org/packages/3.12/DESeq2">DESeq2</a></em> analysis</h2>
<p>With the <code>DESeqDataSet</code> object in place, we can then use the function <code><a href="https://rdrr.io/pkg/DESeq2/man/DESeq.html">DESeq()</a></code> from <em><a href="https://bioconductor.org/packages/3.12/DESeq2">DESeq2</a></em> to perform the differential expression analysis (between groups) as shown below.</p>
<div class="sourceCode" id="cb90"><html><body><pre class="r"><span class="co">## Perform differential expression analysis with DESeq2 at the ER-level</span>
<span class="no">dds_ers</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeq.html">DESeq</a></span>(<span class="no">dds_ers</span>,
    <span class="kw">test</span> <span class="kw">=</span> <span class="st">"LRT"</span>, <span class="kw">reduced</span> <span class="kw">=</span> ~<span class="no">gene_target</span>,
    <span class="kw">fitType</span> <span class="kw">=</span> <span class="st">"local"</span>
)</pre></body></html></div>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates</code></pre>
<pre><code>## fitting model and testing</code></pre>
<div class="sourceCode" id="cb97"><html><body><pre class="r"><span class="no">res_ers</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/results.html">results</a></span>(<span class="no">dds_ers</span>)</pre></body></html></div>
<p>We can then visually explore the results as shown in Figure @ref(fig:maploters), like we did before in Figure @ref(fig:maplot).</p>
<div class="sourceCode" id="cb98"><html><body><pre class="r"><span class="co">## Explore results</span>
<span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/plotMA.html">plotMA</a></span>(<span class="no">res_ers</span>, <span class="kw">main</span> <span class="kw">=</span> <span class="st">"DESeq2 results for SRP009615 (ER-level, chrY)"</span>)</pre></body></html></div>
<div class="figure">
<img src="recount-quickstart_files/figure-html/maploters-1.png" alt="MA plot for group differences adjusted by gene target for SRP009615 using ER-level data (just chrY)." width="700"><p class="caption">
MA plot for group differences adjusted by gene target for SRP009615 using ER-level data (just chrY).
</p>
</div>
<p>We can also use <em><a href="https://bioconductor.org/packages/3.12/regionReport">regionReport</a></em> to create a more extensive exploratory report.</p>
<div class="sourceCode" id="cb99"><html><body><pre class="r"><span class="co">## Create the report</span>
<span class="no">report2</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/regionReport/man/DESeq2Report.html">DESeq2Report</a></span>(<span class="no">dds_ers</span>,
    <span class="kw">res</span> <span class="kw">=</span> <span class="no">res_ers</span>,
    <span class="kw">project</span> <span class="kw">=</span> <span class="st">"SRP009615 (ER-level, chrY)"</span>,
    <span class="kw">intgroup</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"group"</span>, <span class="st">"gene_target"</span>), <span class="kw">outdir</span> <span class="kw">=</span> <span class="st">"."</span>,
    <span class="kw">output</span> <span class="kw">=</span> <span class="st">"SRP009615-results-ER-level-chrY"</span>
)</pre></body></html></div>
</div>
</div>
<div id="annotation-used" class="section level1">
<h1 class="hasAnchor">
<a href="#annotation-used" class="anchor"></a>Annotation used</h1>
<p>This section describes the annotation used in <code>recount</code>.</p>
<p>We used the comprehensive gene annotation for (<code>CHR</code> regions) from <a href="https://www.gencodegenes.org/releases/25.html">Gencode v25</a> (GRCh38.p7) to get the list of genes. Specifically <a href="ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_25/gencode.v25.annotation.gff3.gz">this GFF3</a> file. We then used the <em><a href="https://bioconductor.org/packages/3.12/org.Hs.eg.db">org.Hs.eg.db</a></em> package to get the gene symbols. For each gene, we also counted the total length of exonic base pairs for that given gene and stored this information in the <code>bp_length</code> column. You can see below this information for the <code>rse_gene</code> object included in <em><a href="https://bioconductor.org/packages/3.12/recount">recount</a></em>. The <code><a href="https://rdrr.io/r/base/colnames.html">rownames()</a></code> are the Gencode <code>gene_id</code>.</p>
<div class="sourceCode" id="cb100"><html><body><pre class="r"><span class="co">## Gene annotation information</span>
<span class="fu">rowRanges</span>(<span class="no">rse_gene_SRP009615</span>)</pre></body></html></div>
<pre><code>## GRanges object with 58037 ranges and 3 metadata columns:
##                      seqnames              ranges strand |            gene_id bp_length          symbol
##                         &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; |        &lt;character&gt; &lt;integer&gt; &lt;CharacterList&gt;
##   ENSG00000000003.14     chrX 100627109-100639991      - | ENSG00000000003.14      4535          TSPAN6
##    ENSG00000000005.5     chrX 100584802-100599885      + |  ENSG00000000005.5      1610            TNMD
##   ENSG00000000419.12    chr20   50934867-50958555      - | ENSG00000000419.12      1207            DPM1
##   ENSG00000000457.13     chr1 169849631-169894267      - | ENSG00000000457.13      6883           SCYL3
##   ENSG00000000460.16     chr1 169662007-169854080      + | ENSG00000000460.16      5967        C1orf112
##                  ...      ...                 ...    ... .                ...       ...             ...
##    ENSG00000283695.1    chr19   52865369-52865429      - |  ENSG00000283695.1        61              NA
##    ENSG00000283696.1     chr1 161399409-161422424      + |  ENSG00000283696.1       997              NA
##    ENSG00000283697.1     chrX 149548210-149549852      - |  ENSG00000283697.1      1184           HSFX3
##    ENSG00000283698.1     chr2 112439312-112469687      - |  ENSG00000283698.1       940              NA
##    ENSG00000283699.1    chr10   12653138-12653197      - |  ENSG00000283699.1        60         MIR4481
##   -------
##   seqinfo: 25 sequences (1 circular) from an unspecified genome; no seqlengths</code></pre>
<div class="sourceCode" id="cb102"><html><body><pre class="r"><span class="co">## Also accessible via</span>
<span class="fu">rowData</span>(<span class="no">rse_gene_SRP009615</span>)</pre></body></html></div>
<pre><code>## DataFrame with 58037 rows and 3 columns
##                               gene_id bp_length          symbol
##                           &lt;character&gt; &lt;integer&gt; &lt;CharacterList&gt;
## ENSG00000000003.14 ENSG00000000003.14      4535          TSPAN6
## ENSG00000000005.5   ENSG00000000005.5      1610            TNMD
## ENSG00000000419.12 ENSG00000000419.12      1207            DPM1
## ENSG00000000457.13 ENSG00000000457.13      6883           SCYL3
## ENSG00000000460.16 ENSG00000000460.16      5967        C1orf112
## ...                               ...       ...             ...
## ENSG00000283695.1   ENSG00000283695.1        61              NA
## ENSG00000283696.1   ENSG00000283696.1       997              NA
## ENSG00000283697.1   ENSG00000283697.1      1184           HSFX3
## ENSG00000283698.1   ENSG00000283698.1       940              NA
## ENSG00000283699.1   ENSG00000283699.1        60         MIR4481</code></pre>
<p>For an <code>rse_exon</code> object, the <code><a href="https://rdrr.io/r/base/colnames.html">rownames()</a></code> correspond to the Gencode <code>gene_id</code>. You can add a <code>gene_id</code> column if you are interested in subsetting the whole <code>rse_exon</code> object.</p>
<div class="sourceCode" id="cb104"><html><body><pre class="r"><span class="co">## Get the rse_exon object for study SRP009615</span>
<span class="fu"><a href="../reference/download_study.html">download_study</a></span>(<span class="st">"SRP009615"</span>, <span class="kw">type</span> <span class="kw">=</span> <span class="st">"rse-exon"</span>)</pre></body></html></div>
<pre><code>## 2020-05-13 16:52:46 downloading file rse_exon.Rdata to SRP009615</code></pre>
<div class="sourceCode" id="cb106"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span>(<span class="st">"SRP009615"</span>, <span class="st">"rse_exon.Rdata"</span>))

<span class="co">## Delete it if you don't need it anymore</span>
<span class="fu"><a href="https://rdrr.io/r/base/unlink.html">unlink</a></span>(<span class="st">"SRP009615"</span>, <span class="kw">recursive</span> <span class="kw">=</span> <span class="fl">TRUE</span>)

<span class="co">## Annotation information</span>
<span class="fu">rowRanges</span>(<span class="no">rse_exon</span>)</pre></body></html></div>
<pre><code>## GRanges object with 603139 ranges and 0 metadata columns:
##                      seqnames              ranges strand
##                         &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt;
##   ENSG00000000003.14     chrX 100627109-100628669      -
##   ENSG00000000003.14     chrX 100628670-100629986      -
##   ENSG00000000003.14     chrX 100630759-100630866      -
##   ENSG00000000003.14     chrX 100632063-100632068      -
##   ENSG00000000003.14     chrX 100632485-100632540      -
##                  ...      ...                 ...    ...
##    ENSG00000283698.1     chr2 112440396-112440601      -
##    ENSG00000283698.1     chr2 112440602-112440611      -
##    ENSG00000283698.1     chr2 112463991-112464036      -
##    ENSG00000283698.1     chr2 112469548-112469687      -
##    ENSG00000283699.1    chr10   12653138-12653197      -
##   -------
##   seqinfo: 25 sequences (1 circular) from an unspecified genome; no seqlengths</code></pre>
<div class="sourceCode" id="cb108"><html><body><pre class="r"><span class="co">## Add a gene_id column</span>
<span class="fu">rowRanges</span>(<span class="no">rse_exon</span>)$<span class="no">gene_id</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">rse_exon</span>)

<span class="co">## Example subset</span>
<span class="no">rse_exon_subset</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span>(<span class="no">rse_exon</span>, <span class="kw">subset</span> <span class="kw">=</span> <span class="no">gene_id</span> <span class="kw">==</span> <span class="st">"ENSG00000000003.14"</span>)
<span class="fu">rowRanges</span>(<span class="no">rse_exon_subset</span>)</pre></body></html></div>
<pre><code>## GRanges object with 17 ranges and 1 metadata column:
##                      seqnames              ranges strand |            gene_id
##                         &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; |        &lt;character&gt;
##   ENSG00000000003.14     chrX 100627109-100628669      - | ENSG00000000003.14
##   ENSG00000000003.14     chrX 100628670-100629986      - | ENSG00000000003.14
##   ENSG00000000003.14     chrX 100630759-100630866      - | ENSG00000000003.14
##   ENSG00000000003.14     chrX 100632063-100632068      - | ENSG00000000003.14
##   ENSG00000000003.14     chrX 100632485-100632540      - | ENSG00000000003.14
##                  ...      ...                 ...    ... .                ...
##   ENSG00000000003.14     chrX 100636608-100636689      - | ENSG00000000003.14
##   ENSG00000000003.14     chrX 100636690-100636792      - | ENSG00000000003.14
##   ENSG00000000003.14     chrX 100636793-100636806      - | ENSG00000000003.14
##   ENSG00000000003.14     chrX 100636807-100637104      - | ENSG00000000003.14
##   ENSG00000000003.14     chrX 100639945-100639991      - | ENSG00000000003.14
##   -------
##   seqinfo: 25 sequences (1 circular) from an unspecified genome; no seqlengths</code></pre>
<p>The exons in a <code>rse_exon</code> object are those that have a Gencode <code>gene_id</code> and that were disjointed as described in the <em><a href="https://bioconductor.org/packages/3.12/GenomicRanges">GenomicRanges</a></em> package:</p>
<pre><code><a href="https://rdrr.io/pkg/GenomicRanges/man/inter-range-methods.html">?`inter-range-methods`</a></code></pre>
<p>You can reproduce the gene and exon information using the function <code><a href="../reference/reproduce_ranges.html">reproduce_ranges()</a></code>.</p>
<div id="using-anothernewer-annotation" class="section level2">
<h2 class="hasAnchor">
<a href="#using-anothernewer-annotation" class="anchor"></a>Using another/newer annotation</h2>
<p>If you are interested in using another annotation based on hg38 coordinates you can do so using the function <code><a href="../reference/coverage_matrix.html">coverage_matrix()</a></code> or by running <a href="https://github.com/CRG-Barcelona/bwtool/wiki">bwtool</a> thanks to the BigWig files in <code>recount</code>, in which case <a href="https://github.com/LieberInstitute/recount.bwtool">recount.bwtool</a> will be helpful.</p>
<p>If you are re-processing a small set of samples, it simply might be easier to use <code><a href="../reference/coverage_matrix.html">coverage_matrix()</a></code> as shown below using the annotation information provided by the <em><a href="https://bioconductor.org/packages/3.12/EnsDb.Hsapiens.v79">EnsDb.Hsapiens.v79</a></em> package.</p>
<div class="sourceCode" id="cb111"><html><body><pre class="r"><span class="co">## Get the disjoint exons based on EnsDb.Hsapiens.v79 which matches hg38</span>
<span class="no">exons</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/reproduce_ranges.html">reproduce_ranges</a></span>(<span class="st">"exon"</span>, <span class="kw">db</span> <span class="kw">=</span> <span class="st">"EnsDb.Hsapiens.v79"</span>)

<span class="co">## Change the chromosome names to match those used in the BigWig files</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"GenomeInfoDb"</span>)
<span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqlevelsStyle.html">seqlevelsStyle</a></span>(<span class="no">exons</span>) <span class="kw">&lt;-</span> <span class="st">"UCSC"</span>

<span class="co">## Get the count matrix at the exon level for chrY</span>
<span class="no">exons_chrY</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqlevels-wrappers.html">keepSeqlevels</a></span>(<span class="no">exons</span>, <span class="st">"chrY"</span>, <span class="kw">pruning.mode</span> <span class="kw">=</span> <span class="st">"coarse"</span>)
<span class="no">exonRSE</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/coverage_matrix.html">coverage_matrix</a></span>(<span class="st">"SRP009615"</span>, <span class="st">"chrY"</span>, <span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span>(<span class="no">exons_chrY</span>),
    <span class="kw">chunksize</span> <span class="kw">=</span> <span class="fl">3000</span>
)</pre></body></html></div>
<pre><code>## 2020-05-13 16:53:03 railMatrix: processing regions 1 to 2562</code></pre>
<pre><code>## 2020-05-13 16:53:03 railMatrix: processing file http://duffel.rail.bio/recount/SRP009615/bw/SRR387777.bw</code></pre>
<pre><code>## 2020-05-13 16:53:04 railMatrix: processing file http://duffel.rail.bio/recount/SRP009615/bw/SRR387778.bw</code></pre>
<pre><code>## 2020-05-13 16:53:05 railMatrix: processing file http://duffel.rail.bio/recount/SRP009615/bw/SRR387779.bw</code></pre>
<pre><code>## 2020-05-13 16:53:06 railMatrix: processing file http://duffel.rail.bio/recount/SRP009615/bw/SRR387780.bw</code></pre>
<pre><code>## 2020-05-13 16:53:08 railMatrix: processing file http://duffel.rail.bio/recount/SRP009615/bw/SRR389077.bw</code></pre>
<pre><code>## 2020-05-13 16:53:09 railMatrix: processing file http://duffel.rail.bio/recount/SRP009615/bw/SRR389078.bw</code></pre>
<pre><code>## 2020-05-13 16:53:10 railMatrix: processing file http://duffel.rail.bio/recount/SRP009615/bw/SRR389079.bw</code></pre>
<pre><code>## 2020-05-13 16:53:11 railMatrix: processing file http://duffel.rail.bio/recount/SRP009615/bw/SRR389080.bw</code></pre>
<pre><code>## 2020-05-13 16:53:12 railMatrix: processing file http://duffel.rail.bio/recount/SRP009615/bw/SRR389081.bw</code></pre>
<pre><code>## 2020-05-13 16:53:13 railMatrix: processing file http://duffel.rail.bio/recount/SRP009615/bw/SRR389082.bw</code></pre>
<pre><code>## 2020-05-13 16:53:14 railMatrix: processing file http://duffel.rail.bio/recount/SRP009615/bw/SRR389083.bw</code></pre>
<pre><code>## 2020-05-13 16:53:14 railMatrix: processing file http://duffel.rail.bio/recount/SRP009615/bw/SRR389084.bw</code></pre>
<div class="sourceCode" id="cb125"><html><body><pre class="r"><span class="no">exonMatrix</span> <span class="kw">&lt;-</span> <span class="fu">assays</span>(<span class="no">exonRSE</span>)$<span class="no">counts</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span>(<span class="no">exonMatrix</span>)</pre></body></html></div>
<pre><code>## [1] 2562   12</code></pre>
<div class="sourceCode" id="cb127"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="no">exonMatrix</span>)</pre></body></html></div>
<pre><code>##   SRR387777 SRR387778 SRR387779 SRR387780 SRR389077 SRR389078 SRR389079 SRR389080 SRR389081 SRR389082 SRR389083
## 1         0         0         0         0         0         0 0.0000000 0.0000000         0         0         0
## 2         0         0         0         0         0         0 0.0000000 0.0000000         0         0         0
## 3         0         0         0         0         0         0 0.0000000 0.0000000         0         0         0
## 4         0         0         0         0         0         0 0.0000000 0.0000000         0         0         0
## 5         0         0         0         0         0         0 0.0000000 0.6875641         0         0         0
## 6         0         0         0         0         0         0 0.3256991 0.0000000         0         0         0
##   SRR389084
## 1         0
## 2         0
## 3         0
## 4         0
## 5         0
## 6         0</code></pre>
<div class="sourceCode" id="cb129"><html><body><pre class="r"><span class="co">## Summary the information at the gene level for chrY</span>
<span class="no">exon_gene</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="no">exons_chrY</span>), <span class="fu">elementNROWS</span>(<span class="no">exons_chrY</span>))
<span class="no">geneMatrix</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/do.call.html">do.call</a></span>(<span class="no">rbind</span>, <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/split.html">split</a></span>(
    <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span>(<span class="no">exonMatrix</span>),
    <span class="no">exon_gene</span>
), <span class="no">colSums</span>))
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span>(<span class="no">geneMatrix</span>)</pre></body></html></div>
<pre><code>## [1] 544  12</code></pre>
<div class="sourceCode" id="cb131"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="no">geneMatrix</span>)</pre></body></html></div>
<pre><code>##                 SRR387777 SRR387778  SRR387779  SRR387780 SRR389077 SRR389078  SRR389079  SRR389080 SRR389081
## ENSG00000012817  9.791214 17.655826 17.8671773  6.7090447  4.832332  2.893332 12.7255290  8.1132560 16.988920
## ENSG00000067048 77.902314 69.981889 41.6900803 83.0430650 52.797699 52.000713 52.3677621 53.9737791 80.441512
## ENSG00000067646 44.371293 52.123510 22.6257722 57.2505152 44.251631 44.232314 31.5928125 56.0708495 38.480927
## ENSG00000092377 10.568294  9.722520 13.8867173 16.0271624 11.364928  1.426849 17.0061457 24.2022551 33.773154
## ENSG00000099715  3.613424  1.789214  0.6284937  2.6836179  2.997835  0.000000  0.8375120  0.0000000  1.023429
## ENSG00000099721  0.000000  0.000000  0.0000000  0.7827219  1.342314  0.000000  0.4652844  0.7563205  0.000000
##                  SRR389082 SRR389083 SRR389084
## ENSG00000012817  28.049724  8.830204  9.139425
## ENSG00000067048 106.485062 65.978494 34.679041
## ENSG00000067646  45.905450 35.469641 28.992288
## ENSG00000092377   9.349908  3.571768 16.450965
## ENSG00000099715   2.337477  0.000000  0.000000
## ENSG00000099721   0.000000  0.000000  0.000000</code></pre>
<p>The above solution works well for a small set of samples. However, <code>bwtool</code> is faster for processing large sets. You can see how we used <code>bwtool</code> at <a href="https://github.com/leekgroup/recount-website">leekgroup/recount-website</a>. It’s much faster than R and the small package <a href="https://github.com/LieberInstitute/recount.bwtool">recount.bwtool</a> will be helpful for such scenarios. Note that you will need to run <code><a href="../reference/scale_counts.html">scale_counts()</a></code> after using <code>coverage_matrix_bwtool()</code> and that will have to download the BigWig files first unless you are working at <a href="https://jhpce.jhu.edu/">JHPCE</a> or <a href="http://www.sciserver.org/">SciServer</a>. <code>coverage_matrix_bwtool()</code> will download the files for you, but that till take time.</p>
<p>Finally, the BigWig files hosted by recount have the following chromosome names and sizes as specified in <a href="https://github.com/nellore/runs/blob/master/gtex/hg38.sizes">hg38.sizes</a>. You’ll have to make sure that you match these names when using alternative annotations.</p>
</div>
</div>
<div id="candidate-gene-fusions" class="section level1">
<h1 class="hasAnchor">
<a href="#candidate-gene-fusions" class="anchor"></a>Candidate gene fusions</h1>
<p>If you are interested in finding possible gene fusion events in a particular study, you can download the <em>RangedSummarizedExperiment</em> object at the exon-exon junctions level for that study. The objects we provide classify exon-exon junction events as shown below.</p>
<div class="sourceCode" id="cb133"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"recount"</span>)

<span class="co">## Download and load RSE object for SRP009615 study</span>
<span class="fu"><a href="../reference/download_study.html">download_study</a></span>(<span class="st">"SRP009615"</span>, <span class="kw">type</span> <span class="kw">=</span> <span class="st">"rse-jx"</span>)</pre></body></html></div>
<pre><code>## 2020-05-13 16:53:16 downloading file rse_jx.Rdata to SRP009615</code></pre>
<div class="sourceCode" id="cb135"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span>(<span class="st">"SRP009615"</span>, <span class="st">"rse_jx.Rdata"</span>))

<span class="co">## Delete it if you don't need it anymore</span>
<span class="fu"><a href="https://rdrr.io/r/base/unlink.html">unlink</a></span>(<span class="st">"SRP009615"</span>, <span class="kw">recursive</span> <span class="kw">=</span> <span class="fl">TRUE</span>)

<span class="co">## Exon-exon junctions by class</span>
<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span>(<span class="fu">rowRanges</span>(<span class="no">rse_jx</span>)$<span class="no">class</span>)</pre></body></html></div>
<pre><code>## 
## alternative_end       annotated       exon_skip          fusion           novel 
##           35601          140251            9513             489           26475</code></pre>
<div class="sourceCode" id="cb137"><html><body><pre class="r"><span class="co">## Potential gene fusions by chromosome</span>
<span class="no">fusions</span> <span class="kw">&lt;-</span> <span class="fu">rowRanges</span>(<span class="no">rse_jx</span>)[<span class="fu">rowRanges</span>(<span class="no">rse_jx</span>)$<span class="no">class</span> <span class="kw">==</span> <span class="st">"fusion"</span>]
<span class="no">fusions_by_chr</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sort.html">sort</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html">seqnames</a></span>(<span class="no">fusions</span>)), <span class="kw">decreasing</span> <span class="kw">=</span> <span class="fl">TRUE</span>)
<span class="no">fusions_by_chr</span>[<span class="no">fusions_by_chr</span> <span class="kw">&gt;</span> <span class="fl">0</span>]</pre></body></html></div>
<pre><code>## 
##  chr1 chr19  chr6 chr12 chr22 chr17  chr5 chr16 chr11  chr3 chr10  chr2  chr7  chrX chr14  chr4  chr9  chr8 chr15 chr20 
##    61    54    37    31    27    26    26    24    22    22    21    20    20    16    13    13    12    10     9     9 
## chr13 chr21 chr18 
##     6     6     4</code></pre>
<div class="sourceCode" id="cb139"><html><body><pre class="r"><span class="co">## Genes with the most fusions</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/sort.html">sort</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span>(<span class="no">fusions</span>$<span class="no">symbol_proposed</span>)), <span class="kw">decreasing</span> <span class="kw">=</span> <span class="fl">TRUE</span>))</pre></body></html></div>
<pre><code>## 
##      TMEM14B       SYCP2L LOC100129924        MATR3        ACSM3         BMI1 
##            6            5            4            4            3            3</code></pre>
<p>If you are interested in checking the frequency of a particular exon-exon junction then check the <code><a href="../reference/snaptron_query.html">snaptron_query()</a></code> function described in the following section.</p>
</div>
<div id="snaptron" class="section level1">
<h1 class="hasAnchor">
<a href="#snaptron" class="anchor"></a>Snaptron</h1>
<p>Our collaborators built <em>SnaptronUI</em> to query the <a href="https://github.com/nellore/intropolis">Intropolis</a> database of the exon-exon junctions found in SRA. <em>SnaptronUI</em> allows you to do several types of queries manually and perform analyses. <em><a href="https://bioconductor.org/packages/3.12/recount">recount</a></em> has the function <code><a href="../reference/snaptron_query.html">snaptron_query()</a></code> which uses <a href="https://github.com/ChristopherWilks/snaptron">Snaptron</a> to check if particular exon-exon junctions are present in <em>Intropolis</em>. For example, here we use <code><a href="../reference/snaptron_query.html">snaptron_query()</a></code> to get in <code>R</code>-friendly format the information for 3 exon-exon junctions using <em>Intropolis</em> version 1 which uses hg19 coordinates. Version 2 uses hg38 coordinates and the exon-exon junctions were derived from a larger data set: about 50,000 samples vs 21,000 in version 1. <code><a href="../reference/snaptron_query.html">snaptron_query()</a></code> can also be used to access the 30 million and 36 million exon-exon junctions derived from the GTEx and TCGA consortiums respectively (about 10 and 11 thousand samples respectively).</p>
<div class="sourceCode" id="cb141"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"GenomicRanges"</span>)
<span class="no">junctions</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html">GRanges</a></span>(<span class="kw">seqnames</span> <span class="kw">=</span> <span class="st">"chr2"</span>, <span class="fu">IRanges</span>(
    <span class="kw">start</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">28971711</span>, <span class="fl">29555082</span>, <span class="fl">29754983</span>),
    <span class="kw">end</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">29462418</span>, <span class="fl">29923339</span>, <span class="fl">29917715</span>)
))

<span class="fu"><a href="../reference/snaptron_query.html">snaptron_query</a></span>(<span class="no">junctions</span>)</pre></body></html></div>
<pre><code>## 2020-05-13 16:53:17 querying Snaptron</code></pre>
<pre><code>## 2020-05-13 16:53:18 processing results</code></pre>
<pre><code>## 2020-05-13 16:53:18 extracting information</code></pre>
<pre><code>## GRanges object with 3 ranges and 14 metadata columns:
##       seqnames            ranges strand |     type snaptron_id       annotated  left_motif right_motif
##          &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;factor&gt;   &lt;integer&gt; &lt;CharacterList&gt; &lt;character&gt; &lt;character&gt;
##   [1]     chr2 28971711-29462418      - |  SRAv1:I    22314441              NA          AT          AC
##   [2]     chr2 29555082-29923339      + |  SRAv1:I    22328842              NA          GC          AG
##   [3]     chr2 29754983-29917715      - |  SRAv1:I    22329293               1          GT          AG
##           left_annotated    right_annotated            samples read_coverage_by_sample samples_count coverage_sum
##          &lt;CharacterList&gt;    &lt;CharacterList&gt;      &lt;IntegerList&gt;           &lt;IntegerList&gt;     &lt;integer&gt;    &lt;integer&gt;
##   [1]                 NA                 NA               3481                       1             1            1
##   [2]                 NA                 NA 2643,4867,4945,...               1,1,1,...            13           13
##   [3] aC19,cG19,cG38,... aC19,cG19,cG38,...         3,5,13,...               1,1,1,...          1625         5380
##       coverage_avg coverage_median source_dataset_id
##          &lt;numeric&gt;       &lt;numeric&gt;         &lt;integer&gt;
##   [1]        1.000               1                 0
##   [2]        1.000               1                 0
##   [3]        3.311               1                 0
##   -------
##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</code></pre>
<p>If you use <code><a href="../reference/snaptron_query.html">snaptron_query()</a></code> please cite <a href="http://snaptron.cs.jhu.edu">snaptron.cs.jhu.edu</a>. <em>Snaptron</em> is separate from <em><a href="https://bioconductor.org/packages/3.12/recount">recount</a></em>. Thank you!</p>
</div>
<div id="fantom-cat-annotation" class="section level1">
<h1 class="hasAnchor">
<a href="#fantom-cat-annotation" class="anchor"></a>FANTOM-CAT annotation</h1>
<p>Thanks to <a href="https://www.biorxiv.org/content/10.1101/659490v1">Imada, Sanchez, et al., bioRxiv, 2019</a> you can now use the <code>recount</code> package to access the FANTOM-CAT annotation that was merged with <code>recount2</code>. You can use the main <code>recount2</code> website or the <code><a href="../reference/download_study.html">recount::download_study()</a></code> function by specifying <code>type = 'rse-fc'</code>. See their manuscript for further details on how this annotation was derived.</p>
<div class="sourceCode" id="cb146"><html><body><pre class="r"><span class="co">## Example use of download_study()</span>
<span class="co">## for downloading the FANTOM-CAT `rse_fc`</span>
<span class="co">## file for a given study</span>
<span class="fu"><a href="../reference/download_study.html">download_study</a></span>(<span class="st">"DRP000366"</span>, <span class="kw">type</span> <span class="kw">=</span> <span class="st">"rse-fc"</span>, <span class="kw">download</span> <span class="kw">=</span> <span class="fl">FALSE</span>)</pre></body></html></div>
</div>
<div id="recount-brain" class="section level1">
<h1 class="hasAnchor">
<a href="#recount-brain" class="anchor"></a><code>recount-brain</code>
</h1>
<p>The <code>recount-brain</code> project by <a href="https://www.biorxiv.org/content/10.1101/618025v1">Ramzara et al., bioRxiv, 2019</a> contains curated metadata for brain studies present in <code>recount2</code> that you can access via <code><a href="../reference/add_metadata.html">recount::add_metadata(source = 'recount_brain_v2')</a></code>. The manuscript by Razmara et al describes a curation process that can be used for curating other tissues or projects. If you are interested in doing such a project and/or contributing curated metadata to the <code>recount</code> package via the <code><a href="../reference/add_metadata.html">add_metadata()</a></code> function please let us know.</p>
<div class="sourceCode" id="cb147"><html><body><pre class="r"><span class="co">## recount-brain citation details</span>
<span class="fu"><a href="https://rdrr.io/r/utils/citation.html">citation</a></span>(<span class="st">"recount"</span>)[<span class="fl">5</span>]</pre></body></html></div>
<pre><code>## 
## Frazee AC, Langmead B, Leek JT (2011). "ReCount: A multi-experiment resource of analysis-ready RNA-seq gene
## count datasets." _BMC Bioinformatics_. doi: 10.1186/1471-2105-12-449 (URL:
## https://doi.org/10.1186/1471-2105-12-449), &lt;URL: https://doi.org/10.1186/1471-2105-12-449&gt;.
## 
## A BibTeX entry for LaTeX users is
## 
##   @Article{,
##     title = {ReCount: A multi-experiment resource of analysis-ready RNA-seq gene count datasets},
##     author = {Alyssa C. Frazee and Ben Langmead and Jeffrey T. Leek},
##     year = {2011},
##     journal = {BMC Bioinformatics},
##     doi = {10.1186/1471-2105-12-449},
##     url = {https://doi.org/10.1186/1471-2105-12-449},
##   }</code></pre>
</div>
<div id="download-all-the-data" class="section level1">
<h1 class="hasAnchor">
<a href="#download-all-the-data" class="anchor"></a>Download all the data</h1>
<p>If you are interested in downloading all the data you can do so using the <code>recount_url</code> data.frame included in the package. That is:</p>
<div class="sourceCode" id="cb149"><html><body><pre class="r"><span class="co">## Get data.frame with all the URLs</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"recount"</span>)
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span>(<span class="no">recount_url</span>)</pre></body></html></div>
<pre><code>## [1] 97229     6</code></pre>
<div class="sourceCode" id="cb151"><html><body><pre class="r"><span class="co">## Explore URLs</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="no">recount_url</span>$<span class="no">url</span>)</pre></body></html></div>
<pre><code>## [1] "http://duffel.rail.bio/recount/v2/DRP000366/counts_exon.tsv.gz"
## [2] "http://duffel.rail.bio/recount/v2/DRP000366/counts_gene.tsv.gz"
## [3] "http://duffel.rail.bio/recount/v2/DRP000366/rse_exon.Rdata"    
## [4] "http://duffel.rail.bio/recount/v2/DRP000366/rse_gene.Rdata"    
## [5] "http://duffel.rail.bio/recount/DRP000366/counts_jx.tsv.gz"     
## [6] "http://duffel.rail.bio/recount/DRP000366/rse_jx.Rdata"</code></pre>
<p>You can then download all the data using the <code><a href="../reference/download_study.html">download_study()</a></code> function or however you prefer to do so with <code>recount_url</code>.</p>
<div class="sourceCode" id="cb153"><html><body><pre class="r"><span class="co">## Download all files (will take a while! It's over 8 TB of data)</span>
<span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span>(<span class="no">recount_url</span>$<span class="no">project</span>), <span class="no">download_study</span>, <span class="kw">type</span> <span class="kw">=</span> <span class="st">"all"</span>)</pre></body></html></div>
<p>You can find the code for the website at <a href="https://github.com/leekgroup/recount-website/tree/master/website">leekgroup/recount-website</a> if you want to deploy your own local mirror. Please let us know if you choose to do deploy a mirror.</p>
</div>
<div id="accessing-recount-via-sciserver" class="section level1">
<h1 class="hasAnchor">
<a href="#accessing-recount-via-sciserver" class="anchor"></a>Accessing <em><a href="https://bioconductor.org/packages/3.12/recount">recount</a></em> via <em>SciServer</em>
</h1>
<p>Not everyone has over 8 terabytes of disk space available to download all the data from the <code>recount</code> project. However, thanks to <a href="http://www.sciserver.org/">SciServer</a> you can access it locally via a Jupyter Notebook. If you do so and want to share your work, please let the <a href="http://www.sciserver.org/">SciServer</a> maintainers know via Twitter at <a href="https://twitter.com/IDIESJHU">IDIESJHU</a>.</p>
<div id="step-by-step" class="section level2">
<h2 class="hasAnchor">
<a href="#step-by-step" class="anchor"></a>Step by step</h2>
<p>To use <em>SciServer</em>, you first have to go to <a href="http://www.sciserver.org/" class="uri">http://www.sciserver.org/</a> then click on “Login to SciServer”. If you are a first time user, click on “Register New Account” and enter the information they request as shown on Figure @ref(fig:Figure1).</p>
<div class="figure" style="text-align: center">
<img src="https://raw.githubusercontent.com/leekgroup/recount-website/master/sciserver_demo/register.png" alt="SIgn up to SciServer" width="100%"><p class="caption">
SIgn up to SciServer
</p>
</div>
<p>Once you’ve registered, open the <a href="http://compute.sciserver.org/dashboard/">SciServer compute tool</a> (see Figure @ref(fig:Figure2)).</p>
<div class="figure" style="text-align: center">
<img src="https://raw.githubusercontent.com/leekgroup/recount-website/master/sciserver_demo/sciserver-compute.png" alt="Click on SciServer Compute," width="100%"><p class="caption">
Click on SciServer Compute,
</p>
</div>
<p>Then click on “Create container”, choose a name of your preference and make sure that you choose to load R 3.3.x and the <code>recount</code> public volume (see Figure @ref(fig:Figure3)). This will enable you to access all of the <code>recount</code> data as if it was on your computer. Click “Create” once you are ready.</p>
<div class="figure" style="text-align: center">
<img src="https://raw.githubusercontent.com/leekgroup/recount-website/master/sciserver_demo/new-container.png" alt="Select the appropriate image and load the recount public volume." width="100%"><p class="caption">
Select the appropriate image and load the recount public volume.
</p>
</div>
<p>Once you have a container, create a new R <em>Notebook</em> as shown in Figure @ref(fig:Figure4).</p>
<div class="figure" style="text-align: center">
<img src="https://raw.githubusercontent.com/leekgroup/recount-website/master/sciserver_demo/new-R.png" alt="Create a R notebook." width="100%"><p class="caption">
Create a R notebook.
</p>
</div>
<p>In this R <em>Notebook</em> you can insert new code <em>cells</em> where you can type R code. For example, you can install <em><a href="https://bioconductor.org/packages/3.12/recount">recount</a></em> and <em><a href="https://bioconductor.org/packages/3.12/DESeq2">DESeq2</a></em> to run the code example from this vignette. You first have to install the packages as shown in Figure @ref(fig:Figure5).</p>
<div class="figure" style="text-align: center">
<img src="https://raw.githubusercontent.com/leekgroup/recount-website/master/sciserver_demo/demo1.png" alt="Install recount and DESeq2 on SciServer." width="100%"><p class="caption">
Install recount and DESeq2 on SciServer.
</p>
</div>
<p>Then, the main part when using <em>SciServer</em> is to use to remember that all the <code>recount</code> data is available locally. So instead of using <code><a href="../reference/download_study.html">download_study()</a></code>, you can simply use <code><a href="https://rdrr.io/r/base/load.html">load()</a></code> with the correct path as shown in Figure @ref(fig:Figure6).</p>
<div class="figure" style="text-align: center">
<img src="https://raw.githubusercontent.com/leekgroup/recount-website/master/sciserver_demo/demo2.png" alt="recount files are available locally so remmeber to use the local data when working on SciServer." width="100%"><p class="caption">
recount files are available locally so remmeber to use the local data when working on SciServer.
</p>
</div>
<p>Several functions in the <em><a href="https://bioconductor.org/packages/3.12/recount">recount</a></em> package have the <code>outdir</code> argument, which you can specify to use the data locally. Try using <code><a href="../reference/expressed_regions.html">expressed_regions()</a></code> in <em>SciServer</em> to get started with the following code:</p>
<div class="sourceCode" id="cb154"><html><body><pre class="r"><span class="co">## Expressed-regions SciServer example</span>
<span class="no">regions</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/expressed_regions.html">expressed_regions</a></span>(<span class="st">"SRP009615"</span>, <span class="st">"chrY"</span>,
    <span class="kw">cutoff</span> <span class="kw">=</span> <span class="fl">5L</span>,
    <span class="kw">maxClusterGap</span> <span class="kw">=</span> <span class="fl">3000L</span>, <span class="kw">outdir</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span>(<span class="no">scipath</span>, <span class="st">"SRP009615"</span>)
)
<span class="no">regions</span></pre></body></html></div>
<p>You can find the R code for a full <em>SciServer</em> demo <a href="https://github.com/leekgroup/recount-website/blob/master/sciserver_demo/sciserver_demo.R">here</a>. You can copy and paste it into a cell and run it all to see how <em>SciServer compute</em> works.</p>
</div>
<div id="sciserver-help" class="section level2">
<h2 class="hasAnchor">
<a href="#sciserver-help" class="anchor"></a>SciServer help</h2>
<p>If you encounter problems when using SciServer please check their <a href="http://www.sciserver.org/support/help/">support page</a> and the <a href="http://compute.sciserver.org/dashboard/Home/Help">SciServer compute help section</a>.</p>
</div>
<div id="citing-sciserver" class="section level2">
<h2 class="hasAnchor">
<a href="#citing-sciserver" class="anchor"></a>Citing <em>SciServer</em>
</h2>
<p>If you use <a href="http://www.sciserver.org/">SciServer</a> for your published work, please cite it accordingly. <em>SciServer</em> is administered by the <a href="http://idies.jhu.edu/">Institute for Data Intensive Engineering and Science</a> at <a href="http://www.jhu.edu/">Johns Hopkins University</a> and is funded by National Science Foundation award ACI-1261715.</p>
</div>
</div>
<div id="reproducibility" class="section level1">
<h1 class="hasAnchor">
<a href="#reproducibility" class="anchor"></a>Reproducibility</h1>
<p>The <em><a href="https://bioconductor.org/packages/3.12/recount">recount</a></em> package (<a href="http://www.nature.com/nbt/journal/v35/n4/full/nbt.3838.html">Collado-Torres, Nellore, Kammers, Ellis, et al., 2017</a>) was made possible thanks to:</p>
<ul>
<li>R <a id="cite-R_Core_Team_2020"></a>(<a href="https://www.R-project.org/">R Core Team, 2020</a>)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.12/AnnotationDbi">AnnotationDbi</a></em> <a id="cite-AnnotationDbi"></a>(<a href="https://doi.org/10.18129/B9.bioc.AnnotationDbi">Pagès, Carlson, Falcon, and Li, 2017</a>)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.12/BiocParallel">BiocParallel</a></em> <a id="cite-Morgan_2020"></a>(<a href="https://github.com/Bioconductor/BiocParallel">Morgan, Obenchain, Lang, Thompson, et al., 2020</a>)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.12/BiocStyle">BiocStyle</a></em> <a id="cite-Oles_2020"></a>(<a href="https://github.com/Bioconductor/BiocStyle">Oleś, Morgan, and Huber, 2020</a>)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.12/derfinder">derfinder</a></em> (<a href="http://nar.oxfordjournals.org/content/early/2016/09/29/nar.gkw852">Collado-Torres, Nellore, Frazee, Wilks, et al., 2017</a>)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=sessioninfo">sessioninfo</a></em> <a id="cite-Csardi_2018"></a>(<a href="https://CRAN.R-project.org/package=sessioninfo">Csárdi, core, Wickham, Chang, et al., 2018</a>)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=downloader">downloader</a></em> <a id="cite-Chang_2015"></a>(<a href="https://CRAN.R-project.org/package=downloader">Chang, 2015</a>)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.12/EnsDb.Hsapiens.v79">EnsDb.Hsapiens.v79</a></em> <a id="cite-Rainer_2017"></a>(<a href="#bib-Rainer_2017">Rainer, 2017</a>)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.12/GEOquery">GEOquery</a></em> <a id="cite-Davis_2007"></a>(<a href="#bib-Davis_2007">Davis and Meltzer, 2007</a>)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.12/GenomeInfoDb">GenomeInfoDb</a></em> <a id="cite-GenomeInfoDb"></a>(<a href="https://doi.org/10.18129/B9.bioc.GenomeInfoDb">Arora, Morgan, Carlson, and Pagès, 2017</a>)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.12/GenomicFeatures">GenomicFeatures</a></em> <a id="cite-Lawrence_2013"></a>(<a href="http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003118">Lawrence, Huber, Pagès, Aboyoun, et al., 2013</a>)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.12/GenomicRanges">GenomicRanges</a></em> <a id="cite-Lawrence_2013a"></a>(<a href="http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003118">Lawrence, Huber, Pagès, Aboyoun, et al., 2013</a>)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=knitcitations">knitcitations</a></em> (<a href="https://CRAN.R-project.org/package=knitcitations">Boettiger, 2019</a>)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=knitr">knitr</a></em> (<a href="http://www.crcpress.com/product/isbn/9781466561595">Xie, 2014</a>)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.12/org.Hs.eg.db">org.Hs.eg.db</a></em> <a id="cite-Carlson_2020"></a>(<a href="#bib-Carlson_2020">Carlson, 2020</a>)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=RCurl">RCurl</a></em> <a id="cite-Temple_Lang_2020"></a>(<a href="https://CRAN.R-project.org/package=RCurl">Temple Lang, 2020</a>)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=rentrez">rentrez</a></em> <a id="cite-Winter_2017"></a>(<a href="#bib-Winter_2017">Winter, 2017</a>)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.12/rtracklayer">rtracklayer</a></em> <a id="cite-Lawrence_2009"></a>(<a href="http://bioinformatics.oxfordjournals.org/content/25/14/1841.abstract">Lawrence, Gentleman, and Carey, 2009</a>)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=rmarkdown">rmarkdown</a></em> (<a href="https://github.com/rstudio/rmarkdown">Allaire, Xie, McPherson, Luraschi, et al., 2020</a>)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.12/S4Vectors">S4Vectors</a></em> <a id="cite-S4Vectors"></a>(<a href="https://doi.org/10.18129/B9.bioc.S4Vectors">Pagès, Lawrence, and Aboyoun, 2017</a>)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.12/SummarizedExperiment">SummarizedExperiment</a></em> (<a href="https://doi.org/10.18129/B9.bioc.SummarizedExperiment">Morgan, Obenchain, Hester, and Pagès, 2017</a>)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=testthat">testthat</a></em> <a id="cite-Wickham_2011"></a>(<a href="https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf">Wickham, 2011</a>)</li>
</ul>
<p>Code for creating the vignette</p>
<div class="sourceCode" id="cb155"><html><body><pre class="r"><span class="co">## Create the vignette</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"rmarkdown"</span>)
<span class="fu"><a href="https://rdrr.io/r/base/system.time.html">system.time</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/rmarkdown/man/render.html">render</a></span>(<span class="st">"recount-quickstart.Rmd"</span>, <span class="st">"BiocStyle::html_document"</span>))

<span class="co">## Extract the R code</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"knitr"</span>)
<span class="fu"><a href="https://rdrr.io/pkg/knitr/man/knit.html">knit</a></span>(<span class="st">"recount-quickstart.Rmd"</span>, <span class="kw">tangle</span> <span class="kw">=</span> <span class="fl">TRUE</span>)</pre></body></html></div>
<div class="sourceCode" id="cb156"><html><body><pre class="r"><span class="co">## Clean up</span>
<span class="fu"><a href="https://rdrr.io/r/base/files.html">file.remove</a></span>(<span class="st">"quickstartRef.bib"</span>)</pre></body></html></div>
<pre><code>## [1] TRUE</code></pre>
<p>Date the vignette was generated.</p>
<pre><code>## [1] "2020-05-13 16:53:19 UTC"</code></pre>
<p>Wallclock time spent generating the vignette.</p>
<pre><code>## Time difference of 2.078 mins</code></pre>
<p><code>R</code> session information.</p>
<pre><code>## ─ Session info ───────────────────────────────────────────────────────────────────────────────────────────────────────
##  setting  value                                      
##  version  R version 4.0.0 Patched (2020-05-07 r78385)
##  os       Ubuntu 18.04.4 LTS                         
##  system   x86_64, linux-gnu                          
##  ui       X11                                        
##  language (EN)                                       
##  collate  en_US.UTF-8                                
##  ctype    en_US.UTF-8                                
##  tz       UTC                                        
##  date     2020-05-13                                 
## 
## ─ Packages ───────────────────────────────────────────────────────────────────────────────────────────────────────────
##  package              * version    date       lib source                        
##  acepack                1.4.1      2016-10-29 [1] CRAN (R 4.0.0)                
##  annotate               1.67.0     2020-04-27 [1] Bioconductor                  
##  AnnotationDbi        * 1.51.0     2020-04-27 [1] Bioconductor                  
##  AnnotationFilter       1.13.0     2020-04-27 [1] Bioconductor                  
##  askpass                1.1        2019-01-13 [2] CRAN (R 4.0.0)                
##  assertthat             0.2.1      2019-03-21 [2] CRAN (R 4.0.0)                
##  backports              1.1.6      2020-04-05 [2] CRAN (R 4.0.0)                
##  base64enc              0.1-3      2015-07-28 [1] CRAN (R 4.0.0)                
##  bibtex                 0.4.2.2    2020-01-02 [1] CRAN (R 4.0.0)                
##  Biobase              * 2.49.0     2020-04-27 [1] Bioconductor                  
##  BiocFileCache          1.13.0     2020-04-27 [1] Bioconductor                  
##  BiocGenerics         * 0.35.2     2020-05-06 [1] Bioconductor                  
##  BiocManager            1.30.10    2019-11-16 [2] CRAN (R 4.0.0)                
##  BiocParallel           1.23.0     2020-04-27 [1] Bioconductor                  
##  BiocStyle            * 2.17.0     2020-04-27 [1] Bioconductor                  
##  biomaRt                2.45.0     2020-04-27 [1] Bioconductor                  
##  Biostrings             2.57.0     2020-04-27 [1] Bioconductor                  
##  bit                    1.1-15.2   2020-02-10 [1] CRAN (R 4.0.0)                
##  bit64                  0.9-7      2017-05-08 [1] CRAN (R 4.0.0)                
##  bitops                 1.0-6      2013-08-17 [1] CRAN (R 4.0.0)                
##  blob                   1.2.1      2020-01-20 [1] CRAN (R 4.0.0)                
##  bookdown               0.18       2020-03-05 [1] CRAN (R 4.0.0)                
##  BSgenome               1.57.0     2020-04-27 [1] Bioconductor                  
##  bumphunter             1.31.0     2020-04-28 [1] Bioconductor                  
##  checkmate              2.0.0      2020-02-06 [1] CRAN (R 4.0.0)                
##  cli                    2.0.2      2020-02-28 [2] CRAN (R 4.0.0)                
##  cluster                2.1.0      2019-06-19 [3] CRAN (R 4.0.0)                
##  codetools              0.2-16     2018-12-24 [3] CRAN (R 4.0.0)                
##  colorspace             1.4-1      2019-03-18 [1] CRAN (R 4.0.0)                
##  crayon                 1.3.4      2017-09-16 [2] CRAN (R 4.0.0)                
##  crosstalk              1.1.0.1    2020-03-13 [2] CRAN (R 4.0.0)                
##  curl                   4.3        2019-12-02 [2] CRAN (R 4.0.0)                
##  data.table             1.12.8     2019-12-09 [1] CRAN (R 4.0.0)                
##  DBI                    1.1.0      2019-12-15 [1] CRAN (R 4.0.0)                
##  dbplyr                 1.4.3      2020-04-19 [1] CRAN (R 4.0.0)                
##  DEFormats              1.17.0     2020-04-27 [1] Bioconductor                  
##  DelayedArray         * 0.15.1     2020-05-01 [1] Bioconductor                  
##  derfinder              1.23.1     2020-05-10 [1] Bioconductor                  
##  derfinderHelper        1.23.1     2020-05-10 [1] Bioconductor                  
##  desc                   1.2.0      2018-05-01 [2] CRAN (R 4.0.0)                
##  DESeq2               * 1.29.0     2020-04-27 [1] Bioconductor                  
##  digest                 0.6.25     2020-02-23 [2] CRAN (R 4.0.0)                
##  doRNG                  1.8.2      2020-01-27 [1] CRAN (R 4.0.0)                
##  downloader             0.4        2015-07-09 [1] CRAN (R 4.0.0)                
##  dplyr                  0.8.5      2020-03-07 [1] CRAN (R 4.0.0)                
##  DT                   * 0.13       2020-03-23 [2] CRAN (R 4.0.0)                
##  edgeR                  3.31.0     2020-04-27 [1] Bioconductor                  
##  ellipsis               0.3.0      2019-09-20 [2] CRAN (R 4.0.0)                
##  EnsDb.Hsapiens.v79     2.99.0     2020-05-13 [1] Bioconductor                  
##  ensembldb              2.13.1     2020-05-06 [1] Bioconductor                  
##  evaluate               0.14       2019-05-28 [2] CRAN (R 4.0.0)                
##  fansi                  0.4.1      2020-01-08 [2] CRAN (R 4.0.0)                
##  farver                 2.0.3      2020-01-16 [1] CRAN (R 4.0.0)                
##  foreach                1.5.0      2020-03-30 [1] CRAN (R 4.0.0)                
##  foreign                0.8-79     2020-04-26 [3] CRAN (R 4.0.0)                
##  Formula                1.2-3      2018-05-03 [1] CRAN (R 4.0.0)                
##  fs                     1.4.1      2020-04-04 [2] CRAN (R 4.0.0)                
##  genefilter             1.71.0     2020-04-27 [1] Bioconductor                  
##  geneplotter            1.67.0     2020-04-27 [1] Bioconductor                  
##  generics               0.0.2      2018-11-29 [1] CRAN (R 4.0.0)                
##  GenomeInfoDb         * 1.25.0     2020-04-27 [1] Bioconductor                  
##  GenomeInfoDbData       1.2.3      2020-05-13 [1] Bioconductor                  
##  GenomicAlignments      1.25.0     2020-04-27 [1] Bioconductor                  
##  GenomicFeatures        1.41.0     2020-04-27 [1] Bioconductor                  
##  GenomicFiles           1.25.0     2020-04-27 [1] Bioconductor                  
##  GenomicRanges        * 1.41.1     2020-05-02 [1] Bioconductor                  
##  GEOquery               2.57.0     2020-04-27 [1] Bioconductor                  
##  ggplot2              * 3.3.0      2020-03-05 [1] CRAN (R 4.0.0)                
##  glue                   1.4.1      2020-05-13 [1] CRAN (R 4.0.0)                
##  gridExtra              2.3        2017-09-09 [1] CRAN (R 4.0.0)                
##  gtable                 0.3.0      2019-03-25 [1] CRAN (R 4.0.0)                
##  highr                  0.8        2019-03-20 [2] CRAN (R 4.0.0)                
##  Hmisc                  4.4-0      2020-03-23 [1] CRAN (R 4.0.0)                
##  hms                    0.5.3      2020-01-08 [1] CRAN (R 4.0.0)                
##  htmlTable              1.13.3     2019-12-04 [1] CRAN (R 4.0.0)                
##  htmltools              0.4.0      2019-10-04 [2] CRAN (R 4.0.0)                
##  htmlwidgets            1.5.1      2019-10-08 [2] CRAN (R 4.0.0)                
##  httr                   1.4.1      2019-08-05 [2] CRAN (R 4.0.0)                
##  IRanges              * 2.23.4     2020-05-03 [1] Bioconductor                  
##  iterators              1.0.12     2019-07-26 [1] CRAN (R 4.0.0)                
##  jpeg                   0.1-8.1    2019-10-24 [1] CRAN (R 4.0.0)                
##  jsonlite               1.6.1      2020-02-02 [2] CRAN (R 4.0.0)                
##  knitcitations        * 1.0.10     2019-09-15 [1] CRAN (R 4.0.0)                
##  knitr                * 1.28       2020-02-06 [2] CRAN (R 4.0.0)                
##  knitrBootstrap         1.0.2      2018-05-24 [1] CRAN (R 4.0.0)                
##  labeling               0.3        2014-08-23 [1] CRAN (R 4.0.0)                
##  lattice                0.20-41    2020-04-02 [3] CRAN (R 4.0.0)                
##  latticeExtra           0.6-29     2019-12-19 [1] CRAN (R 4.0.0)                
##  lazyeval               0.2.2      2019-03-15 [2] CRAN (R 4.0.0)                
##  lifecycle              0.2.0      2020-03-06 [2] CRAN (R 4.0.0)                
##  limma                  3.45.0     2020-04-27 [1] Bioconductor                  
##  locfit                 1.5-9.4    2020-03-25 [1] CRAN (R 4.0.0)                
##  lubridate              1.7.8      2020-04-06 [1] CRAN (R 4.0.0)                
##  magrittr               1.5        2014-11-22 [2] CRAN (R 4.0.0)                
##  markdown               1.1        2019-08-07 [2] CRAN (R 4.0.0)                
##  MASS                   7.3-51.6   2020-04-26 [3] CRAN (R 4.0.0)                
##  Matrix                 1.2-18     2019-11-27 [3] CRAN (R 4.0.0)                
##  matrixStats          * 0.56.0     2020-03-13 [1] CRAN (R 4.0.0)                
##  memoise                1.1.0      2017-04-21 [2] CRAN (R 4.0.0)                
##  munsell                0.5.0      2018-06-12 [1] CRAN (R 4.0.0)                
##  nnet                   7.3-14     2020-04-26 [3] CRAN (R 4.0.0)                
##  openssl                1.4.1      2019-07-18 [2] CRAN (R 4.0.0)                
##  org.Hs.eg.db         * 3.11.1     2020-05-13 [1] Bioconductor                  
##  pheatmap             * 1.0.12     2019-01-04 [1] CRAN (R 4.0.0)                
##  pillar                 1.4.4      2020-05-05 [2] CRAN (R 4.0.0)                
##  pkgconfig              2.0.3      2019-09-22 [2] CRAN (R 4.0.0)                
##  pkgdown                1.5.1.9000 2020-05-13 [1] Github (r-lib/pkgdown@c181e5c)
##  plyr                   1.8.6      2020-03-03 [1] CRAN (R 4.0.0)                
##  png                    0.1-7      2013-12-03 [1] CRAN (R 4.0.0)                
##  prettyunits            1.1.1      2020-01-24 [2] CRAN (R 4.0.0)                
##  progress               1.2.2      2019-05-16 [1] CRAN (R 4.0.0)                
##  ProtGenerics           1.21.0     2020-04-27 [1] Bioconductor                  
##  purrr                  0.3.4      2020-04-17 [2] CRAN (R 4.0.0)                
##  qvalue                 2.21.0     2020-04-27 [1] Bioconductor                  
##  R6                     2.4.1      2019-11-12 [2] CRAN (R 4.0.0)                
##  rappdirs               0.3.1      2016-03-28 [1] CRAN (R 4.0.0)                
##  RColorBrewer         * 1.1-2      2014-12-07 [1] CRAN (R 4.0.0)                
##  Rcpp                   1.0.4.6    2020-04-09 [2] CRAN (R 4.0.0)                
##  RCurl                  1.98-1.2   2020-04-18 [1] CRAN (R 4.0.0)                
##  readr                  1.3.1      2018-12-21 [1] CRAN (R 4.0.0)                
##  recount              * 1.15.1     2020-05-13 [1] Bioconductor                  
##  RefManageR             1.2.12     2019-04-03 [1] CRAN (R 4.0.0)                
##  regionReport         * 1.23.2     2020-05-10 [1] Bioconductor                  
##  rentrez                1.2.2      2019-05-02 [1] CRAN (R 4.0.0)                
##  reshape2               1.4.4      2020-04-09 [1] CRAN (R 4.0.0)                
##  rlang                  0.4.6      2020-05-02 [2] CRAN (R 4.0.0)                
##  rmarkdown              2.1        2020-01-20 [1] CRAN (R 4.0.0)                
##  rngtools               1.5        2020-01-23 [1] CRAN (R 4.0.0)                
##  rpart                  4.1-15     2019-04-12 [3] CRAN (R 4.0.0)                
##  rprojroot              1.3-2      2018-01-03 [2] CRAN (R 4.0.0)                
##  Rsamtools              2.5.0      2020-04-27 [1] Bioconductor                  
##  RSQLite                2.2.0      2020-01-07 [1] CRAN (R 4.0.0)                
##  rstudioapi             0.11       2020-02-07 [2] CRAN (R 4.0.0)                
##  rtracklayer            1.49.1     2020-05-02 [1] Bioconductor                  
##  S4Vectors            * 0.27.5     2020-05-06 [1] Bioconductor                  
##  scales                 1.1.1      2020-05-11 [1] CRAN (R 4.0.0)                
##  sessioninfo          * 1.1.1      2018-11-05 [2] CRAN (R 4.0.0)                
##  stringi                1.4.6      2020-02-17 [2] CRAN (R 4.0.0)                
##  stringr                1.4.0      2019-02-10 [2] CRAN (R 4.0.0)                
##  SummarizedExperiment * 1.19.2     2020-05-01 [1] Bioconductor                  
##  survival               3.1-12     2020-04-10 [3] CRAN (R 4.0.0)                
##  tibble                 3.0.1      2020-04-20 [2] CRAN (R 4.0.0)                
##  tidyr                  1.0.3      2020-05-07 [1] CRAN (R 4.0.0)                
##  tidyselect             1.1.0      2020-05-11 [1] CRAN (R 4.0.0)                
##  VariantAnnotation      1.35.0     2020-04-27 [1] Bioconductor                  
##  vctrs                  0.3.0      2020-05-11 [1] CRAN (R 4.0.0)                
##  withr                  2.2.0      2020-04-20 [2] CRAN (R 4.0.0)                
##  xfun                   0.13       2020-04-13 [2] CRAN (R 4.0.0)                
##  XML                    3.99-0.3   2020-01-20 [1] CRAN (R 4.0.0)                
##  xml2                   1.3.2      2020-04-23 [2] CRAN (R 4.0.0)                
##  xtable                 1.8-4      2019-04-21 [1] CRAN (R 4.0.0)                
##  XVector                0.29.0     2020-04-27 [1] Bioconductor                  
##  yaml                   2.2.1      2020-02-01 [2] CRAN (R 4.0.0)                
##  zlibbioc               1.35.0     2020-04-27 [1] Bioconductor                  
## 
## [1] /usr/local/lib/R/host-site-library
## [2] /usr/local/lib/R/site-library
## [3] /usr/local/lib/R/library</code></pre>
</div>
<div id="bibliography" class="section level1">
<h1 class="hasAnchor">
<a href="#bibliography" class="anchor"></a>Bibliography</h1>
<p>This vignette was generated using <em><a href="https://bioconductor.org/packages/3.12/BiocStyle">BiocStyle</a></em> (<a href="https://github.com/Bioconductor/BiocStyle">Oleś, Morgan, and Huber, 2020</a>) with <em><a href="https://CRAN.R-project.org/package=knitr">knitr</a></em> (<a href="http://www.crcpress.com/product/isbn/9781466561595">Xie, 2014</a>) and <em><a href="https://CRAN.R-project.org/package=rmarkdown">rmarkdown</a></em> (<a href="https://github.com/rstudio/rmarkdown">Allaire, Xie, McPherson, Luraschi, et al., 2020</a>) running behind the scenes.</p>
<p>Citations made with <em><a href="https://CRAN.R-project.org/package=knitcitations">knitcitations</a></em> (<a href="https://CRAN.R-project.org/package=knitcitations">Boettiger, 2019</a>).</p>
<p>[1] J. Allaire, Y. Xie, J. McPherson, J. Luraschi, et al. <em>rmarkdown: Dynamic Documents for R</em>. R package version 2.1. 2020. &lt;URL: <a href="https://github.com/rstudio/rmarkdown" class="uri">https://github.com/rstudio/rmarkdown</a>&gt;.</p>
<p>[2] S. Arora, M. Morgan, M. Carlson, and H. Pagès. <em>GenomeInfoDb: Utilities for manipulating chromosome and other ‘seqname’ identifiers</em>. 2017. DOI: 10.18129/B9.bioc.GenomeInfoDb.</p>
<p>[3] C. Boettiger. <em>knitcitations: Citations for ‘Knitr’ Markdown Files</em>. R package version 1.0.10. 2019. &lt;URL: <a href="https://CRAN.R-project.org/package=knitcitations" class="uri">https://CRAN.R-project.org/package=knitcitations</a>&gt;.</p>
<p>[4] M. Carlson. <em>org.Hs.eg.db: Genome wide annotation for Human</em>. R package version 3.11.1. 2020.</p>
<p>[5] W. Chang. <em>downloader: Download Files over HTTP and HTTPS</em>. R package version 0.4. 2015. &lt;URL: <a href="https://CRAN.R-project.org/package=downloader" class="uri">https://CRAN.R-project.org/package=downloader</a>&gt;.</p>
<p>[6] L. Collado-Torres, A. E. Jaffe, and J. T. Leek. “regionReport: Interactive reports for region-level and feature-level genomic analyses [version2; referees: 2 approved, 1 approved with reservations]”. In: <em>F1000Research</em> 4 (2016), p. 105. DOI: 10.12688/f1000research.6379.2. &lt;URL: <a href="http://f1000research.com/articles/4-105/v2" class="uri">http://f1000research.com/articles/4-105/v2</a>&gt;.</p>
<p>[7] L. Collado-Torres, A. Nellore, A. C. Frazee, C. Wilks, et al. “Flexible expressed region analysis for RNA-seq with derfinder”. In: <em>Nucl. Acids Res.</em> (2017). DOI: 10.1093/nar/gkw852. &lt;URL: <a href="http://nar.oxfordjournals.org/content/early/2016/09/29/nar.gkw852" class="uri">http://nar.oxfordjournals.org/content/early/2016/09/29/nar.gkw852</a>&gt;.</p>
<p>[8] L. Collado-Torres, A. Nellore, K. Kammers, S. E. Ellis, et al. “Reproducible RNA-seq analysis using recount2”. In: <em>Nature Biotechnology</em> (2017). DOI: 10.1038/nbt.3838. &lt;URL: <a href="http://www.nature.com/nbt/journal/v35/n4/full/nbt.3838.html" class="uri">http://www.nature.com/nbt/journal/v35/n4/full/nbt.3838.html</a>&gt;.</p>
<p>[9] G. Csárdi, R. core, H. Wickham, W. Chang, et al. <em>sessioninfo: R Session Information</em>. R package version 1.1.1. 2018. &lt;URL: <a href="https://CRAN.R-project.org/package=sessioninfo" class="uri">https://CRAN.R-project.org/package=sessioninfo</a>&gt;.</p>
<p>[10] S. Davis and P. Meltzer. “GEOquery: a bridge between the Gene Expression Omnibus (GEO) and BioConductor”. In: <em>Bioinformatics</em> 14 (2007), pp. 1846-1847.</p>
<p>[11] R. Kolde. <em>pheatmap: Pretty Heatmaps</em>. R package version 1.0.12. 2019. &lt;URL: <a href="https://CRAN.R-project.org/package=pheatmap" class="uri">https://CRAN.R-project.org/package=pheatmap</a>&gt;.</p>
<p>[12] M. Lawrence, R. Gentleman, and V. Carey. “rtracklayer: an R package for interfacing with genome browsers”. In: <em>Bioinformatics</em> 25 (2009), pp. 1841-1842. DOI: 10.1093/bioinformatics/btp328. &lt;URL: <a href="http://bioinformatics.oxfordjournals.org/content/25/14/1841.abstract" class="uri">http://bioinformatics.oxfordjournals.org/content/25/14/1841.abstract</a>&gt;.</p>
<p>[13] M. Lawrence, W. Huber, H. Pagès, P. Aboyoun, et al. “Software for Computing and Annotating Genomic Ranges”. In: <em>PLoS Computational Biology</em> 9 (8 2013). DOI: 10.1371/journal.pcbi.1003118. &lt;URL: <a href="http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003118" class="uri">http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003118</a>}.&gt;</p>
<p>[14] M. Lawrence, W. Huber, H. Pagès, P. Aboyoun, et al. “Software for Computing and Annotating Genomic Ranges”. In: <em>PLoS Computational Biology</em> 9 (8 2013). DOI: 10.1371/journal.pcbi.1003118. &lt;URL: <a href="http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003118" class="uri">http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003118</a>}.&gt;</p>
<p>[15] M. I. Love, W. Huber, and S. Anders. “Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2”. In: <em>Genome Biology</em> 15 (12 2014), p. 550. DOI: 10.1186/s13059-014-0550-8.</p>
<p>[16] M. Morgan, V. Obenchain, J. Hester, and H. Pagès. <em>SummarizedExperiment: SummarizedExperiment container</em>. 2017. DOI: 10.18129/B9.bioc.SummarizedExperiment.</p>
<p>[17] M. Morgan, V. Obenchain, M. Lang, R. Thompson, et al. <em>BiocParallel: Bioconductor facilities for parallel evaluation</em>. R package version 1.23.0. 2020. &lt;URL: <a href="https://github.com/Bioconductor/BiocParallel" class="uri">https://github.com/Bioconductor/BiocParallel</a>&gt;.</p>
<p>[18] A. Oleś, M. Morgan, and W. Huber. <em>BiocStyle: Standard styles for vignettes and other Bioconductor documents</em>. R package version 2.17.0. 2020. &lt;URL: <a href="https://github.com/Bioconductor/BiocStyle" class="uri">https://github.com/Bioconductor/BiocStyle</a>&gt;.</p>
<p>[19] H. Pagès, M. Carlson, S. Falcon, and N. Li. <em>AnnotationDbi: Annotation Database Interface</em>. 2017. DOI: 10.18129/B9.bioc.AnnotationDbi.</p>
<p>[20] H. Pagès, M. Lawrence, and P. Aboyoun. <em>S4Vectors: S4 implementation of vector-like and list-like objects</em>. 2017. DOI: 10.18129/B9.bioc.S4Vectors.</p>
<p>[21] R Core Team. <em>R: A Language and Environment for Statistical Computing</em>. R Foundation for Statistical Computing. Vienna, Austria, 2020. &lt;URL: <a href="https://www.R-project.org/" class="uri">https://www.R-project.org/</a>&gt;.</p>
<p>[22] J. Rainer. <em>EnsDb.Hsapiens.v79: Ensembl based annotation package</em>. R package version 2.99.0. 2017.</p>
<p>[23] D. Temple Lang. <em>RCurl: General Network (HTTP/FTP/…) Client Interface for R</em>. R package version 1.98-1.2. 2020. &lt;URL: <a href="https://CRAN.R-project.org/package=RCurl" class="uri">https://CRAN.R-project.org/package=RCurl</a>&gt;.</p>
<p>[24] H. Wickham. <em>ggplot2: Elegant Graphics for Data Analysis</em>. Springer-Verlag New York, 2016. ISBN: 978-3-319-24277-4. &lt;URL: <a href="https://ggplot2.tidyverse.org" class="uri">https://ggplot2.tidyverse.org</a>&gt;.</p>
<p>[25] H. Wickham. “testthat: Get Started with Testing”. In: <em>The R Journal</em> 3 (2011), pp. 5-10. &lt;URL: <a href="https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf" class="uri">https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf</a>&gt;.</p>
<p>[26] D. J. Winter. “rentrez: an R package for the NCBI eUtils API”. In: <em>The R Journal</em> 9 (2 2017), pp. 520-526.</p>
<p>[27] Y. Xie. “knitr: A Comprehensive Tool for Reproducible Research in R”. In: <em>Implementing Reproducible Computational Research</em>. Ed. by V. Stodden, F. Leisch and R. D. Peng. ISBN 978-1466561595. Chapman and Hall/CRC, 2014. &lt;URL: <a href="http://www.crcpress.com/product/isbn/9781466561595" class="uri">http://www.crcpress.com/product/isbn/9781466561595</a>&gt;.</p>
<p>[28] Y. Xie, J. Cheng, and X. Tan. <em>DT: A Wrapper of the JavaScript Library ‘DataTables’</em>. R package version 0.13. 2020. &lt;URL: <a href="https://CRAN.R-project.org/package=DT" class="uri">https://CRAN.R-project.org/package=DT</a>&gt;.</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Leonardo Collado-Torres, Jeffrey T. Leek.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.9000.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
